{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Using TensorFlow backend.\n"
     ]
    }
   ],
   "source": [
    "from __future__ import print_function\n",
    "import keras\n",
    "from keras.models import Sequential, Model, load_model\n",
    "from keras import backend as K\n",
    "\n",
    "import tensorflow as tf\n",
    "\n",
    "import os\n",
    "import pandas as pd\n",
    "\n",
    "import numpy as np\n",
    "\n",
    "import matplotlib.pyplot as plt\n",
    "import matplotlib.cm as cm\n",
    "\n",
    "#import aparent.visualization as vis\n",
    "\n",
    "#from aparent_predictor import *\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<h2>Load APARENT model</h2>\n",
    "<br/>\n",
    "-- Load APARENT, which has been trained to predict the isoform abundance and cut profile of a proximal PAS given a fixed background distal PAS (trained on random 3' UTR APA MPRA data).<br/>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WARNING:tensorflow:From /home/ubuntu/anaconda3/envs/tensorflow_p36/lib/python3.6/site-packages/tensorflow/python/framework/op_def_library.py:263: colocate_with (from tensorflow.python.framework.ops) is deprecated and will be removed in a future version.\n",
      "Instructions for updating:\n",
      "Colocations handled automatically by placer.\n",
      "WARNING:tensorflow:From /home/ubuntu/anaconda3/envs/tensorflow_p36/lib/python3.6/site-packages/keras/backend/tensorflow_backend.py:3445: calling dropout (from tensorflow.python.ops.nn_ops) with keep_prob is deprecated and will be removed in a future version.\n",
      "Instructions for updating:\n",
      "Please use `rate` instead of `keep_prob`. Rate should be set to `rate = 1 - keep_prob`.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/ubuntu/anaconda3/envs/tensorflow_p36/lib/python3.6/site-packages/keras/engine/saving.py:292: UserWarning: No training configuration found in save file: the model was *not* compiled. Compile it manually.\n",
      "  warnings.warn('No training configuration found in save file: '\n"
     ]
    }
   ],
   "source": [
    "#Load APARENT Resnet\n",
    "\n",
    "#model_name = 'aparent_all_libs_resnet_no_clinvar_wt_ep_5'\n",
    "#save_dir = os.path.join(os.getcwd(), './')\n",
    "\n",
    "model_name = 'aparent_all_libs_resnet_no_clinvar_wt_ep_5_var_batch_size'\n",
    "save_dir = os.path.join(os.getcwd(), '../../autoscrambler/analysis/apa/saved_models')\n",
    "\n",
    "model_path = os.path.join(save_dir, model_name + '.h5')\n",
    "\n",
    "aparent_model = load_model(model_path)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "len(polyadb_df) = 175451\n"
     ]
    }
   ],
   "source": [
    "#Load APADB Data and filter on targeted genes\n",
    "\n",
    "#genes = ['RUNX1', 'CEBPA', 'GATA2', 'ANKRD26', 'DDX41', 'ETV6', 'PTEN', 'BRCA1', 'BRCA2', 'TP53', 'APC', 'ATM', 'PALB2', 'MSH2', 'MLH1', 'MSH6', 'PMS2', 'MUTYH']\n",
    "\n",
    "polyadb_df = pd.read_csv('polyadb_processed.csv', sep=',')\n",
    "\n",
    "#polyadb_df = polyadb_df.loc[polyadb_df['gene'].isin(genes)].reset_index(drop=True).copy()\n",
    "polyadb_df = polyadb_df.loc[((~polyadb_df['gene'].isnull()) & (polyadb_df['gene'] != 'na')) & (polyadb_df['pas'] != -1)].reset_index(drop=True).copy()\n",
    "\n",
    "print('len(polyadb_df) = ' + str(len(polyadb_df)))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "polyadb_df_minus = polyadb_df.query(\"strand == '-'\").copy().reset_index(drop=True)\n",
    "polyadb_df_plus = polyadb_df.query(\"strand == '+'\").copy().reset_index(drop=True)\n",
    "\n",
    "polyadb_df_minus = polyadb_df_minus.sort_values(by='pas_pos', ascending=False).copy().reset_index(drop=True)\n",
    "polyadb_df_plus = polyadb_df_plus.sort_values(by='pas_pos', ascending=True).copy().reset_index(drop=True)\n",
    "\n",
    "new_gene_id_list_plus = []\n",
    "sitenum_list_plus = []\n",
    "gene_id_dict = {}\n",
    "for _, row in polyadb_df_plus.iterrows() :\n",
    "\n",
    "    gene = row['gene']\n",
    "\n",
    "    if gene not in gene_id_dict :\n",
    "        gene_id_dict[gene] = 0\n",
    "\n",
    "    gene_id_dict[gene] += 1\n",
    "\n",
    "    new_gene_id_list_plus.append(gene + \".\" + str(gene_id_dict[gene]))\n",
    "    sitenum_list_plus.append(gene_id_dict[gene])\n",
    "\n",
    "polyadb_df_plus['gene_id'] = new_gene_id_list_plus\n",
    "polyadb_df_plus['sitenum'] = sitenum_list_plus\n",
    "\n",
    "new_gene_id_list_minus = []\n",
    "sitenum_list_minus = []\n",
    "gene_id_dict = {}\n",
    "for _, row in polyadb_df_minus.iterrows() :\n",
    "\n",
    "    gene = row['gene']\n",
    "\n",
    "    if gene not in gene_id_dict :\n",
    "        gene_id_dict[gene] = 0\n",
    "\n",
    "    gene_id_dict[gene] += 1\n",
    "\n",
    "    new_gene_id_list_minus.append(gene + \".\" + str(gene_id_dict[gene]))\n",
    "    sitenum_list_minus.append(gene_id_dict[gene])\n",
    "\n",
    "polyadb_df_minus['gene_id'] = new_gene_id_list_minus\n",
    "polyadb_df_minus['sitenum'] = sitenum_list_minus\n",
    "\n",
    "polyadb_df = pd.concat([polyadb_df_plus, polyadb_df_minus])\n",
    "\n",
    "polyadb_df = polyadb_df.sort_values(by=['gene', 'sitenum'], ascending=True).reset_index(drop=True).copy()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "len(polyadb_df) = 175451\n"
     ]
    }
   ],
   "source": [
    "\n",
    "print('len(polyadb_df) = ' + str(len(polyadb_df)))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "class OneHotEncoder :\n",
    "    def __init__(self, seq_length=100, default_fill_value=0) :\n",
    "        self.seq_length = seq_length\n",
    "        self.default_fill_value = default_fill_value\n",
    "        self.encode_map = {\n",
    "            'A' : 0,\n",
    "            'C' : 1,\n",
    "            'G' : 2,\n",
    "            'T' : 3\n",
    "        }\n",
    "        self.decode_map = {\n",
    "            0 : 'A',\n",
    "            1 : 'C',\n",
    "            2 : 'G',\n",
    "            3 : 'T',\n",
    "            -1 : 'X'\n",
    "        }\n",
    "\n",
    "    def encode(self, seq) :\n",
    "        one_hot = np.zeros((self.seq_length, 4))\n",
    "        self.encode_inplace(seq, one_hot)\n",
    "\n",
    "        return one_hot\n",
    "\n",
    "    def encode_inplace(self, seq, encoding) :\n",
    "        for pos, nt in enumerate(list(seq)) :\n",
    "            if nt in self.encode_map :\n",
    "                encoding[pos, self.encode_map[nt]] = 1\n",
    "            elif self.default_fill_value != 0 :\n",
    "                encoding[pos, :] = self.default_fill_value\n",
    "\n",
    "    def __call__(self, seq) :\n",
    "        return self.encode(seq)\n",
    "\n",
    "def get_aparent_encoder(lib_bias=11) :\n",
    "    onehot_encoder = OneHotEncoder(205)\n",
    "\n",
    "    def encode_for_aparent(sequences) :\n",
    "        one_hots = np.concatenate([onehot_encoder(sequence)[None, None, ...] for sequence in sequences], axis=0)\n",
    "\n",
    "        fake_lib = np.zeros((len(sequences), 13))\n",
    "\n",
    "        if lib_bias is not None :\n",
    "            fake_lib[:, lib_bias] = 1.\n",
    "\n",
    "        return [\n",
    "            one_hots,\n",
    "            fake_lib\n",
    "        ]\n",
    "\n",
    "    return encode_for_aparent\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Predicting variants for PAS = A1BG.1 (0)\n",
      "Predicting variants for PAS = AAK1.10 (100)\n",
      "Predicting variants for PAS = AASDH.14 (200)\n",
      "Predicting variants for PAS = AC007663.1 (1200)\n",
      "Predicting variants for PAS = AC010542.2 (1300)\n",
      "Predicting variants for PAS = AC020893.1 (1400)\n",
      "Predicting variants for PAS = AC061975.1 (1500)\n",
      "Predicting variants for PAS = AC091551.6 (1600)\n",
      "Predicting variants for PAS = AC098934.2 (1700)\n",
      "Predicting variants for PAS = AC117498.1 (1800)\n",
      "Predicting variants for PAS = ACAA2.2 (1900)\n",
      "Predicting variants for PAS = ACAD8.11 (2000)\n",
      "Predicting variants for PAS = ACAT1.5 (2100)\n",
      "Predicting variants for PAS = ACER2.3 (2200)\n",
      "Predicting variants for PAS = ACO2.12 (2300)\n",
      "Predicting variants for PAS = ACOX1.21 (2400)\n",
      "Predicting variants for PAS = ACSBG1.3 (2500)\n",
      "Predicting variants for PAS = ACSM3.3 (2600)\n",
      "Predicting variants for PAS = ACTL6A.6 (2700)\n",
      "Predicting variants for PAS = ACTR2.7 (2800)\n",
      "Predicting variants for PAS = ACTR8.18 (2900)\n",
      "Predicting variants for PAS = ACYP2.25 (3000)\n",
      "Predicting variants for PAS = ADAM17.4 (3100)\n",
      "Predicting variants for PAS = ADAM9.9 (3200)\n",
      "Predicting variants for PAS = ADAMTS18.2 (3300)\n",
      "Predicting variants for PAS = ADAMTS9.4 (3400)\n",
      "Predicting variants for PAS = ADARB2.12 (3500)\n",
      "Predicting variants for PAS = ADCY1.8 (3600)\n",
      "Predicting variants for PAS = ADCYAP1R1.6 (3700)\n",
      "Predicting variants for PAS = ADGRA3.18 (3800)\n",
      "Predicting variants for PAS = ADGRG1.3 (3900)\n",
      "Predicting variants for PAS = ADGRL3.21 (4000)\n",
      "Predicting variants for PAS = ADK.7 (4100)\n",
      "Predicting variants for PAS = ADO.10 (4200)\n",
      "Predicting variants for PAS = ADSL.2 (4300)\n",
      "Predicting variants for PAS = AF274858.1 (4400)\n",
      "Predicting variants for PAS = AFF1.16 (4500)\n",
      "Predicting variants for PAS = AFG1L.16 (4600)\n",
      "Predicting variants for PAS = AGAP3.1 (4700)\n",
      "Predicting variants for PAS = AGFG1.35 (4800)\n",
      "Predicting variants for PAS = AGO1.13 (4900)\n",
      "Predicting variants for PAS = AGPAT3.8 (5000)\n",
      "Predicting variants for PAS = AGTPBP1.16 (5100)\n",
      "Predicting variants for PAS = AHNAK.2 (5200)\n",
      "Predicting variants for PAS = AIDA.28 (5300)\n",
      "Predicting variants for PAS = AJAP1.4 (5400)\n",
      "Predicting variants for PAS = AK6.9 (5500)\n",
      "Predicting variants for PAS = AKAP12.3 (5600)\n",
      "Predicting variants for PAS = AKAP8.6 (5700)\n",
      "Predicting variants for PAS = AKR1E2.10 (5800)\n",
      "Predicting variants for PAS = AL022237.1 (5900)\n",
      "Predicting variants for PAS = AL136452.1 (6000)\n",
      "Predicting variants for PAS = AL355916.1 (6100)\n",
      "Predicting variants for PAS = AL590143.1 (6200)\n",
      "Predicting variants for PAS = ALDH1A3.19 (6300)\n",
      "Predicting variants for PAS = ALDH5A1.9 (6400)\n",
      "Predicting variants for PAS = ALG10B.8 (6500)\n",
      "Predicting variants for PAS = ALG6.3 (6600)\n",
      "Predicting variants for PAS = ALKBH5.7 (6700)\n",
      "Predicting variants for PAS = ALPK1.5 (6800)\n",
      "Predicting variants for PAS = AMACR.2 (6900)\n",
      "Predicting variants for PAS = AMMECR1.7 (7000)\n",
      "Predicting variants for PAS = AMZ1.1 (7100)\n",
      "Predicting variants for PAS = ANAPC16.16 (7200)\n",
      "Predicting variants for PAS = ANGPTL2.7 (7300)\n",
      "Predicting variants for PAS = ANK3.44 (7400)\n",
      "Predicting variants for PAS = ANKH.14 (7500)\n",
      "Predicting variants for PAS = ANKMY1.5 (7600)\n",
      "Predicting variants for PAS = ANKRD12.13 (7700)\n",
      "Predicting variants for PAS = ANKRD17.25 (7800)\n",
      "Predicting variants for PAS = ANKRD26.22 (7900)\n",
      "Predicting variants for PAS = ANKRD36.5 (8000)\n",
      "Predicting variants for PAS = ANKRD49.4 (8100)\n",
      "Predicting variants for PAS = ANKS1B.11 (8200)\n",
      "Predicting variants for PAS = ANO2.10 (8300)\n",
      "Predicting variants for PAS = ANP32B.15 (8400)\n",
      "Predicting variants for PAS = ANXA1.8 (8500)\n",
      "Predicting variants for PAS = ANXA4.20 (8600)\n",
      "Predicting variants for PAS = AP001024.1 (8700)\n",
      "Predicting variants for PAS = AP1S2.8 (8800)\n",
      "Predicting variants for PAS = AP3B1.18 (8900)\n",
      "Predicting variants for PAS = AP4M1.3 (9000)\n",
      "Predicting variants for PAS = APBA3.5 (9100)\n",
      "Predicting variants for PAS = APEX1.4 (9200)\n",
      "Predicting variants for PAS = APOBEC2.1 (9300)\n",
      "Predicting variants for PAS = APOO.7 (9400)\n",
      "Predicting variants for PAS = APPL1.18 (9500)\n",
      "Predicting variants for PAS = AQR.2 (9600)\n",
      "Predicting variants for PAS = ARF1.7 (9700)\n",
      "Predicting variants for PAS = ARFGEF2.5 (9800)\n",
      "Predicting variants for PAS = ARHGAP1.2 (9900)\n",
      "Predicting variants for PAS = ARHGAP21.2 (10000)\n",
      "Predicting variants for PAS = ARHGAP29.11 (10100)\n",
      "Predicting variants for PAS = ARHGAP42.1 (10200)\n",
      "Predicting variants for PAS = ARHGEF10.8 (10300)\n",
      "Predicting variants for PAS = ARHGEF2.11 (10400)\n",
      "Predicting variants for PAS = ARHGEF40.7 (10500)\n",
      "Predicting variants for PAS = ARID2.10 (10600)\n",
      "Predicting variants for PAS = ARIH2.1 (10700)\n",
      "Predicting variants for PAS = ARL17B.2 (10800)\n",
      "Predicting variants for PAS = ARL5A.34 (10900)\n",
      "Predicting variants for PAS = ARL8B.5 (11000)\n",
      "Predicting variants for PAS = ARMC3.1 (11100)\n",
      "Predicting variants for PAS = ARMCX2.5 (11200)\n",
      "Predicting variants for PAS = ARNT.5 (11300)\n",
      "Predicting variants for PAS = ARPC3.8 (11400)\n",
      "Predicting variants for PAS = ARRDC1.6 (11500)\n",
      "Predicting variants for PAS = ARSG.25 (11600)\n",
      "Predicting variants for PAS = ASAH1.10 (11700)\n",
      "Predicting variants for PAS = ASB7.4 (11800)\n",
      "Predicting variants for PAS = ASH1L.2 (11900)\n",
      "Predicting variants for PAS = ASPH.11 (12000)\n",
      "Predicting variants for PAS = ASPSCR1.3 (12100)\n",
      "Predicting variants for PAS = ASXL2.15 (12200)\n",
      "Predicting variants for PAS = ATAD5.6 (12300)\n",
      "Predicting variants for PAS = ATF4.8 (12400)\n",
      "Predicting variants for PAS = ATG10.17 (12500)\n",
      "Predicting variants for PAS = ATG4A.3 (12600)\n",
      "Predicting variants for PAS = ATIC.16 (12700)\n",
      "Predicting variants for PAS = ATN1.9 (12800)\n",
      "Predicting variants for PAS = ATP11B.1 (12900)\n",
      "Predicting variants for PAS = ATP1A3.1 (13000)\n",
      "Predicting variants for PAS = ATP2B1.12 (13100)\n",
      "Predicting variants for PAS = ATP5A1.3 (13200)\n",
      "Predicting variants for PAS = ATP5G3.6 (13300)\n",
      "Predicting variants for PAS = ATP6AP1L.6 (13400)\n",
      "Predicting variants for PAS = ATP6V1A.8 (13500)\n",
      "Predicting variants for PAS = ATP6V1G1.1 (13600)\n",
      "Predicting variants for PAS = ATP8B3.1 (13700)\n",
      "Predicting variants for PAS = ATR.12 (13800)\n",
      "Predicting variants for PAS = ATXN1.23 (13900)\n",
      "Predicting variants for PAS = ATXN3.10 (14000)\n",
      "Predicting variants for PAS = AUH.34 (14100)\n",
      "Predicting variants for PAS = AVPR2.1 (14200)\n",
      "Predicting variants for PAS = B2M.8 (14300)\n",
      "Predicting variants for PAS = B3GAT3.2 (14400)\n",
      "Predicting variants for PAS = B4GALNT4.1 (14500)\n",
      "Predicting variants for PAS = BAALC.5 (14600)\n",
      "Predicting variants for PAS = BACH1.3 (14700)\n",
      "Predicting variants for PAS = BAG5.13 (14800)\n",
      "Predicting variants for PAS = BAP1.4 (14900)\n",
      "Predicting variants for PAS = BAZ2B.13 (15000)\n",
      "Predicting variants for PAS = BBS7.4 (15100)\n",
      "Predicting variants for PAS = BCAP29.25 (15200)\n",
      "Predicting variants for PAS = BCAS1.8 (15300)\n",
      "Predicting variants for PAS = BCCIP.3 (15400)\n",
      "Predicting variants for PAS = BCL2.1 (15500)\n",
      "Predicting variants for PAS = BCL2L2-PABPN1.25 (15600)\n",
      "Predicting variants for PAS = BCLAF1.23 (15700)\n",
      "Predicting variants for PAS = BDH2.7 (15800)\n",
      "Predicting variants for PAS = BEND6.10 (15900)\n",
      "Predicting variants for PAS = BFAR.15 (16000)\n",
      "Predicting variants for PAS = BICD1.20 (16100)\n",
      "Predicting variants for PAS = BIN2.1 (16200)\n",
      "Predicting variants for PAS = BLCAP.5 (16300)\n",
      "Predicting variants for PAS = BLZF1.1 (16400)\n",
      "Predicting variants for PAS = BMP7.5 (16500)\n",
      "Predicting variants for PAS = BMPR2.29 (16600)\n",
      "Predicting variants for PAS = BNIP2.10 (16700)\n",
      "Predicting variants for PAS = BORCS5.1 (16800)\n",
      "Predicting variants for PAS = BPTF.9 (16900)\n",
      "Predicting variants for PAS = BRCC3.19 (17000)\n",
      "Predicting variants for PAS = BRK1.6 (17200)\n",
      "Predicting variants for PAS = BRWD3.2 (17300)\n",
      "Predicting variants for PAS = BTBD1.5 (17400)\n",
      "Predicting variants for PAS = BTBD9.8 (17500)\n",
      "Predicting variants for PAS = BTN2A1.9 (17600)\n",
      "Predicting variants for PAS = BUB1B-PAK6.10 (17700)\n",
      "Predicting variants for PAS = C10orf143.2 (17800)\n",
      "Predicting variants for PAS = C11orf52.4 (17900)\n",
      "Predicting variants for PAS = C11orf80.3 (18000)\n",
      "Predicting variants for PAS = C12orf43.8 (18100)\n",
      "Predicting variants for PAS = C12orf76.5 (18200)\n",
      "Predicting variants for PAS = C15orf38-AP3S2.12 (18300)\n",
      "Predicting variants for PAS = C15orf62.1 (18400)\n",
      "Predicting variants for PAS = C16orf72.6 (18500)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Predicting variants for PAS = C17orf49.3 (18600)\n",
      "Predicting variants for PAS = C18orf25.10 (18700)\n",
      "Predicting variants for PAS = C19orf66.2 (18800)\n",
      "Predicting variants for PAS = C1QTNF3.10 (18900)\n",
      "Predicting variants for PAS = C1orf122.2 (19000)\n",
      "Predicting variants for PAS = C1orf228.4 (19100)\n",
      "Predicting variants for PAS = C20orf144.2 (19200)\n",
      "Predicting variants for PAS = C21orf58.24 (19300)\n",
      "Predicting variants for PAS = C2CD2.5 (19400)\n",
      "Predicting variants for PAS = C2orf27A.24 (19500)\n",
      "Predicting variants for PAS = C2orf76.6 (19600)\n",
      "Predicting variants for PAS = C3orf67.11 (19700)\n",
      "Predicting variants for PAS = C5AR1.1 (19800)\n",
      "Predicting variants for PAS = C5orf49.2 (19900)\n",
      "Predicting variants for PAS = C6orf120.8 (20000)\n",
      "Predicting variants for PAS = C6orf62.3 (20100)\n",
      "Predicting variants for PAS = C7orf49.15 (20200)\n",
      "Predicting variants for PAS = C8orf33.12 (20300)\n",
      "Predicting variants for PAS = C8orf88.10 (20400)\n",
      "Predicting variants for PAS = C9orf43.2 (20500)\n",
      "Predicting variants for PAS = CA4.5 (20600)\n",
      "Predicting variants for PAS = CAB39.3 (20700)\n",
      "Predicting variants for PAS = CACNA1A.4 (20800)\n",
      "Predicting variants for PAS = CACNA2D1.29 (20900)\n",
      "Predicting variants for PAS = CACNG7.4 (21000)\n",
      "Predicting variants for PAS = CADM2.14 (21100)\n",
      "Predicting variants for PAS = CALB1.5 (21200)\n",
      "Predicting variants for PAS = CALM1.2 (21300)\n",
      "Predicting variants for PAS = CALR.12 (21400)\n",
      "Predicting variants for PAS = CAMK2D.14 (21500)\n",
      "Predicting variants for PAS = CAMKMT.18 (21600)\n",
      "Predicting variants for PAS = CAMTA2.7 (21700)\n",
      "Predicting variants for PAS = CAP2.3 (21800)\n",
      "Predicting variants for PAS = CAPN6.1 (21900)\n",
      "Predicting variants for PAS = CAPS2.18 (22000)\n",
      "Predicting variants for PAS = CARD8.20 (22100)\n",
      "Predicting variants for PAS = CARS2.3 (22200)\n",
      "Predicting variants for PAS = CASK.15 (22300)\n",
      "Predicting variants for PAS = CASP9.7 (22400)\n",
      "Predicting variants for PAS = CATSPERE.3 (22500)\n",
      "Predicting variants for PAS = CBFB.4 (22600)\n",
      "Predicting variants for PAS = CBR3.3 (22700)\n",
      "Predicting variants for PAS = CBX5.5 (22800)\n",
      "Predicting variants for PAS = CC2D2A.6 (22900)\n",
      "Predicting variants for PAS = CCDC113.2 (23000)\n",
      "Predicting variants for PAS = CCDC127.7 (23100)\n",
      "Predicting variants for PAS = CCDC146.1 (23200)\n",
      "Predicting variants for PAS = CCDC171.6 (23300)\n",
      "Predicting variants for PAS = CCDC189.2 (23400)\n",
      "Predicting variants for PAS = CCDC34.1 (23500)\n",
      "Predicting variants for PAS = CCDC50.26 (23600)\n",
      "Predicting variants for PAS = CCDC66.15 (23700)\n",
      "Predicting variants for PAS = CCDC82.4 (23800)\n",
      "Predicting variants for PAS = CCDC88C.13 (23900)\n",
      "Predicting variants for PAS = CCDC97.8 (24000)\n",
      "Predicting variants for PAS = CCNB2.6 (24100)\n",
      "Predicting variants for PAS = CCNF.3 (24200)\n",
      "Predicting variants for PAS = CCNJL.2 (24300)\n",
      "Predicting variants for PAS = CCNY.7 (24400)\n",
      "Predicting variants for PAS = CCSER1.16 (24500)\n",
      "Predicting variants for PAS = CCT6P3.1 (24600)\n",
      "Predicting variants for PAS = CD177P1.1 (24700)\n",
      "Predicting variants for PAS = CD36.6 (24800)\n",
      "Predicting variants for PAS = CD55.1 (24900)\n",
      "Predicting variants for PAS = CD68.5 (25000)\n",
      "Predicting variants for PAS = CDC14B.4 (25100)\n",
      "Predicting variants for PAS = CDC34.7 (25200)\n",
      "Predicting variants for PAS = CDC42BPG.3 (25300)\n",
      "Predicting variants for PAS = CDC45.11 (25400)\n",
      "Predicting variants for PAS = CDCA2.3 (25500)\n",
      "Predicting variants for PAS = CDCP1.17 (25600)\n",
      "Predicting variants for PAS = CDH2.12 (25700)\n",
      "Predicting variants for PAS = CDH6.4 (25800)\n",
      "Predicting variants for PAS = CDK11A.1 (25900)\n",
      "Predicting variants for PAS = CDK17.14 (26000)\n",
      "Predicting variants for PAS = CDK5R1.2 (26100)\n",
      "Predicting variants for PAS = CDKAL1.7 (26200)\n",
      "Predicting variants for PAS = CDKN1B.14 (26300)\n",
      "Predicting variants for PAS = CDON.9 (26400)\n",
      "Predicting variants for PAS = CDX2.4 (26500)\n",
      "Predicting variants for PAS = CEBPZOS.18 (26600)\n",
      "Predicting variants for PAS = CELSR1.16 (26700)\n",
      "Predicting variants for PAS = CENPF.16 (26800)\n",
      "Predicting variants for PAS = CENPN.5 (26900)\n",
      "Predicting variants for PAS = CENPU.7 (27000)\n",
      "Predicting variants for PAS = CEP126.4 (27100)\n",
      "Predicting variants for PAS = CEP290.10 (27200)\n",
      "Predicting variants for PAS = CEP57.24 (27300)\n",
      "Predicting variants for PAS = CEP76.3 (27400)\n",
      "Predicting variants for PAS = CEP89.9 (27500)\n",
      "Predicting variants for PAS = CERS2.6 (27600)\n",
      "Predicting variants for PAS = CFAP20.2 (27700)\n",
      "Predicting variants for PAS = CFAP46.15 (27800)\n",
      "Predicting variants for PAS = CFAP70.9 (27900)\n",
      "Predicting variants for PAS = CFI.5 (28000)\n",
      "Predicting variants for PAS = CGA.8 (28100)\n",
      "Predicting variants for PAS = CHAF1A.9 (28200)\n",
      "Predicting variants for PAS = CHCHD7.6 (28300)\n",
      "Predicting variants for PAS = CHD6.8 (28400)\n",
      "Predicting variants for PAS = CHD9.38 (28500)\n",
      "Predicting variants for PAS = CHIC2.13 (28600)\n",
      "Predicting variants for PAS = CHMP2A.1 (28700)\n",
      "Predicting variants for PAS = CHN2.17 (28800)\n",
      "Predicting variants for PAS = CHRM3.9 (28900)\n",
      "Predicting variants for PAS = CHST11.4 (29000)\n",
      "Predicting variants for PAS = CHST8.6 (29100)\n",
      "Predicting variants for PAS = CHURC1-FNTB.9 (29200)\n",
      "Predicting variants for PAS = CIITA.7 (29300)\n",
      "Predicting variants for PAS = CISD1.7 (29400)\n",
      "Predicting variants for PAS = CKAP2.4 (29500)\n",
      "Predicting variants for PAS = CLASP1.9 (29600)\n",
      "Predicting variants for PAS = CLCN3.13 (29700)\n",
      "Predicting variants for PAS = CLDN14.2 (29800)\n",
      "Predicting variants for PAS = CLEC1A.2 (29900)\n",
      "Predicting variants for PAS = CLIC4.22 (30000)\n",
      "Predicting variants for PAS = CLK4.8 (30100)\n",
      "Predicting variants for PAS = CLNS1A.7 (30200)\n",
      "Predicting variants for PAS = CLRN3.1 (30300)\n",
      "Predicting variants for PAS = CLTC.31 (30400)\n",
      "Predicting variants for PAS = CLVS2.7 (30500)\n",
      "Predicting variants for PAS = CMC2.13 (30600)\n",
      "Predicting variants for PAS = CMTM6.9 (30700)\n",
      "Predicting variants for PAS = CNGA1.7 (30800)\n",
      "Predicting variants for PAS = CNKSR2.18 (30900)\n",
      "Predicting variants for PAS = CNOT1.15 (31000)\n",
      "Predicting variants for PAS = CNOT4.24 (31100)\n",
      "Predicting variants for PAS = CNPPD1.4 (31200)\n",
      "Predicting variants for PAS = CNTFR.5 (31300)\n",
      "Predicting variants for PAS = CNTNAP2.11 (31400)\n",
      "Predicting variants for PAS = COA1.50 (31500)\n",
      "Predicting variants for PAS = COBLL1.9 (31600)\n",
      "Predicting variants for PAS = COG6.4 (31700)\n",
      "Predicting variants for PAS = COL15A1.9 (31800)\n",
      "Predicting variants for PAS = COL25A1.11 (31900)\n",
      "Predicting variants for PAS = COL4A2.30 (32000)\n",
      "Predicting variants for PAS = COL5A1.15 (32100)\n",
      "Predicting variants for PAS = COL9A2.3 (32200)\n",
      "Predicting variants for PAS = COLQ.2 (32300)\n",
      "Predicting variants for PAS = COMMD6.6 (32400)\n",
      "Predicting variants for PAS = COPG1.5 (32500)\n",
      "Predicting variants for PAS = COPS8.6 (32600)\n",
      "Predicting variants for PAS = COQ7.10 (32700)\n",
      "Predicting variants for PAS = CORO7-PAM16.10 (32800)\n",
      "Predicting variants for PAS = COX17.1 (32900)\n",
      "Predicting variants for PAS = COX6B2.4 (33000)\n",
      "Predicting variants for PAS = CPA4.7 (33100)\n",
      "Predicting variants for PAS = CPEB4.7 (33200)\n",
      "Predicting variants for PAS = CPNE3.5 (33300)\n",
      "Predicting variants for PAS = CPS1.11 (33400)\n",
      "Predicting variants for PAS = CPSF6.9 (33500)\n",
      "Predicting variants for PAS = CPXM1.2 (33600)\n",
      "Predicting variants for PAS = CRBN.9 (33700)\n",
      "Predicting variants for PAS = CREB3L1.9 (33800)\n",
      "Predicting variants for PAS = CREBRF.10 (33900)\n",
      "Predicting variants for PAS = CREM.26 (34000)\n",
      "Predicting variants for PAS = CRK.1 (34100)\n",
      "Predicting variants for PAS = CRNKL1.11 (34200)\n",
      "Predicting variants for PAS = CRTAP.24 (34300)\n",
      "Predicting variants for PAS = CRYBG3.1 (34400)\n",
      "Predicting variants for PAS = CSAG4.1 (34500)\n",
      "Predicting variants for PAS = CSMD1.3 (34600)\n",
      "Predicting variants for PAS = CSNK1A1.44 (34700)\n",
      "Predicting variants for PAS = CSNK2A2.10 (34800)\n",
      "Predicting variants for PAS = CSTF1.3 (34900)\n",
      "Predicting variants for PAS = CTBP2.17 (35000)\n",
      "Predicting variants for PAS = CTDSPL2.2 (35100)\n",
      "Predicting variants for PAS = CTNNA1.21 (35200)\n",
      "Predicting variants for PAS = CTNS.3 (35300)\n",
      "Predicting variants for PAS = CTSD.3 (35400)\n",
      "Predicting variants for PAS = CTTNBP2NL.3 (35500)\n",
      "Predicting variants for PAS = CUL3.3 (35600)\n",
      "Predicting variants for PAS = CUTC.5 (35700)\n",
      "Predicting variants for PAS = CWC27.33 (35800)\n",
      "Predicting variants for PAS = CXCR2.1 (35900)\n",
      "Predicting variants for PAS = CYB5B.11 (36000)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Predicting variants for PAS = CYBRD1.1 (36100)\n",
      "Predicting variants for PAS = CYP11A1.3 (36200)\n",
      "Predicting variants for PAS = CYP2C9.2 (36300)\n",
      "Predicting variants for PAS = CYP4V2.10 (36400)\n",
      "Predicting variants for PAS = CYTH3.7 (36500)\n",
      "Predicting variants for PAS = DAB2IP.11 (36600)\n",
      "Predicting variants for PAS = DAP3.7 (36700)\n",
      "Predicting variants for PAS = DAZAP1.11 (36800)\n",
      "Predicting variants for PAS = DBNL.23 (36900)\n",
      "Predicting variants for PAS = DCAF12.5 (37000)\n",
      "Predicting variants for PAS = DCAF6.16 (37100)\n",
      "Predicting variants for PAS = DCBLD2.39 (37200)\n",
      "Predicting variants for PAS = DCLRE1C.6 (37300)\n",
      "Predicting variants for PAS = DCT.4 (37400)\n",
      "Predicting variants for PAS = DCTN6.4 (37500)\n",
      "Predicting variants for PAS = DCUN1D5.1 (37600)\n",
      "Predicting variants for PAS = DDHD1.10 (37700)\n",
      "Predicting variants for PAS = DDR1.6 (37800)\n",
      "Predicting variants for PAS = DDX19A.10 (37900)\n",
      "Predicting variants for PAS = DDX31.9 (38000)\n",
      "Predicting variants for PAS = DDX47.6 (38100)\n",
      "Predicting variants for PAS = DDX59.5 (38200)\n",
      "Predicting variants for PAS = DEDD.11 (38300)\n",
      "Predicting variants for PAS = DENND1A.25 (38400)\n",
      "Predicting variants for PAS = DENND4A.7 (38500)\n",
      "Predicting variants for PAS = DENND6A.7 (38600)\n",
      "Predicting variants for PAS = DEPDC5.4 (38700)\n",
      "Predicting variants for PAS = DEXI.5 (38800)\n",
      "Predicting variants for PAS = DGCR8.3 (38900)\n",
      "Predicting variants for PAS = DGKI.6 (39000)\n",
      "Predicting variants for PAS = DHFR.10 (39100)\n",
      "Predicting variants for PAS = DHRS7.11 (39200)\n",
      "Predicting variants for PAS = DHX33.11 (39300)\n",
      "Predicting variants for PAS = DHX9.2 (39400)\n",
      "Predicting variants for PAS = DIDO1.2 (39500)\n",
      "Predicting variants for PAS = DIP2A.9 (39600)\n",
      "Predicting variants for PAS = DIS3.9 (39700)\n",
      "Predicting variants for PAS = DKFZP586I1420.7 (39800)\n",
      "Predicting variants for PAS = DLD.20 (39900)\n",
      "Predicting variants for PAS = DLG5.11 (40000)\n",
      "Predicting variants for PAS = DLST.8 (40100)\n",
      "Predicting variants for PAS = DMKN.4 (40200)\n",
      "Predicting variants for PAS = DNA2.2 (40300)\n",
      "Predicting variants for PAS = DNAH10.5 (40400)\n",
      "Predicting variants for PAS = DNAI1.2 (40500)\n",
      "Predicting variants for PAS = DNAJB14.8 (40600)\n",
      "Predicting variants for PAS = DNAJC10.14 (40700)\n",
      "Predicting variants for PAS = DNAJC16.8 (40800)\n",
      "Predicting variants for PAS = DNAJC22.8 (40900)\n",
      "Predicting variants for PAS = DNAJC5.10 (41000)\n",
      "Predicting variants for PAS = DNASE1L2.1 (41100)\n",
      "Predicting variants for PAS = DNM1L.22 (41200)\n",
      "Predicting variants for PAS = DNPH1.4 (41300)\n",
      "Predicting variants for PAS = DOCK3.9 (41400)\n",
      "Predicting variants for PAS = DOHH.2 (41500)\n",
      "Predicting variants for PAS = DPAGT1.1 (41600)\n",
      "Predicting variants for PAS = DPM1.2 (41700)\n",
      "Predicting variants for PAS = DPPA4.3 (41800)\n",
      "Predicting variants for PAS = DPY19L4.8 (41900)\n",
      "Predicting variants for PAS = DR1.6 (42000)\n",
      "Predicting variants for PAS = DRG1.3 (42100)\n",
      "Predicting variants for PAS = DSCAM.8 (42200)\n",
      "Predicting variants for PAS = DSEL.20 (42300)\n",
      "Predicting variants for PAS = DSTN.18 (42400)\n",
      "Predicting variants for PAS = DTNB.34 (42500)\n",
      "Predicting variants for PAS = DUS2.6 (42600)\n",
      "Predicting variants for PAS = DUSP22.4 (42700)\n",
      "Predicting variants for PAS = DUXAP10.3 (42800)\n",
      "Predicting variants for PAS = DYNC1LI1.6 (42900)\n",
      "Predicting variants for PAS = DYNLL2.17 (43000)\n",
      "Predicting variants for PAS = DYRK4.7 (43100)\n",
      "Predicting variants for PAS = E2F3.5 (43200)\n",
      "Predicting variants for PAS = EBF1.1 (43300)\n",
      "Predicting variants for PAS = ECE1.2 (43400)\n",
      "Predicting variants for PAS = ECM2.3 (43500)\n",
      "Predicting variants for PAS = EDIL3.7 (43600)\n",
      "Predicting variants for PAS = EEF1A1.13 (43700)\n",
      "Predicting variants for PAS = EEF1D.7 (43800)\n",
      "Predicting variants for PAS = EEF2K.19 (43900)\n",
      "Predicting variants for PAS = EFCAB14.14 (44000)\n",
      "Predicting variants for PAS = EFEMP2.8 (44100)\n",
      "Predicting variants for PAS = EFNB2.6 (44200)\n",
      "Predicting variants for PAS = EGFR.5 (44300)\n",
      "Predicting variants for PAS = EGLN3.17 (44400)\n",
      "Predicting variants for PAS = EHD4.10 (44500)\n",
      "Predicting variants for PAS = EID2.5 (44600)\n",
      "Predicting variants for PAS = EIF2AK1.4 (44700)\n",
      "Predicting variants for PAS = EIF2B3.7 (44800)\n",
      "Predicting variants for PAS = EIF2S3.28 (44900)\n",
      "Predicting variants for PAS = EIF3G.7 (45000)\n",
      "Predicting variants for PAS = EIF4A1.12 (45100)\n",
      "Predicting variants for PAS = EIF4E2.10 (45200)\n",
      "Predicting variants for PAS = EIF4G2.7 (45300)\n",
      "Predicting variants for PAS = EIF5A2.19 (45400)\n",
      "Predicting variants for PAS = ELAVL1.3 (45500)\n",
      "Predicting variants for PAS = ELF3.11 (45600)\n",
      "Predicting variants for PAS = ELMOD1.3 (45700)\n",
      "Predicting variants for PAS = ELOVL5.8 (45800)\n",
      "Predicting variants for PAS = ELP4.4 (45900)\n",
      "Predicting variants for PAS = EMC10.7 (46000)\n",
      "Predicting variants for PAS = EMC8.6 (46100)\n",
      "Predicting variants for PAS = EML2.4 (46200)\n",
      "Predicting variants for PAS = EMSY.31 (46300)\n",
      "Predicting variants for PAS = ENDOD1.13 (46400)\n",
      "Predicting variants for PAS = ENO4.1 (46500)\n",
      "Predicting variants for PAS = ENPP4.6 (46600)\n",
      "Predicting variants for PAS = ENTPD4.14 (46700)\n",
      "Predicting variants for PAS = EP300.1 (46800)\n",
      "Predicting variants for PAS = EPB41.29 (46900)\n",
      "Predicting variants for PAS = EPB41L5.1 (47000)\n",
      "Predicting variants for PAS = EPGN.12 (47100)\n",
      "Predicting variants for PAS = EPHA6.33 (47200)\n",
      "Predicting variants for PAS = EPHX4.3 (47300)\n",
      "Predicting variants for PAS = EPPK1.16 (47400)\n",
      "Predicting variants for PAS = ERAP2.2 (47500)\n",
      "Predicting variants for PAS = ERCC1.6 (47600)\n",
      "Predicting variants for PAS = ERCC6L2.18 (47700)\n",
      "Predicting variants for PAS = ERGIC1.4 (47800)\n",
      "Predicting variants for PAS = ERI1.31 (47900)\n",
      "Predicting variants for PAS = ERLIN1.1 (48000)\n",
      "Predicting variants for PAS = ERO1A.9 (48100)\n",
      "Predicting variants for PAS = ERV3-1-ZNF117.20 (48200)\n",
      "Predicting variants for PAS = ESR1.7 (48300)\n",
      "Predicting variants for PAS = ESYT2.23 (48400)\n",
      "Predicting variants for PAS = ETFDH.12 (48500)\n",
      "Predicting variants for PAS = ETV1.19 (48600)\n",
      "Predicting variants for PAS = EVC2.4 (48700)\n",
      "Predicting variants for PAS = EXO1.18 (48800)\n",
      "Predicting variants for PAS = EXOC5.16 (48900)\n",
      "Predicting variants for PAS = EXOSC1.7 (49000)\n",
      "Predicting variants for PAS = EXOSC7.9 (49100)\n",
      "Predicting variants for PAS = EYA3.1 (49200)\n",
      "Predicting variants for PAS = F2R.10 (49300)\n",
      "Predicting variants for PAS = FABP5.1 (49400)\n",
      "Predicting variants for PAS = FAF2.7 (49500)\n",
      "Predicting variants for PAS = FAM104A.9 (49600)\n",
      "Predicting variants for PAS = FAM111A.9 (49700)\n",
      "Predicting variants for PAS = FAM120A.7 (49800)\n",
      "Predicting variants for PAS = FAM122B.1 (49900)\n",
      "Predicting variants for PAS = FAM129B.5 (50000)\n",
      "Predicting variants for PAS = FAM13A.17 (50100)\n",
      "Predicting variants for PAS = FAM153C.1 (50200)\n",
      "Predicting variants for PAS = FAM168A.2 (50300)\n",
      "Predicting variants for PAS = FAM172A.32 (50400)\n",
      "Predicting variants for PAS = FAM178B.6 (50500)\n",
      "Predicting variants for PAS = FAM193A.3 (50600)\n",
      "Predicting variants for PAS = FAM19A5.15 (50700)\n",
      "Predicting variants for PAS = FAM208B.6 (50800)\n",
      "Predicting variants for PAS = FAM210B.8 (50900)\n",
      "Predicting variants for PAS = FAM219A.2 (51000)\n",
      "Predicting variants for PAS = FARS2.9 (52000)\n",
      "Predicting variants for PAS = FAT1.5 (52100)\n",
      "Predicting variants for PAS = FBLN1.1 (52200)\n",
      "Predicting variants for PAS = FBRSL1.9 (52300)\n",
      "Predicting variants for PAS = FBXL17.45 (52400)\n",
      "Predicting variants for PAS = FBXL3.3 (52500)\n",
      "Predicting variants for PAS = FBXO11.23 (52600)\n",
      "Predicting variants for PAS = FBXO22.24 (52700)\n",
      "Predicting variants for PAS = FBXO31.6 (52800)\n",
      "Predicting variants for PAS = FBXO42.2 (52900)\n",
      "Predicting variants for PAS = FBXW11.13 (53000)\n",
      "Predicting variants for PAS = FCF1.3 (53100)\n",
      "Predicting variants for PAS = FCRLA.2 (53200)\n",
      "Predicting variants for PAS = FEM1A.1 (53300)\n",
      "Predicting variants for PAS = FERMT3.2 (53400)\n",
      "Predicting variants for PAS = FGF1.1 (53500)\n",
      "Predicting variants for PAS = FGF5.14 (53600)\n",
      "Predicting variants for PAS = FGFR1OP2.8 (53700)\n",
      "Predicting variants for PAS = FHDC1.7 (53800)\n",
      "Predicting variants for PAS = FIG4.2 (53900)\n",
      "Predicting variants for PAS = FITM2.4 (54000)\n",
      "Predicting variants for PAS = FKBP4.4 (54100)\n",
      "Predicting variants for PAS = FLII.6 (54200)\n",
      "Predicting variants for PAS = FLT1.8 (54300)\n",
      "Predicting variants for PAS = FMNL2.14 (54400)\n",
      "Predicting variants for PAS = FN3KRP.5 (54500)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Predicting variants for PAS = FNDC3B.10 (54600)\n",
      "Predicting variants for PAS = FOLH1.7 (54700)\n",
      "Predicting variants for PAS = FOXB1.1 (54800)\n",
      "Predicting variants for PAS = FOXM1.6 (54900)\n",
      "Predicting variants for PAS = FOXP1.10 (55000)\n",
      "Predicting variants for PAS = FRA10AC1.6 (55100)\n",
      "Predicting variants for PAS = FRG1BP.22 (55200)\n",
      "Predicting variants for PAS = FRMD8.5 (55300)\n",
      "Predicting variants for PAS = FSBP.2 (55400)\n",
      "Predicting variants for PAS = FSTL4.4 (55500)\n",
      "Predicting variants for PAS = FTO.5 (55600)\n",
      "Predicting variants for PAS = FUNDC2.2 (55700)\n",
      "Predicting variants for PAS = FUT2.2 (55800)\n",
      "Predicting variants for PAS = FXR1.27 (55900)\n",
      "Predicting variants for PAS = FYTTD1.25 (56000)\n",
      "Predicting variants for PAS = FZD6.16 (56100)\n",
      "Predicting variants for PAS = G3BP2.20 (56200)\n",
      "Predicting variants for PAS = GABBR2.8 (56300)\n",
      "Predicting variants for PAS = GABRB1.3 (56400)\n",
      "Predicting variants for PAS = GABRQ.5 (56500)\n",
      "Predicting variants for PAS = GALK2.10 (56600)\n",
      "Predicting variants for PAS = GALNT13.13 (56700)\n",
      "Predicting variants for PAS = GALNT2.6 (56800)\n",
      "Predicting variants for PAS = GAMT.2 (56900)\n",
      "Predicting variants for PAS = GAR1.3 (57000)\n",
      "Predicting variants for PAS = GAS2L3.15 (57100)\n",
      "Predicting variants for PAS = GATAD2A.4 (57200)\n",
      "Predicting variants for PAS = GBA2.2 (57300)\n",
      "Predicting variants for PAS = GCC1.1 (57400)\n",
      "Predicting variants for PAS = GCLM.4 (57500)\n",
      "Predicting variants for PAS = GCSHP3.3 (57600)\n",
      "Predicting variants for PAS = GDF11.12 (57700)\n",
      "Predicting variants for PAS = GEM.2 (57800)\n",
      "Predicting variants for PAS = GET4.6 (57900)\n",
      "Predicting variants for PAS = GFPT1.18 (58000)\n",
      "Predicting variants for PAS = GGCX.7 (58100)\n",
      "Predicting variants for PAS = GHITM.11 (58200)\n",
      "Predicting variants for PAS = GIGYF2.26 (58300)\n",
      "Predicting variants for PAS = GINS4.19 (58400)\n",
      "Predicting variants for PAS = GK.13 (58500)\n",
      "Predicting variants for PAS = GLCCI1.2 (58600)\n",
      "Predicting variants for PAS = GLI3.4 (58700)\n",
      "Predicting variants for PAS = GLMN.9 (58800)\n",
      "Predicting variants for PAS = GLRX3.15 (58900)\n",
      "Predicting variants for PAS = GLUD1.4 (59000)\n",
      "Predicting variants for PAS = GMCL1.10 (59100)\n",
      "Predicting variants for PAS = GMPPB.4 (59200)\n",
      "Predicting variants for PAS = GNAI1.7 (59300)\n",
      "Predicting variants for PAS = GNAS.1 (59400)\n",
      "Predicting variants for PAS = GNB5.9 (59500)\n",
      "Predicting variants for PAS = GNG7.5 (59600)\n",
      "Predicting variants for PAS = GNPDA1.4 (59700)\n",
      "Predicting variants for PAS = GNPTG.6 (59800)\n",
      "Predicting variants for PAS = GOLGA4.17 (59900)\n",
      "Predicting variants for PAS = GOLPH3.9 (60000)\n",
      "Predicting variants for PAS = GORASP2.3 (60100)\n",
      "Predicting variants for PAS = GOT2.15 (60200)\n",
      "Predicting variants for PAS = GPATCH2.20 (60300)\n",
      "Predicting variants for PAS = GPBP1L1.2 (60400)\n",
      "Predicting variants for PAS = GPC6.22 (60500)\n",
      "Predicting variants for PAS = GPM6B.7 (60600)\n",
      "Predicting variants for PAS = GPR108.1 (60700)\n",
      "Predicting variants for PAS = GPR155.4 (60800)\n",
      "Predicting variants for PAS = GPR180.13 (60900)\n",
      "Predicting variants for PAS = GPR75-ASB3.8 (61000)\n",
      "Predicting variants for PAS = GPRIN2.3 (61100)\n",
      "Predicting variants for PAS = GPX2.1 (61200)\n",
      "Predicting variants for PAS = GRB2.2 (61300)\n",
      "Predicting variants for PAS = GRHL2.4 (61400)\n",
      "Predicting variants for PAS = GRIA4.5 (61500)\n",
      "Predicting variants for PAS = GRIK3.3 (61600)\n",
      "Predicting variants for PAS = GRK3.8 (61700)\n",
      "Predicting variants for PAS = GRM7.5 (61800)\n",
      "Predicting variants for PAS = GS1-124K5.6 (61900)\n",
      "Predicting variants for PAS = GSK3B.21 (62000)\n",
      "Predicting variants for PAS = GSTCD.8 (62100)\n",
      "Predicting variants for PAS = GTDC1.8 (62200)\n",
      "Predicting variants for PAS = GTF2F1.16 (62300)\n",
      "Predicting variants for PAS = GTF2IRD1P1.2 (62400)\n",
      "Predicting variants for PAS = GTPBP1.5 (62500)\n",
      "Predicting variants for PAS = GTSE1.8 (62600)\n",
      "Predicting variants for PAS = GUF1.13 (62700)\n",
      "Predicting variants for PAS = GUSBP11.20 (62800)\n",
      "Predicting variants for PAS = GYS1.6 (62900)\n",
      "Predicting variants for PAS = H2AFY.10 (63000)\n",
      "Predicting variants for PAS = HACD1.13 (63100)\n",
      "Predicting variants for PAS = HADHB.3 (63200)\n",
      "Predicting variants for PAS = HASPIN.6 (63300)\n",
      "Predicting variants for PAS = HBEGF.2 (63400)\n",
      "Predicting variants for PAS = HCN1.8 (63500)\n",
      "Predicting variants for PAS = HDAC4.24 (63600)\n",
      "Predicting variants for PAS = HDGFL3.12 (63700)\n",
      "Predicting variants for PAS = HEATR1.23 (63800)\n",
      "Predicting variants for PAS = HEBP2.14 (63900)\n",
      "Predicting variants for PAS = HECW2.9 (64000)\n",
      "Predicting variants for PAS = HELZ.24 (64100)\n",
      "Predicting variants for PAS = HERC4.1 (64200)\n",
      "Predicting variants for PAS = HEXA.7 (64300)\n",
      "Predicting variants for PAS = HGSNAT.3 (64400)\n",
      "Predicting variants for PAS = HID1.2 (64500)\n",
      "Predicting variants for PAS = HILPDA.3 (64600)\n",
      "Predicting variants for PAS = HIPK2.3 (64700)\n",
      "Predicting variants for PAS = HIST1H2AG.5 (64800)\n",
      "Predicting variants for PAS = HIST1H2BJ.3 (64900)\n",
      "Predicting variants for PAS = HIST1H4B.4 (65000)\n",
      "Predicting variants for PAS = HIVEP1.6 (65100)\n",
      "Predicting variants for PAS = HKR1.19 (65200)\n",
      "Predicting variants for PAS = HLCS.24 (65300)\n",
      "Predicting variants for PAS = HMCES.6 (65400)\n",
      "Predicting variants for PAS = HMGB1.19 (65500)\n",
      "Predicting variants for PAS = HMGCS1.3 (65600)\n",
      "Predicting variants for PAS = HMOX1.2 (65700)\n",
      "Predicting variants for PAS = HNRNPA1P33.2 (65800)\n",
      "Predicting variants for PAS = HNRNPA3.29 (65900)\n",
      "Predicting variants for PAS = HNRNPDL.28 (66000)\n",
      "Predicting variants for PAS = HNRNPLL.15 (66100)\n",
      "Predicting variants for PAS = HNRNPU.51 (66200)\n",
      "Predicting variants for PAS = HOMEZ.1 (66300)\n",
      "Predicting variants for PAS = HOXA3.6 (66400)\n",
      "Predicting variants for PAS = HOXD4.1 (66500)\n",
      "Predicting variants for PAS = HPN.2 (66600)\n",
      "Predicting variants for PAS = HRASLS2.1 (66700)\n",
      "Predicting variants for PAS = HS3ST3A1.12 (66800)\n",
      "Predicting variants for PAS = HSCB.9 (66900)\n",
      "Predicting variants for PAS = HSDL1.8 (67000)\n",
      "Predicting variants for PAS = HSP90AB1.1 (67100)\n",
      "Predicting variants for PAS = HSPA13.11 (67200)\n",
      "Predicting variants for PAS = HSPA5.18 (67300)\n",
      "Predicting variants for PAS = HSPD1.15 (67400)\n",
      "Predicting variants for PAS = HTD2.2 (67500)\n",
      "Predicting variants for PAS = HTRA2.13 (67600)\n",
      "Predicting variants for PAS = HYI.1 (67700)\n",
      "Predicting variants for PAS = IBTK.3 (67800)\n",
      "Predicting variants for PAS = ICOSLG.2 (67900)\n",
      "Predicting variants for PAS = IDH3G.7 (68000)\n",
      "Predicting variants for PAS = IER5L.5 (68100)\n",
      "Predicting variants for PAS = IFNAR1.3 (68200)\n",
      "Predicting variants for PAS = IFT140.8 (68300)\n",
      "Predicting variants for PAS = IFT74.15 (68400)\n",
      "Predicting variants for PAS = IGF1R.28 (68500)\n",
      "Predicting variants for PAS = IGF2BP3.24 (68600)\n",
      "Predicting variants for PAS = IGFL2.1 (68700)\n",
      "Predicting variants for PAS = IGSF3.19 (68800)\n",
      "Predicting variants for PAS = IL10RB.4 (68900)\n",
      "Predicting variants for PAS = IL17RB.3 (69000)\n",
      "Predicting variants for PAS = IL1RN.2 (69100)\n",
      "Predicting variants for PAS = IL6R.5 (69200)\n",
      "Predicting variants for PAS = IMMP2L.5 (69300)\n",
      "Predicting variants for PAS = IMPA2.3 (69400)\n",
      "Predicting variants for PAS = INCENP.11 (69500)\n",
      "Predicting variants for PAS = INHBA.7 (69600)\n",
      "Predicting variants for PAS = INPP4B.14 (69700)\n",
      "Predicting variants for PAS = INSIG1.14 (69800)\n",
      "Predicting variants for PAS = INTS2.3 (69900)\n",
      "Predicting variants for PAS = INTS7.19 (70000)\n",
      "Predicting variants for PAS = IPCEF1.2 (70100)\n",
      "Predicting variants for PAS = IPO5P1.6 (70200)\n",
      "Predicting variants for PAS = IPP.5 (70300)\n",
      "Predicting variants for PAS = IQCG.18 (70400)\n",
      "Predicting variants for PAS = IQSEC1.16 (70500)\n",
      "Predicting variants for PAS = IREB2.23 (70600)\n",
      "Predicting variants for PAS = IRS1.14 (70700)\n",
      "Predicting variants for PAS = ISG20.1 (70800)\n",
      "Predicting variants for PAS = ITCH.9 (70900)\n",
      "Predicting variants for PAS = ITGA1.23 (71000)\n",
      "Predicting variants for PAS = ITGA4.18 (71100)\n",
      "Predicting variants for PAS = ITGB1.18 (71200)\n",
      "Predicting variants for PAS = ITGB6.24 (71300)\n",
      "Predicting variants for PAS = ITPK1.3 (71400)\n",
      "Predicting variants for PAS = ITPR3.12 (71500)\n",
      "Predicting variants for PAS = IVNS1ABP.1 (71600)\n",
      "Predicting variants for PAS = JAGN1.4 (71700)\n",
      "Predicting variants for PAS = JAM2.10 (71800)\n",
      "Predicting variants for PAS = JMJD1C.4 (71900)\n",
      "Predicting variants for PAS = JPH4.1 (72000)\n",
      "Predicting variants for PAS = KAAG1.1 (72100)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Predicting variants for PAS = KANSL1L.21 (72200)\n",
      "Predicting variants for PAS = KAT7.9 (72300)\n",
      "Predicting variants for PAS = KAZN.5 (72400)\n",
      "Predicting variants for PAS = KCMF1.19 (72500)\n",
      "Predicting variants for PAS = KCND1.1 (72600)\n",
      "Predicting variants for PAS = KCNIP3.4 (72700)\n",
      "Predicting variants for PAS = KCNJ9.3 (72800)\n",
      "Predicting variants for PAS = KCNMA1.9 (72900)\n",
      "Predicting variants for PAS = KCNQ3.11 (73000)\n",
      "Predicting variants for PAS = KCTD1.22 (73100)\n",
      "Predicting variants for PAS = KCTD20.15 (73200)\n",
      "Predicting variants for PAS = KDELR2.11 (73300)\n",
      "Predicting variants for PAS = KDM3A.11 (73400)\n",
      "Predicting variants for PAS = KDM5A.31 (73500)\n",
      "Predicting variants for PAS = KDSR.9 (73600)\n",
      "Predicting variants for PAS = KHNYN.8 (73700)\n",
      "Predicting variants for PAS = KIAA0355.4 (73800)\n",
      "Predicting variants for PAS = KIAA0753.9 (73900)\n",
      "Predicting variants for PAS = KIAA1147.10 (74000)\n",
      "Predicting variants for PAS = KIAA1257.26 (74100)\n",
      "Predicting variants for PAS = KIAA1468.19 (74200)\n",
      "Predicting variants for PAS = KIAA1671.11 (74300)\n",
      "Predicting variants for PAS = KIAA2026.14 (74400)\n",
      "Predicting variants for PAS = KIF15.9 (74500)\n",
      "Predicting variants for PAS = KIF20A.6 (74600)\n",
      "Predicting variants for PAS = KIF2A.7 (74700)\n",
      "Predicting variants for PAS = KIF5B.22 (74800)\n",
      "Predicting variants for PAS = KIN.12 (74900)\n",
      "Predicting variants for PAS = KLC1.19 (75000)\n",
      "Predicting variants for PAS = KLF3.23 (75100)\n",
      "Predicting variants for PAS = KLHDC4.1 (75200)\n",
      "Predicting variants for PAS = KLHL18.9 (75300)\n",
      "Predicting variants for PAS = KLHL31.2 (75400)\n",
      "Predicting variants for PAS = KLHL8.8 (75500)\n",
      "Predicting variants for PAS = KLRG1.8 (75600)\n",
      "Predicting variants for PAS = KMT2E.26 (75700)\n",
      "Predicting variants for PAS = KNSTRN.12 (75800)\n",
      "Predicting variants for PAS = KPNA4.9 (75900)\n",
      "Predicting variants for PAS = KRBA1.3 (76000)\n",
      "Predicting variants for PAS = KRIT1.18 (76100)\n",
      "Predicting variants for PAS = KRT19.6 (76200)\n",
      "Predicting variants for PAS = KRT80.5 (76300)\n",
      "Predicting variants for PAS = KYAT3.2 (76400)\n",
      "Predicting variants for PAS = L3HYPDH.12 (76500)\n",
      "Predicting variants for PAS = LAMA1.11 (76600)\n",
      "Predicting variants for PAS = LAMB1.24 (76700)\n",
      "Predicting variants for PAS = LAMP2.14 (76800)\n",
      "Predicting variants for PAS = LAP3.9 (76900)\n",
      "Predicting variants for PAS = LARP4.14 (77000)\n",
      "Predicting variants for PAS = LARS2.15 (77100)\n",
      "Predicting variants for PAS = LBR.13 (77200)\n",
      "Predicting variants for PAS = LCOR.26 (77300)\n",
      "Predicting variants for PAS = LDHA.1 (77400)\n",
      "Predicting variants for PAS = LDLRAD3.18 (77500)\n",
      "Predicting variants for PAS = LENG1.3 (77600)\n",
      "Predicting variants for PAS = LETM2.3 (77700)\n",
      "Predicting variants for PAS = LGI1.7 (77800)\n",
      "Predicting variants for PAS = LHPP.1 (77900)\n",
      "Predicting variants for PAS = LIG1.7 (78000)\n",
      "Predicting variants for PAS = LIMK1.3 (78100)\n",
      "Predicting variants for PAS = LIN7A.20 (78200)\n",
      "Predicting variants for PAS = LINC02210-CRHR1.30 (78300)\n",
      "Predicting variants for PAS = LIPT2.2 (78400)\n",
      "Predicting variants for PAS = LMBR1.7 (78500)\n",
      "Predicting variants for PAS = LMBRD2.15 (78600)\n",
      "Predicting variants for PAS = LMNTD1.3 (78700)\n",
      "Predicting variants for PAS = LMX1B.6 (78800)\n",
      "Predicting variants for PAS = LOC100129484.1 (78900)\n",
      "Predicting variants for PAS = LOC100288842.2 (79000)\n",
      "Predicting variants for PAS = LOC101927572.1 (79100)\n",
      "Predicting variants for PAS = LOC220729.8 (79200)\n",
      "Predicting variants for PAS = LOC389906.8 (79300)\n",
      "Predicting variants for PAS = LOC554206.1 (79400)\n",
      "Predicting variants for PAS = LOC728743.3 (79500)\n",
      "Predicting variants for PAS = LOX.2 (79600)\n",
      "Predicting variants for PAS = LPCAT1.12 (79700)\n",
      "Predicting variants for PAS = LPIN3.1 (79800)\n",
      "Predicting variants for PAS = LRCH1.11 (79900)\n",
      "Predicting variants for PAS = LRIG3.9 (80000)\n",
      "Predicting variants for PAS = LRP2.12 (80100)\n",
      "Predicting variants for PAS = LRP8.26 (80200)\n",
      "Predicting variants for PAS = LRRC14.5 (80300)\n",
      "Predicting variants for PAS = LRRC28.37 (80400)\n",
      "Predicting variants for PAS = LRRC43.1 (80500)\n",
      "Predicting variants for PAS = LRRC6.3 (80600)\n",
      "Predicting variants for PAS = LRRC8B.11 (80700)\n",
      "Predicting variants for PAS = LRRIQ1.4 (80800)\n",
      "Predicting variants for PAS = LRRTM3.8 (80900)\n",
      "Predicting variants for PAS = LSM14A.3 (81000)\n",
      "Predicting variants for PAS = LSM8.14 (81100)\n",
      "Predicting variants for PAS = LTBR.8 (81200)\n",
      "Predicting variants for PAS = LUZP1.6 (81300)\n",
      "Predicting variants for PAS = LYG2.1 (81400)\n",
      "Predicting variants for PAS = LYRM1.5 (81500)\n",
      "Predicting variants for PAS = LYST.6 (81600)\n",
      "Predicting variants for PAS = MACC1.4 (81700)\n",
      "Predicting variants for PAS = MAD1L1.9 (81800)\n",
      "Predicting variants for PAS = MAGEA10-MAGEA5.3 (81900)\n",
      "Predicting variants for PAS = MAGI3.3 (82000)\n",
      "Predicting variants for PAS = MAK16.2 (82100)\n",
      "Predicting variants for PAS = MAMDC2.18 (82200)\n",
      "Predicting variants for PAS = MAN1A2.32 (82300)\n",
      "Predicting variants for PAS = MANBAL.4 (82400)\n",
      "Predicting variants for PAS = MAP1LC3B.8 (82500)\n",
      "Predicting variants for PAS = MAP2K6.12 (82600)\n",
      "Predicting variants for PAS = MAP3K20.7 (82700)\n",
      "Predicting variants for PAS = MAP3K8.6 (82800)\n",
      "Predicting variants for PAS = MAP6.4 (82900)\n",
      "Predicting variants for PAS = MAPK10.2 (83000)\n",
      "Predicting variants for PAS = MAPK1IP1L.21 (83100)\n",
      "Predicting variants for PAS = MAPK9.18 (83200)\n",
      "Predicting variants for PAS = MAPRE3.3 (83300)\n",
      "Predicting variants for PAS = MARCH6.6 (83400)\n",
      "Predicting variants for PAS = MARF1.2 (83500)\n",
      "Predicting variants for PAS = MARVELD3.7 (83600)\n",
      "Predicting variants for PAS = MAT2B.9 (83700)\n",
      "Predicting variants for PAS = MAVS.20 (83800)\n",
      "Predicting variants for PAS = MBD5.9 (83900)\n",
      "Predicting variants for PAS = MBNL2.10 (84000)\n",
      "Predicting variants for PAS = MBP.28 (84100)\n",
      "Predicting variants for PAS = MCC.8 (84200)\n",
      "Predicting variants for PAS = MCFD2.5 (84300)\n",
      "Predicting variants for PAS = MCM4.1 (84400)\n",
      "Predicting variants for PAS = MCMBP.21 (84500)\n",
      "Predicting variants for PAS = MCTP1.29 (84600)\n",
      "Predicting variants for PAS = MDFIC.10 (84700)\n",
      "Predicting variants for PAS = MDM2.28 (84800)\n",
      "Predicting variants for PAS = MEA1.2 (84900)\n",
      "Predicting variants for PAS = MED12L.10 (85000)\n",
      "Predicting variants for PAS = MED15.2 (85100)\n",
      "Predicting variants for PAS = MED28.25 (85200)\n",
      "Predicting variants for PAS = MED6.18 (85300)\n",
      "Predicting variants for PAS = MEGF11.3 (85400)\n",
      "Predicting variants for PAS = MELK.5 (85500)\n",
      "Predicting variants for PAS = MESD.20 (85600)\n",
      "Predicting variants for PAS = METRN.2 (85700)\n",
      "Predicting variants for PAS = METTL21EP.1 (85800)\n",
      "Predicting variants for PAS = METTL5.2 (85900)\n",
      "Predicting variants for PAS = MFAP1.1 (86000)\n",
      "Predicting variants for PAS = MFN1.18 (86100)\n",
      "Predicting variants for PAS = MFSD3.4 (86200)\n",
      "Predicting variants for PAS = MGA.20 (86300)\n",
      "Predicting variants for PAS = MGAT5.7 (86400)\n",
      "Predicting variants for PAS = MGST1.12 (86500)\n",
      "Predicting variants for PAS = MICAL1.16 (86600)\n",
      "Predicting variants for PAS = MICU1.20 (86700)\n",
      "Predicting variants for PAS = MIEF1.21 (86800)\n",
      "Predicting variants for PAS = MILR1.2 (86900)\n",
      "Predicting variants for PAS = MIOS.15 (87000)\n",
      "Predicting variants for PAS = MISP3.1 (87100)\n",
      "Predicting variants for PAS = MKLN1.3 (87200)\n",
      "Predicting variants for PAS = MKX.1 (87300)\n",
      "Predicting variants for PAS = MLKL.5 (87400)\n",
      "Predicting variants for PAS = MLX.7 (87500)\n",
      "Predicting variants for PAS = MMD.1 (87600)\n",
      "Predicting variants for PAS = MMP16.12 (87700)\n",
      "Predicting variants for PAS = MMP8.3 (87800)\n",
      "Predicting variants for PAS = MOAP1.5 (87900)\n",
      "Predicting variants for PAS = MOCS2.8 (88000)\n",
      "Predicting variants for PAS = MORC3.8 (88100)\n",
      "Predicting variants for PAS = MORN3.1 (88200)\n",
      "Predicting variants for PAS = MPDZ.8 (88300)\n",
      "Predicting variants for PAS = MPI.17 (88400)\n",
      "Predicting variants for PAS = MPP6.19 (88500)\n",
      "Predicting variants for PAS = MPRIP.31 (88600)\n",
      "Predicting variants for PAS = MRAS.1 (88700)\n",
      "Predicting variants for PAS = MRI1.6 (88800)\n",
      "Predicting variants for PAS = MRPL14.10 (88900)\n",
      "Predicting variants for PAS = MRPL23.2 (89000)\n",
      "Predicting variants for PAS = MRPL32.1 (89100)\n",
      "Predicting variants for PAS = MRPL42.26 (89200)\n",
      "Predicting variants for PAS = MRPL48.16 (89300)\n",
      "Predicting variants for PAS = MRPS11.6 (89400)\n",
      "Predicting variants for PAS = MRPS18C.12 (89500)\n",
      "Predicting variants for PAS = MRPS28.16 (89600)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Predicting variants for PAS = MRPS7.2 (89700)\n",
      "Predicting variants for PAS = MSANTD3-TMEFF1.8 (89800)\n",
      "Predicting variants for PAS = MSH6.15 (89900)\n",
      "Predicting variants for PAS = MSRA.3 (90000)\n",
      "Predicting variants for PAS = MT1P3.1 (90100)\n",
      "Predicting variants for PAS = MTATP6P2.1 (90200)\n",
      "Predicting variants for PAS = MTCO1P40.3 (90300)\n",
      "Predicting variants for PAS = MTF2.10 (90400)\n",
      "Predicting variants for PAS = MTHFD1.8 (90500)\n",
      "Predicting variants for PAS = MTHFD2L.30 (90600)\n",
      "Predicting variants for PAS = MTMR10.12 (90700)\n",
      "Predicting variants for PAS = MTMR7.13 (90800)\n",
      "Predicting variants for PAS = MTND5P11.8 (90900)\n",
      "Predicting variants for PAS = MTRF1L.12 (91000)\n",
      "Predicting variants for PAS = MTSS1L.8 (91100)\n",
      "Predicting variants for PAS = MUC20.2 (91200)\n",
      "Predicting variants for PAS = MVD.4 (91300)\n",
      "Predicting variants for PAS = MYBBP1A.11 (91400)\n",
      "Predicting variants for PAS = MYEF2.6 (91500)\n",
      "Predicting variants for PAS = MYH9.27 (91600)\n",
      "Predicting variants for PAS = MYLK2.1 (91700)\n",
      "Predicting variants for PAS = MYO18A.8 (91800)\n",
      "Predicting variants for PAS = MYO1E.22 (91900)\n",
      "Predicting variants for PAS = MYO6.11 (92000)\n",
      "Predicting variants for PAS = MYOZ2.3 (92100)\n",
      "Predicting variants for PAS = MZF1.1 (92200)\n",
      "Predicting variants for PAS = N4BP2L2.26 (92300)\n",
      "Predicting variants for PAS = NAA25.11 (92400)\n",
      "Predicting variants for PAS = NAA50.3 (92500)\n",
      "Predicting variants for PAS = NAB1.2 (92600)\n",
      "Predicting variants for PAS = NADK2.19 (92700)\n",
      "Predicting variants for PAS = NALCN.23 (92800)\n",
      "Predicting variants for PAS = NAP1L1.41 (92900)\n",
      "Predicting variants for PAS = NARS.6 (93000)\n",
      "Predicting variants for PAS = NAV1.12 (93100)\n",
      "Predicting variants for PAS = NBDY.18 (93200)\n",
      "Predicting variants for PAS = NBPF1.16 (93300)\n",
      "Predicting variants for PAS = NBPF19.9 (93400)\n",
      "Predicting variants for PAS = NBPF20.27 (93500)\n",
      "Predicting variants for PAS = NCAM1.9 (93600)\n",
      "Predicting variants for PAS = NCAN.12 (93700)\n",
      "Predicting variants for PAS = NCAPH2.10 (93800)\n",
      "Predicting variants for PAS = NCK1.9 (93900)\n",
      "Predicting variants for PAS = NCL.18 (94000)\n",
      "Predicting variants for PAS = NCOA5.4 (94100)\n",
      "Predicting variants for PAS = NCOR2.2 (94200)\n",
      "Predicting variants for PAS = NDE1.10 (94300)\n",
      "Predicting variants for PAS = NDRG2.4 (94400)\n",
      "Predicting variants for PAS = NDUFA10.34 (94500)\n",
      "Predicting variants for PAS = NDUFA9.6 (94600)\n",
      "Predicting variants for PAS = NDUFAF6.34 (94700)\n",
      "Predicting variants for PAS = NDUFB7.4 (94800)\n",
      "Predicting variants for PAS = NDUFV1.2 (94900)\n",
      "Predicting variants for PAS = NECAP2.10 (95000)\n",
      "Predicting variants for PAS = NEDD4.26 (95100)\n",
      "Predicting variants for PAS = NEIL1.5 (95200)\n",
      "Predicting variants for PAS = NEK4.7 (95300)\n",
      "Predicting variants for PAS = NELFA.6 (95400)\n",
      "Predicting variants for PAS = NEMP2.14 (95500)\n",
      "Predicting variants for PAS = NETO1.20 (95600)\n",
      "Predicting variants for PAS = NEURL4.1 (95700)\n",
      "Predicting variants for PAS = NFAT5.3 (95800)\n",
      "Predicting variants for PAS = NFE2L2.17 (95900)\n",
      "Predicting variants for PAS = NFIX.4 (96000)\n",
      "Predicting variants for PAS = NFXL1.16 (96100)\n",
      "Predicting variants for PAS = NHEJ1.13 (96200)\n",
      "Predicting variants for PAS = NHSL1.8 (96300)\n",
      "Predicting variants for PAS = NIP7.4 (96400)\n",
      "Predicting variants for PAS = NIPAL3.31 (96500)\n",
      "Predicting variants for PAS = NIT2.17 (96600)\n",
      "Predicting variants for PAS = NKD2.5 (96700)\n",
      "Predicting variants for PAS = NLGN1.5 (96800)\n",
      "Predicting variants for PAS = NLRC5.9 (96900)\n",
      "Predicting variants for PAS = NME7.4 (97000)\n",
      "Predicting variants for PAS = NMT2.7 (97100)\n",
      "Predicting variants for PAS = NOC4L.2 (97200)\n",
      "Predicting variants for PAS = NOL6.4 (97300)\n",
      "Predicting variants for PAS = NONO.10 (97400)\n",
      "Predicting variants for PAS = NOTCH1.6 (97500)\n",
      "Predicting variants for PAS = NOXRED1.4 (97600)\n",
      "Predicting variants for PAS = NPC1.27 (97700)\n",
      "Predicting variants for PAS = NPHP4.12 (97800)\n",
      "Predicting variants for PAS = NPR3.9 (97900)\n",
      "Predicting variants for PAS = NR1D2.4 (98000)\n",
      "Predicting variants for PAS = NR3C1.12 (98100)\n",
      "Predicting variants for PAS = NRBP1.1 (98200)\n",
      "Predicting variants for PAS = NRF1.12 (98300)\n",
      "Predicting variants for PAS = NRIP1.11 (98400)\n",
      "Predicting variants for PAS = NRTN.1 (98500)\n",
      "Predicting variants for PAS = NSD2.7 (98600)\n",
      "Predicting variants for PAS = NSL1.3 (98700)\n",
      "Predicting variants for PAS = NSRP1.1 (98800)\n",
      "Predicting variants for PAS = NSUN7.16 (98900)\n",
      "Predicting variants for PAS = NT5DC1.18 (99000)\n",
      "Predicting variants for PAS = NTMT1.3 (99100)\n",
      "Predicting variants for PAS = NUAK1.2 (99200)\n",
      "Predicting variants for PAS = NUCKS1.29 (99300)\n",
      "Predicting variants for PAS = NUDT15.9 (99400)\n",
      "Predicting variants for PAS = NUDT5.29 (99500)\n",
      "Predicting variants for PAS = NUMA1.8 (99600)\n",
      "Predicting variants for PAS = NUP155.23 (99700)\n",
      "Predicting variants for PAS = NUP37.10 (99800)\n",
      "Predicting variants for PAS = NUP88.20 (99900)\n",
      "Predicting variants for PAS = NVL.7 (100000)\n",
      "Predicting variants for PAS = OARD1.9 (100100)\n",
      "Predicting variants for PAS = OBSL1.2 (100200)\n",
      "Predicting variants for PAS = ODF2.10 (100300)\n",
      "Predicting variants for PAS = OGFOD2.4 (100400)\n",
      "Predicting variants for PAS = OGT.20 (100500)\n",
      "Predicting variants for PAS = OLR1.5 (100600)\n",
      "Predicting variants for PAS = OPCML.9 (100700)\n",
      "Predicting variants for PAS = OR51B4.3 (100800)\n",
      "Predicting variants for PAS = ORC1.9 (100900)\n",
      "Predicting variants for PAS = ORMDL1.9 (101000)\n",
      "Predicting variants for PAS = OSBPL11.5 (101100)\n",
      "Predicting variants for PAS = OSBPL6.2 (101200)\n",
      "Predicting variants for PAS = OSGIN1.1 (101300)\n",
      "Predicting variants for PAS = OSTF1.3 (101400)\n",
      "Predicting variants for PAS = OTUD6B.3 (101500)\n",
      "Predicting variants for PAS = OXA1L.3 (101600)\n",
      "Predicting variants for PAS = OXR1.18 (101700)\n",
      "Predicting variants for PAS = P2RY2.8 (101800)\n",
      "Predicting variants for PAS = P4HTM.6 (101900)\n",
      "Predicting variants for PAS = PABPC4.8 (102000)\n",
      "Predicting variants for PAS = PADI2.5 (102100)\n",
      "Predicting variants for PAS = PAGR1.9 (102200)\n",
      "Predicting variants for PAS = PAK3.14 (102300)\n",
      "Predicting variants for PAS = PAM.20 (102400)\n",
      "Predicting variants for PAS = PANK3.27 (102500)\n",
      "Predicting variants for PAS = PAPOLA.6 (102600)\n",
      "Predicting variants for PAS = PAQR4.3 (102700)\n",
      "Predicting variants for PAS = PARD6G.10 (102800)\n",
      "Predicting variants for PAS = PARP15.2 (102900)\n",
      "Predicting variants for PAS = PARVA.23 (103000)\n",
      "Predicting variants for PAS = PATJ.47 (103100)\n",
      "Predicting variants for PAS = PAX6.12 (103200)\n",
      "Predicting variants for PAS = PBRM1.1 (103300)\n",
      "Predicting variants for PAS = PC.5 (103400)\n",
      "Predicting variants for PAS = PCBP2.27 (103500)\n",
      "Predicting variants for PAS = PCDH11Y.7 (103600)\n",
      "Predicting variants for PAS = PCDHA4.1 (103700)\n",
      "Predicting variants for PAS = PCDHB8.3 (103800)\n",
      "Predicting variants for PAS = PCGF3.5 (103900)\n",
      "Predicting variants for PAS = PCLAF.24 (104000)\n",
      "Predicting variants for PAS = PCMTD2.16 (104100)\n",
      "Predicting variants for PAS = PCNX2.25 (104200)\n",
      "Predicting variants for PAS = PCSK5.7 (104300)\n",
      "Predicting variants for PAS = PCYOX1.19 (104400)\n",
      "Predicting variants for PAS = PDCD1LG2.8 (104500)\n",
      "Predicting variants for PAS = PDCD6IPP2.7 (104600)\n",
      "Predicting variants for PAS = PDE12.23 (104700)\n",
      "Predicting variants for PAS = PDE3B.19 (104800)\n",
      "Predicting variants for PAS = PDE5A.4 (104900)\n",
      "Predicting variants for PAS = PDE9A.17 (105000)\n",
      "Predicting variants for PAS = PDHB.4 (105100)\n",
      "Predicting variants for PAS = PDIK1L.3 (105200)\n",
      "Predicting variants for PAS = PDLIM5.6 (105300)\n",
      "Predicting variants for PAS = PDRG1.3 (105400)\n",
      "Predicting variants for PAS = PDSS2.19 (105500)\n",
      "Predicting variants for PAS = PDZD8.12 (105600)\n",
      "Predicting variants for PAS = PELI2.2 (105700)\n",
      "Predicting variants for PAS = PET100.1 (105800)\n",
      "Predicting variants for PAS = PEX19.10 (105900)\n",
      "Predicting variants for PAS = PEX6.4 (106000)\n",
      "Predicting variants for PAS = PFKFB4.2 (106100)\n",
      "Predicting variants for PAS = PGAM2.1 (106200)\n",
      "Predicting variants for PAS = PGGT1B.9 (106300)\n",
      "Predicting variants for PAS = PGM3.6 (106400)\n",
      "Predicting variants for PAS = PHACTR2.2 (106500)\n",
      "Predicting variants for PAS = PHC3.6 (106600)\n",
      "Predicting variants for PAS = PHF19.4 (106700)\n",
      "Predicting variants for PAS = PHF21A.5 (106800)\n",
      "Predicting variants for PAS = PHF8.15 (106900)\n",
      "Predicting variants for PAS = PHKB.10 (107000)\n",
      "Predicting variants for PAS = PHLDB2.24 (107100)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Predicting variants for PAS = PHTF1.14 (107200)\n",
      "Predicting variants for PAS = PHYKPL.18 (107300)\n",
      "Predicting variants for PAS = PIAS1.31 (107400)\n",
      "Predicting variants for PAS = PID1.5 (107500)\n",
      "Predicting variants for PAS = PIGG.4 (107600)\n",
      "Predicting variants for PAS = PIGN.9 (107700)\n",
      "Predicting variants for PAS = PIGX.8 (107800)\n",
      "Predicting variants for PAS = PIK3CA.11 (107900)\n",
      "Predicting variants for PAS = PILRB.10 (108000)\n",
      "Predicting variants for PAS = PIP4K2C.6 (108100)\n",
      "Predicting variants for PAS = PITPNA.6 (108200)\n",
      "Predicting variants for PAS = PIWIL2.3 (108300)\n",
      "Predicting variants for PAS = PKIA.17 (108400)\n",
      "Predicting variants for PAS = PKN3.2 (108500)\n",
      "Predicting variants for PAS = PKP4.36 (108600)\n",
      "Predicting variants for PAS = PLA2G7.5 (108700)\n",
      "Predicting variants for PAS = PLAUR.13 (108800)\n",
      "Predicting variants for PAS = PLCE1.10 (108900)\n",
      "Predicting variants for PAS = PLD3.6 (109000)\n",
      "Predicting variants for PAS = PLEKHA4.6 (109100)\n",
      "Predicting variants for PAS = PLEKHB2.1 (109200)\n",
      "Predicting variants for PAS = PLEKHH2.5 (109300)\n",
      "Predicting variants for PAS = PLK2.1 (109400)\n",
      "Predicting variants for PAS = PLPP3.1 (109500)\n",
      "Predicting variants for PAS = PLSCR1.9 (109600)\n",
      "Predicting variants for PAS = PLXNA3.4 (109700)\n",
      "Predicting variants for PAS = PMFBP1.3 (109800)\n",
      "Predicting variants for PAS = PMS2P1.6 (109900)\n",
      "Predicting variants for PAS = PNN.7 (110000)\n",
      "Predicting variants for PAS = PNPO.5 (110100)\n",
      "Predicting variants for PAS = PODXL.10 (110200)\n",
      "Predicting variants for PAS = POGZ.8 (110300)\n",
      "Predicting variants for PAS = POLDIP2.13 (110400)\n",
      "Predicting variants for PAS = POLH.14 (110500)\n",
      "Predicting variants for PAS = POLQ.6 (110600)\n",
      "Predicting variants for PAS = POLR1E.19 (110700)\n",
      "Predicting variants for PAS = POLR2J4.9 (110800)\n",
      "Predicting variants for PAS = POLR3F.5 (110900)\n",
      "Predicting variants for PAS = POMK.4 (111000)\n",
      "Predicting variants for PAS = POP5.1 (111100)\n",
      "Predicting variants for PAS = POU3F2.7 (111200)\n",
      "Predicting variants for PAS = PPARA.1 (111300)\n",
      "Predicting variants for PAS = PPCDC.1 (111400)\n",
      "Predicting variants for PAS = PPFIA2.37 (111500)\n",
      "Predicting variants for PAS = PPIAP73.1 (111600)\n",
      "Predicting variants for PAS = PPIL2.7 (111700)\n",
      "Predicting variants for PAS = PPM1A.5 (111800)\n",
      "Predicting variants for PAS = PPM1K.1 (111900)\n",
      "Predicting variants for PAS = PPP1CC.21 (112000)\n",
      "Predicting variants for PAS = PPP1R14C.3 (112100)\n",
      "Predicting variants for PAS = PPP1R35.3 (112200)\n",
      "Predicting variants for PAS = PPP1R9A.16 (112300)\n",
      "Predicting variants for PAS = PPP2R1B.37 (112400)\n",
      "Predicting variants for PAS = PPP2R5A.6 (112500)\n",
      "Predicting variants for PAS = PPP3CB.8 (112600)\n",
      "Predicting variants for PAS = PPP4R3B.10 (112700)\n",
      "Predicting variants for PAS = PPP6R2.13 (112800)\n",
      "Predicting variants for PAS = PQLC3.4 (112900)\n",
      "Predicting variants for PAS = PRDM11.15 (113000)\n",
      "Predicting variants for PAS = PRDX2.5 (113100)\n",
      "Predicting variants for PAS = PREP.6 (113200)\n",
      "Predicting variants for PAS = PRICKLE1.6 (113300)\n",
      "Predicting variants for PAS = PRKAA2.14 (113400)\n",
      "Predicting variants for PAS = PRKAR1A.9 (113500)\n",
      "Predicting variants for PAS = PRKCA.20 (113600)\n",
      "Predicting variants for PAS = PRKD1.1 (113700)\n",
      "Predicting variants for PAS = PRKX.4 (113800)\n",
      "Predicting variants for PAS = PRMT3.15 (113900)\n",
      "Predicting variants for PAS = PRORSD1P.2 (114000)\n",
      "Predicting variants for PAS = PRPF38A.10 (114100)\n",
      "Predicting variants for PAS = PRPF4B.2 (114200)\n",
      "Predicting variants for PAS = PRR13.1 (114300)\n",
      "Predicting variants for PAS = PRRC1.5 (114400)\n",
      "Predicting variants for PAS = PRSS12.2 (114500)\n",
      "Predicting variants for PAS = PRSS56.4 (114600)\n",
      "Predicting variants for PAS = PSAT1.4 (114700)\n",
      "Predicting variants for PAS = PSG5.3 (114800)\n",
      "Predicting variants for PAS = PSMA3.11 (114900)\n",
      "Predicting variants for PAS = PSMB5.1 (115000)\n",
      "Predicting variants for PAS = PSMD11.5 (115100)\n",
      "Predicting variants for PAS = PSMD6.16 (115200)\n",
      "Predicting variants for PAS = PSMG1.11 (115300)\n",
      "Predicting variants for PAS = PSRC1.1 (115400)\n",
      "Predicting variants for PAS = PTBP3.19 (115500)\n",
      "Predicting variants for PAS = PTCHD4.3 (115600)\n",
      "Predicting variants for PAS = PTEN.28 (115700)\n",
      "Predicting variants for PAS = PTGES3.12 (115800)\n",
      "Predicting variants for PAS = PTGS2.3 (115900)\n",
      "Predicting variants for PAS = PTN.2 (116000)\n",
      "Predicting variants for PAS = PTPDC1.14 (116100)\n",
      "Predicting variants for PAS = PTPN14.12 (116200)\n",
      "Predicting variants for PAS = PTPN4.30 (116300)\n",
      "Predicting variants for PAS = PTPRE.10 (116400)\n",
      "Predicting variants for PAS = PTPRK.48 (116500)\n",
      "Predicting variants for PAS = PTPRS.11 (116600)\n",
      "Predicting variants for PAS = PUDP.8 (116700)\n",
      "Predicting variants for PAS = PURA.8 (116800)\n",
      "Predicting variants for PAS = PUS7L.5 (116900)\n",
      "Predicting variants for PAS = PXDC1.2 (117000)\n",
      "Predicting variants for PAS = PYCR2.1 (117100)\n",
      "Predicting variants for PAS = PYURF.5 (117200)\n",
      "Predicting variants for PAS = QKI.63 (117300)\n",
      "Predicting variants for PAS = QTRT1.5 (117400)\n",
      "Predicting variants for PAS = RAB11A.5 (117500)\n",
      "Predicting variants for PAS = RAB12.10 (117600)\n",
      "Predicting variants for PAS = RAB21.18 (117700)\n",
      "Predicting variants for PAS = RAB2A.9 (117800)\n",
      "Predicting variants for PAS = RAB36.7 (117900)\n",
      "Predicting variants for PAS = RAB3GAP2.7 (118000)\n",
      "Predicting variants for PAS = RAB5A.9 (118100)\n",
      "Predicting variants for PAS = RAB9A.1 (118200)\n",
      "Predicting variants for PAS = RABGAP1L.13 (118300)\n",
      "Predicting variants for PAS = RABL2B.5 (118400)\n",
      "Predicting variants for PAS = RAD1.9 (118500)\n",
      "Predicting variants for PAS = RAD50.6 (118600)\n",
      "Predicting variants for PAS = RAD51C.3 (118700)\n",
      "Predicting variants for PAS = RAD9B.4 (118800)\n",
      "Predicting variants for PAS = RALBP1.13 (118900)\n",
      "Predicting variants for PAS = RALGPS2.19 (119000)\n",
      "Predicting variants for PAS = RANBP17.22 (119100)\n",
      "Predicting variants for PAS = RAP1A.7 (119200)\n",
      "Predicting variants for PAS = RAP2C.5 (119300)\n",
      "Predicting variants for PAS = RAPH1.18 (119400)\n",
      "Predicting variants for PAS = RASA1.17 (119500)\n",
      "Predicting variants for PAS = RASEF.19 (119600)\n",
      "Predicting variants for PAS = RASGRP3.7 (119700)\n",
      "Predicting variants for PAS = RAVER2.3 (119800)\n",
      "Predicting variants for PAS = RBBP5.3 (119900)\n",
      "Predicting variants for PAS = RBFOX1.5 (120000)\n",
      "Predicting variants for PAS = RBM11.1 (120100)\n",
      "Predicting variants for PAS = RBM15.17 (120200)\n",
      "Predicting variants for PAS = RBM25.11 (120300)\n",
      "Predicting variants for PAS = RBM28.3 (120400)\n",
      "Predicting variants for PAS = RBM41.5 (120500)\n",
      "Predicting variants for PAS = RBM4B.1 (120600)\n",
      "Predicting variants for PAS = RBMS3.4 (120700)\n",
      "Predicting variants for PAS = RBPMS.11 (120800)\n",
      "Predicting variants for PAS = RCAN2.1 (120900)\n",
      "Predicting variants for PAS = RCHY1.1 (121000)\n",
      "Predicting variants for PAS = RCOR3.16 (121100)\n",
      "Predicting variants for PAS = RDX.16 (121200)\n",
      "Predicting variants for PAS = REEP3.14 (121300)\n",
      "Predicting variants for PAS = RELT.2 (121400)\n",
      "Predicting variants for PAS = RET.5 (121500)\n",
      "Predicting variants for PAS = REXO4.12 (121600)\n",
      "Predicting variants for PAS = RFC5.26 (121700)\n",
      "Predicting variants for PAS = RFTN1.11 (121800)\n",
      "Predicting variants for PAS = RFX5.11 (121900)\n",
      "Predicting variants for PAS = RGMB.18 (122000)\n",
      "Predicting variants for PAS = RGS20.5 (122100)\n",
      "Predicting variants for PAS = RGS8.2 (122200)\n",
      "Predicting variants for PAS = RHCE.3 (122300)\n",
      "Predicting variants for PAS = RHOBTB3.23 (122400)\n",
      "Predicting variants for PAS = RHOU.9 (122500)\n",
      "Predicting variants for PAS = RIC8B.3 (122600)\n",
      "Predicting variants for PAS = RIMBP2.3 (122700)\n",
      "Predicting variants for PAS = RIN2.2 (122800)\n",
      "Predicting variants for PAS = RIPK1.15 (122900)\n",
      "Predicting variants for PAS = RLIM.8 (123000)\n",
      "Predicting variants for PAS = RMND1.8 (123100)\n",
      "Predicting variants for PAS = RNASEK.3 (123200)\n",
      "Predicting variants for PAS = RNF11.15 (123300)\n",
      "Predicting variants for PAS = RNF135.7 (123400)\n",
      "Predicting variants for PAS = RNF144B.4 (123500)\n",
      "Predicting variants for PAS = RNF167.5 (123600)\n",
      "Predicting variants for PAS = RNF19A.6 (123700)\n",
      "Predicting variants for PAS = RNF213.30 (123800)\n",
      "Predicting variants for PAS = RNF220.7 (123900)\n",
      "Predicting variants for PAS = RNF4.3 (124000)\n",
      "Predicting variants for PAS = RNFT2.5 (124100)\n",
      "Predicting variants for PAS = ROBO1.25 (124200)\n",
      "Predicting variants for PAS = ROR2.2 (124300)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Predicting variants for PAS = RPAP1.3 (124400)\n",
      "Predicting variants for PAS = RPF2.4 (124500)\n",
      "Predicting variants for PAS = RPL10A.6 (124600)\n",
      "Predicting variants for PAS = RPL13A.16 (124700)\n",
      "Predicting variants for PAS = RPL18P11.1 (124800)\n",
      "Predicting variants for PAS = RPL23A.8 (124900)\n",
      "Predicting variants for PAS = RPL27A.9 (125000)\n",
      "Predicting variants for PAS = RPL31.32 (125100)\n",
      "Predicting variants for PAS = RPL35A.1 (125200)\n",
      "Predicting variants for PAS = RPL37A.21 (125300)\n",
      "Predicting variants for PAS = RPL5.4 (125400)\n",
      "Predicting variants for PAS = RPL8.2 (125500)\n",
      "Predicting variants for PAS = RPP30.10 (125600)\n",
      "Predicting variants for PAS = RPS11.2 (125700)\n",
      "Predicting variants for PAS = RPS18P9.3 (125800)\n",
      "Predicting variants for PAS = RPS24.8 (125900)\n",
      "Predicting variants for PAS = RPS2P32.2 (126000)\n",
      "Predicting variants for PAS = RPS4Y1.9 (126100)\n",
      "Predicting variants for PAS = RPS6KA5.12 (126200)\n",
      "Predicting variants for PAS = RPSA.7 (126300)\n",
      "Predicting variants for PAS = RPUSD3.11 (126400)\n",
      "Predicting variants for PAS = RRM1.8 (126500)\n",
      "Predicting variants for PAS = RRP1.10 (126600)\n",
      "Predicting variants for PAS = RRP8.12 (126700)\n",
      "Predicting variants for PAS = RSL1D1.17 (126800)\n",
      "Predicting variants for PAS = RSRP1.8 (126900)\n",
      "Predicting variants for PAS = RTFDC1.16 (127000)\n",
      "Predicting variants for PAS = RTN3.7 (127100)\n",
      "Predicting variants for PAS = RUBCNL.4 (127200)\n",
      "Predicting variants for PAS = RUNX1.4 (127300)\n",
      "Predicting variants for PAS = RWDD1.5 (127400)\n",
      "Predicting variants for PAS = RYR1.4 (127500)\n",
      "Predicting variants for PAS = S100P.2 (127600)\n",
      "Predicting variants for PAS = SAE1.11 (127700)\n",
      "Predicting variants for PAS = SAMD3.2 (127800)\n",
      "Predicting variants for PAS = SAP30.3 (127900)\n",
      "Predicting variants for PAS = SARNP.3 (128000)\n",
      "Predicting variants for PAS = SATB1.9 (128100)\n",
      "Predicting variants for PAS = SBK1.1 (128200)\n",
      "Predicting variants for PAS = SCAI.6 (128300)\n",
      "Predicting variants for PAS = SCAPER.14 (128400)\n",
      "Predicting variants for PAS = SCD.21 (128500)\n",
      "Predicting variants for PAS = SCG5.7 (128600)\n",
      "Predicting variants for PAS = SCMH1.12 (128700)\n",
      "Predicting variants for PAS = SCN7A.5 (128800)\n",
      "Predicting variants for PAS = SCRG1.3 (128900)\n",
      "Predicting variants for PAS = SCYL3.9 (129000)\n",
      "Predicting variants for PAS = SDCCAG8.4 (129100)\n",
      "Predicting variants for PAS = SDHAF2.14 (129200)\n",
      "Predicting variants for PAS = SDK1.22 (129300)\n",
      "Predicting variants for PAS = SEC14L1.17 (129400)\n",
      "Predicting variants for PAS = SEC23A.1 (129500)\n",
      "Predicting variants for PAS = SEC31A.23 (129600)\n",
      "Predicting variants for PAS = SEC63.13 (129700)\n",
      "Predicting variants for PAS = SELENOF.1 (129800)\n",
      "Predicting variants for PAS = SELENOT.9 (129900)\n",
      "Predicting variants for PAS = SEMA3C.23 (130000)\n",
      "Predicting variants for PAS = SEMA6A.2 (130100)\n",
      "Predicting variants for PAS = SENP5.10 (130200)\n",
      "Predicting variants for PAS = SEPT10.11 (130300)\n",
      "Predicting variants for PAS = SEPT6.3 (130400)\n",
      "Predicting variants for PAS = SERBP1.17 (130500)\n",
      "Predicting variants for PAS = SERINC3.5 (130600)\n",
      "Predicting variants for PAS = SERPINB8.3 (130700)\n",
      "Predicting variants for PAS = SERTAD4.14 (130800)\n",
      "Predicting variants for PAS = SETD1B.4 (130900)\n",
      "Predicting variants for PAS = SETD7.17 (131000)\n",
      "Predicting variants for PAS = SETX.16 (131100)\n",
      "Predicting variants for PAS = SF3B3.16 (131200)\n",
      "Predicting variants for PAS = SFR1.9 (131300)\n",
      "Predicting variants for PAS = SFXN1.12 (131400)\n",
      "Predicting variants for PAS = SGF29.2 (131500)\n",
      "Predicting variants for PAS = SGO1.13 (131600)\n",
      "Predicting variants for PAS = SGTA.5 (131700)\n",
      "Predicting variants for PAS = SH3BP4.11 (131800)\n",
      "Predicting variants for PAS = SH3GLB1.7 (131900)\n",
      "Predicting variants for PAS = SH3RF3.3 (132000)\n",
      "Predicting variants for PAS = SHANK2.19 (132100)\n",
      "Predicting variants for PAS = SHCBP1.6 (132200)\n",
      "Predicting variants for PAS = SHISA9.11 (132300)\n",
      "Predicting variants for PAS = SHROOM1.1 (132400)\n",
      "Predicting variants for PAS = SIAH1.9 (132500)\n",
      "Predicting variants for PAS = SIK3.14 (132600)\n",
      "Predicting variants for PAS = SIN3A.14 (132700)\n",
      "Predicting variants for PAS = SIRPAP1.2 (132800)\n",
      "Predicting variants for PAS = SIX4.9 (132900)\n",
      "Predicting variants for PAS = SKIL.1 (133000)\n",
      "Predicting variants for PAS = SLAIN2.12 (133100)\n",
      "Predicting variants for PAS = SLC12A2.8 (133200)\n",
      "Predicting variants for PAS = SLC15A2.4 (133300)\n",
      "Predicting variants for PAS = SLC16A4.6 (133400)\n",
      "Predicting variants for PAS = SLC19A1.9 (133500)\n",
      "Predicting variants for PAS = SLC20A1.4 (133600)\n",
      "Predicting variants for PAS = SLC22A3.2 (133700)\n",
      "Predicting variants for PAS = SLC24A5.2 (133800)\n",
      "Predicting variants for PAS = SLC25A18.2 (133900)\n",
      "Predicting variants for PAS = SLC25A29.7 (134000)\n",
      "Predicting variants for PAS = SLC25A36.26 (134100)\n",
      "Predicting variants for PAS = SLC25A48.8 (134200)\n",
      "Predicting variants for PAS = SLC27A1.8 (134300)\n",
      "Predicting variants for PAS = SLC2A1.31 (134400)\n",
      "Predicting variants for PAS = SLC2A4RG.3 (134500)\n",
      "Predicting variants for PAS = SLC30A5.20 (134600)\n",
      "Predicting variants for PAS = SLC30A9.32 (134700)\n",
      "Predicting variants for PAS = SLC35B2.3 (134800)\n",
      "Predicting variants for PAS = SLC35D2.11 (134900)\n",
      "Predicting variants for PAS = SLC35F3.23 (135000)\n",
      "Predicting variants for PAS = SLC36A1.13 (135100)\n",
      "Predicting variants for PAS = SLC38A1.27 (135200)\n",
      "Predicting variants for PAS = SLC38A9.11 (135300)\n",
      "Predicting variants for PAS = SLC39A7.1 (135400)\n",
      "Predicting variants for PAS = SLC41A3.8 (135500)\n",
      "Predicting variants for PAS = SLC45A3.3 (135600)\n",
      "Predicting variants for PAS = SLC4A5.7 (135700)\n",
      "Predicting variants for PAS = SLC5A8.1 (135800)\n",
      "Predicting variants for PAS = SLC6A3.6 (135900)\n",
      "Predicting variants for PAS = SLC7A2.16 (136000)\n",
      "Predicting variants for PAS = SLC8A1.10 (136100)\n",
      "Predicting variants for PAS = SLC9A8.7 (136200)\n",
      "Predicting variants for PAS = SLCO3A1.6 (136300)\n",
      "Predicting variants for PAS = SLFN12.5 (136400)\n",
      "Predicting variants for PAS = SLIT2.29 (136500)\n",
      "Predicting variants for PAS = SLMAP.13 (136600)\n",
      "Predicting variants for PAS = SMAD2.48 (136700)\n",
      "Predicting variants for PAS = SMAP1.2 (136800)\n",
      "Predicting variants for PAS = SMARCAL1.12 (136900)\n",
      "Predicting variants for PAS = SMC1A.11 (137000)\n",
      "Predicting variants for PAS = SMCHD1.4 (137100)\n",
      "Predicting variants for PAS = SMG7.4 (137200)\n",
      "Predicting variants for PAS = SMIM14.11 (137300)\n",
      "Predicting variants for PAS = SMIM29.1 (137400)\n",
      "Predicting variants for PAS = SMOC1.18 (137500)\n",
      "Predicting variants for PAS = SMURF1.1 (137600)\n",
      "Predicting variants for PAS = SMYD5.3 (137700)\n",
      "Predicting variants for PAS = SNAP91.3 (137800)\n",
      "Predicting variants for PAS = SNAPC5.21 (137900)\n",
      "Predicting variants for PAS = SNF8.11 (138000)\n",
      "Predicting variants for PAS = SNRNP48.15 (138100)\n",
      "Predicting variants for PAS = SNRPF.2 (138200)\n",
      "Predicting variants for PAS = SNTG2.7 (138300)\n",
      "Predicting variants for PAS = SNX14.5 (138400)\n",
      "Predicting variants for PAS = SNX2.18 (138500)\n",
      "Predicting variants for PAS = SNX29P2.1 (138600)\n",
      "Predicting variants for PAS = SOAT1.3 (138700)\n",
      "Predicting variants for PAS = SOCS5.16 (138800)\n",
      "Predicting variants for PAS = SOGA3.16 (138900)\n",
      "Predicting variants for PAS = SORBS2.28 (139000)\n",
      "Predicting variants for PAS = SOS1.9 (139100)\n",
      "Predicting variants for PAS = SOX21.1 (139200)\n",
      "Predicting variants for PAS = SP110.5 (139300)\n",
      "Predicting variants for PAS = SPA17.3 (139400)\n",
      "Predicting variants for PAS = SPAG9.2 (139500)\n",
      "Predicting variants for PAS = SPATA13.4 (139600)\n",
      "Predicting variants for PAS = SPATA5.12 (139700)\n",
      "Predicting variants for PAS = SPC24.2 (139800)\n",
      "Predicting variants for PAS = SPDYA.3 (139900)\n",
      "Predicting variants for PAS = SPG11.26 (140000)\n",
      "Predicting variants for PAS = SPIDR.10 (140100)\n",
      "Predicting variants for PAS = SPINT2.4 (140200)\n",
      "Predicting variants for PAS = SPON2.12 (140300)\n",
      "Predicting variants for PAS = SPRED1.23 (140400)\n",
      "Predicting variants for PAS = SPRYD7.7 (140500)\n",
      "Predicting variants for PAS = SPTLC1.8 (140600)\n",
      "Predicting variants for PAS = SQSTM1.4 (140700)\n",
      "Predicting variants for PAS = SRD5A1.6 (140800)\n",
      "Predicting variants for PAS = SRF.5 (140900)\n",
      "Predicting variants for PAS = SRI.20 (141000)\n",
      "Predicting variants for PAS = SRPK1.5 (141100)\n",
      "Predicting variants for PAS = SRPX2.6 (141200)\n",
      "Predicting variants for PAS = SRSF1.18 (141300)\n",
      "Predicting variants for PAS = SRSF3.11 (141400)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Predicting variants for PAS = SRSF9.14 (141500)\n",
      "Predicting variants for PAS = SSBP2.26 (141600)\n",
      "Predicting variants for PAS = SSH3.1 (141700)\n",
      "Predicting variants for PAS = SSR4.4 (141800)\n",
      "Predicting variants for PAS = ST13.15 (141900)\n",
      "Predicting variants for PAS = ST3GAL3.17 (142000)\n",
      "Predicting variants for PAS = ST6GALNAC4.4 (142100)\n",
      "Predicting variants for PAS = ST8SIA1.4 (142200)\n",
      "Predicting variants for PAS = STAC.13 (142300)\n",
      "Predicting variants for PAS = STAM2.15 (142400)\n",
      "Predicting variants for PAS = STARD4.1 (142500)\n",
      "Predicting variants for PAS = STAT2.6 (142600)\n",
      "Predicting variants for PAS = STAU2.36 (142700)\n",
      "Predicting variants for PAS = STEAP3.3 (142800)\n",
      "Predicting variants for PAS = STK17A.15 (142900)\n",
      "Predicting variants for PAS = STK32A.2 (143000)\n",
      "Predicting variants for PAS = STK39.18 (143100)\n",
      "Predicting variants for PAS = STMP1.7 (143200)\n",
      "Predicting variants for PAS = STON2.11 (143300)\n",
      "Predicting variants for PAS = STRIP1.1 (143400)\n",
      "Predicting variants for PAS = STT3A.9 (143500)\n",
      "Predicting variants for PAS = STX17.18 (143600)\n",
      "Predicting variants for PAS = STX6.28 (143700)\n",
      "Predicting variants for PAS = STXBP3.12 (143800)\n",
      "Predicting variants for PAS = STXBP5L.6 (143900)\n",
      "Predicting variants for PAS = SUB1.10 (144000)\n",
      "Predicting variants for PAS = SUDS3.19 (144100)\n",
      "Predicting variants for PAS = SULF2.2 (144200)\n",
      "Predicting variants for PAS = SUMO2.9 (144300)\n",
      "Predicting variants for PAS = SUPT3H.8 (144400)\n",
      "Predicting variants for PAS = SURF6.6 (144500)\n",
      "Predicting variants for PAS = SUZ12.15 (144600)\n",
      "Predicting variants for PAS = SVIL.17 (144700)\n",
      "Predicting variants for PAS = SYAP1.12 (144800)\n",
      "Predicting variants for PAS = SYK.2 (144900)\n",
      "Predicting variants for PAS = SYNCRIP.8 (145000)\n",
      "Predicting variants for PAS = SYNE2.18 (145100)\n",
      "Predicting variants for PAS = SYNJ2BP-COX16.12 (145200)\n",
      "Predicting variants for PAS = SYPL1P2.5 (145300)\n",
      "Predicting variants for PAS = SYT2.9 (145400)\n",
      "Predicting variants for PAS = SZT2.10 (145500)\n",
      "Predicting variants for PAS = TACC2.22 (145600)\n",
      "Predicting variants for PAS = TAF11.6 (145700)\n",
      "Predicting variants for PAS = TAF1D.6 (145800)\n",
      "Predicting variants for PAS = TAF8.14 (145900)\n",
      "Predicting variants for PAS = TANGO6.14 (146000)\n",
      "Predicting variants for PAS = TAPBP.8 (146100)\n",
      "Predicting variants for PAS = TARSL2.15 (146200)\n",
      "Predicting variants for PAS = TAX1BP1.25 (146300)\n",
      "Predicting variants for PAS = TBC1D16.14 (146400)\n",
      "Predicting variants for PAS = TBC1D22B.7 (146500)\n",
      "Predicting variants for PAS = TBC1D32.6 (146600)\n",
      "Predicting variants for PAS = TBC1D8.23 (146700)\n",
      "Predicting variants for PAS = TBCD.3 (146800)\n",
      "Predicting variants for PAS = TBL1X.14 (146900)\n",
      "Predicting variants for PAS = TBRG1.11 (147000)\n",
      "Predicting variants for PAS = TBXAS1.16 (147100)\n",
      "Predicting variants for PAS = TCEAL1.1 (147200)\n",
      "Predicting variants for PAS = TCERG1.11 (147300)\n",
      "Predicting variants for PAS = TCF4.8 (147400)\n",
      "Predicting variants for PAS = TCN2.3 (147500)\n",
      "Predicting variants for PAS = TCTN3.11 (147600)\n",
      "Predicting variants for PAS = TDRKH.5 (147700)\n",
      "Predicting variants for PAS = TECPR2.1 (147800)\n",
      "Predicting variants for PAS = TENM1.4 (147900)\n",
      "Predicting variants for PAS = TERF2.10 (148000)\n",
      "Predicting variants for PAS = TET1.6 (148100)\n",
      "Predicting variants for PAS = TEX2.15 (148200)\n",
      "Predicting variants for PAS = TFAM.14 (148300)\n",
      "Predicting variants for PAS = TFCP2L1.3 (148400)\n",
      "Predicting variants for PAS = TFPI.24 (148500)\n",
      "Predicting variants for PAS = TGFB1I1.3 (148600)\n",
      "Predicting variants for PAS = TGFBR2.17 (148700)\n",
      "Predicting variants for PAS = TGOLN2.4 (148800)\n",
      "Predicting variants for PAS = THAP3.4 (148900)\n",
      "Predicting variants for PAS = THBS1.34 (149000)\n",
      "Predicting variants for PAS = THOC2.3 (149100)\n",
      "Predicting variants for PAS = THRB.25 (149200)\n",
      "Predicting variants for PAS = THY1.2 (149300)\n",
      "Predicting variants for PAS = TIFA.7 (149400)\n",
      "Predicting variants for PAS = TIMM10B.2 (149500)\n",
      "Predicting variants for PAS = TIMM44.5 (149600)\n",
      "Predicting variants for PAS = TIMP2.16 (149700)\n",
      "Predicting variants for PAS = TIPRL.5 (149800)\n",
      "Predicting variants for PAS = TKT.12 (149900)\n",
      "Predicting variants for PAS = TLK1.11 (150000)\n",
      "Predicting variants for PAS = TLNRD1.1 (150100)\n",
      "Predicting variants for PAS = TM2D3.8 (150200)\n",
      "Predicting variants for PAS = TM9SF3.2 (150300)\n",
      "Predicting variants for PAS = TMBIM4.11 (150400)\n",
      "Predicting variants for PAS = TMCC3.6 (150500)\n",
      "Predicting variants for PAS = TMED10P2.1 (150600)\n",
      "Predicting variants for PAS = TMED7-TICAM2.19 (150700)\n",
      "Predicting variants for PAS = TMEM105.1 (150800)\n",
      "Predicting variants for PAS = TMEM116.15 (150900)\n",
      "Predicting variants for PAS = TMEM131.1 (151000)\n",
      "Predicting variants for PAS = TMEM138.7 (151100)\n",
      "Predicting variants for PAS = TMEM154.9 (151200)\n",
      "Predicting variants for PAS = TMEM167A.11 (151300)\n",
      "Predicting variants for PAS = TMEM175.9 (151400)\n",
      "Predicting variants for PAS = TMEM182.9 (151500)\n",
      "Predicting variants for PAS = TMEM189-UBE2V1.6 (151600)\n",
      "Predicting variants for PAS = TMEM200A.3 (151700)\n",
      "Predicting variants for PAS = TMEM218.6 (151800)\n",
      "Predicting variants for PAS = TMEM234.1 (151900)\n",
      "Predicting variants for PAS = TMEM245.18 (152000)\n",
      "Predicting variants for PAS = TMEM256.3 (152100)\n",
      "Predicting variants for PAS = TMEM268.8 (152200)\n",
      "Predicting variants for PAS = TMEM37.3 (152300)\n",
      "Predicting variants for PAS = TMEM43.12 (152400)\n",
      "Predicting variants for PAS = TMEM54.1 (152500)\n",
      "Predicting variants for PAS = TMEM65.9 (152600)\n",
      "Predicting variants for PAS = TMEM86A.1 (152700)\n",
      "Predicting variants for PAS = TMEM94.4 (152800)\n",
      "Predicting variants for PAS = TMOD2.3 (152900)\n",
      "Predicting variants for PAS = TMPRSS15.5 (153000)\n",
      "Predicting variants for PAS = TMTC2.10 (153100)\n",
      "Predicting variants for PAS = TMX3.11 (153200)\n",
      "Predicting variants for PAS = TNFAIP8L2-SCNM1.2 (153300)\n",
      "Predicting variants for PAS = TNFRSF1A.6 (153400)\n",
      "Predicting variants for PAS = TNIK.15 (153500)\n",
      "Predicting variants for PAS = TNPO1.4 (153600)\n",
      "Predicting variants for PAS = TNS1.9 (153700)\n",
      "Predicting variants for PAS = TOLLIP.15 (153800)\n",
      "Predicting variants for PAS = TOMM34.5 (153900)\n",
      "Predicting variants for PAS = TOP2A.3 (154000)\n",
      "Predicting variants for PAS = TOR1A.11 (154100)\n",
      "Predicting variants for PAS = TOX.5 (154200)\n",
      "Predicting variants for PAS = TP53I3.2 (154300)\n",
      "Predicting variants for PAS = TPD52.9 (154400)\n",
      "Predicting variants for PAS = TPI1.8 (154500)\n",
      "Predicting variants for PAS = TPM4.7 (154600)\n",
      "Predicting variants for PAS = TPRA1.6 (154700)\n",
      "Predicting variants for PAS = TPT1P4.1 (154800)\n",
      "Predicting variants for PAS = TRA2B.19 (154900)\n",
      "Predicting variants for PAS = TRAF3IP2.12 (155000)\n",
      "Predicting variants for PAS = TRAM1.3 (155100)\n",
      "Predicting variants for PAS = TRAPPC12.5 (155200)\n",
      "Predicting variants for PAS = TRAPPC8.11 (155300)\n",
      "Predicting variants for PAS = TREML2.3 (155400)\n",
      "Predicting variants for PAS = TRIM14.14 (155500)\n",
      "Predicting variants for PAS = TRIM25.10 (155600)\n",
      "Predicting variants for PAS = TRIM37.9 (155700)\n",
      "Predicting variants for PAS = TRIM5.9 (155800)\n",
      "Predicting variants for PAS = TRIM71.5 (155900)\n",
      "Predicting variants for PAS = TRIOBP.16 (156000)\n",
      "Predicting variants for PAS = TRIR.3 (156100)\n",
      "Predicting variants for PAS = TRMT12.12 (156200)\n",
      "Predicting variants for PAS = TRMT61B.6 (156300)\n",
      "Predicting variants for PAS = TRPC1.6 (156400)\n",
      "Predicting variants for PAS = TRPM7.11 (156500)\n",
      "Predicting variants for PAS = TSC1.4 (156600)\n",
      "Predicting variants for PAS = TSFM.9 (156700)\n",
      "Predicting variants for PAS = TSKU.9 (156800)\n",
      "Predicting variants for PAS = TSPAN14.5 (156900)\n",
      "Predicting variants for PAS = TSPAN5.2 (157000)\n",
      "Predicting variants for PAS = TSR1.6 (157100)\n",
      "Predicting variants for PAS = TTBK1.8 (157200)\n",
      "Predicting variants for PAS = TTC17.19 (157300)\n",
      "Predicting variants for PAS = TTC26.4 (157400)\n",
      "Predicting variants for PAS = TTC31.5 (157500)\n",
      "Predicting variants for PAS = TTC39B.11 (157600)\n",
      "Predicting variants for PAS = TTC6.26 (157700)\n",
      "Predicting variants for PAS = TTF2.14 (157800)\n",
      "Predicting variants for PAS = TTLL11.17 (157900)\n",
      "Predicting variants for PAS = TTPAL.13 (158000)\n",
      "Predicting variants for PAS = TUBB2A.4 (158100)\n",
      "Predicting variants for PAS = TUBGCP2.7 (158200)\n",
      "Predicting variants for PAS = TULP2.1 (158300)\n",
      "Predicting variants for PAS = TVP23A.8 (158400)\n",
      "Predicting variants for PAS = TXLNA.5 (158500)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Predicting variants for PAS = TXNDC17.10 (158600)\n",
      "Predicting variants for PAS = TXNRD1.6 (158700)\n",
      "Predicting variants for PAS = TYR.13 (158800)\n",
      "Predicting variants for PAS = U2SURP.13 (158900)\n",
      "Predicting variants for PAS = UBA5.13 (159000)\n",
      "Predicting variants for PAS = UBALD2.3 (159100)\n",
      "Predicting variants for PAS = UBE2B.22 (159200)\n",
      "Predicting variants for PAS = UBE2E1.8 (159300)\n",
      "Predicting variants for PAS = UBE2J1.1 (159400)\n",
      "Predicting variants for PAS = UBE2O.2 (159500)\n",
      "Predicting variants for PAS = UBE2W.17 (159600)\n",
      "Predicting variants for PAS = UBE3C.35 (159700)\n",
      "Predicting variants for PAS = UBL3.11 (159800)\n",
      "Predicting variants for PAS = UBP1.15 (159900)\n",
      "Predicting variants for PAS = UBR4.4 (160000)\n",
      "Predicting variants for PAS = UBXN2A.7 (160100)\n",
      "Predicting variants for PAS = UCHL5.3 (160200)\n",
      "Predicting variants for PAS = UFD1.9 (160300)\n",
      "Predicting variants for PAS = UGGT1.18 (160400)\n",
      "Predicting variants for PAS = UGT3A2.3 (160500)\n",
      "Predicting variants for PAS = UHRF2.3 (160600)\n",
      "Predicting variants for PAS = UMAD1.17 (160700)\n",
      "Predicting variants for PAS = UNC5A.5 (160800)\n",
      "Predicting variants for PAS = UNK.4 (160900)\n",
      "Predicting variants for PAS = UPRT.5 (161000)\n",
      "Predicting variants for PAS = UQCRC1.3 (161100)\n",
      "Predicting variants for PAS = URGCP-MRPS24.3 (161200)\n",
      "Predicting variants for PAS = USHBP1.1 (161300)\n",
      "Predicting variants for PAS = USP13.13 (161400)\n",
      "Predicting variants for PAS = USP24.9 (161500)\n",
      "Predicting variants for PAS = USP33.8 (161600)\n",
      "Predicting variants for PAS = USP38.19 (161700)\n",
      "Predicting variants for PAS = USP46.10 (161800)\n",
      "Predicting variants for PAS = USP54.1 (161900)\n",
      "Predicting variants for PAS = USP9X.22 (162000)\n",
      "Predicting variants for PAS = UTP20.2 (162100)\n",
      "Predicting variants for PAS = UVRAG.2 (162200)\n",
      "Predicting variants for PAS = VAMP4.5 (162300)\n",
      "Predicting variants for PAS = VASH2.4 (162400)\n",
      "Predicting variants for PAS = VAX2.8 (162500)\n",
      "Predicting variants for PAS = VDAC1.10 (162600)\n",
      "Predicting variants for PAS = VEZT.3 (162700)\n",
      "Predicting variants for PAS = VIM.16 (162800)\n",
      "Predicting variants for PAS = VLDLR.4 (162900)\n",
      "Predicting variants for PAS = VOPP1.12 (163000)\n",
      "Predicting variants for PAS = VPS13C.15 (163100)\n",
      "Predicting variants for PAS = VPS33A.1 (163200)\n",
      "Predicting variants for PAS = VPS37C.1 (163300)\n",
      "Predicting variants for PAS = VPS4B.9 (163400)\n",
      "Predicting variants for PAS = VPS54.15 (163500)\n",
      "Predicting variants for PAS = VSIG10.19 (163600)\n",
      "Predicting variants for PAS = VTCN1.9 (163700)\n",
      "Predicting variants for PAS = VWA8.17 (163800)\n",
      "Predicting variants for PAS = WASF1.1 (163900)\n",
      "Predicting variants for PAS = WASHC5.11 (164000)\n",
      "Predicting variants for PAS = WDFY2.13 (164100)\n",
      "Predicting variants for PAS = WDR1.1 (164200)\n",
      "Predicting variants for PAS = WDR26.2 (164300)\n",
      "Predicting variants for PAS = WDR33.2 (164400)\n",
      "Predicting variants for PAS = WDR37.6 (164500)\n",
      "Predicting variants for PAS = WDR43.15 (164600)\n",
      "Predicting variants for PAS = WDR5B.1 (164700)\n",
      "Predicting variants for PAS = WDR70.8 (164800)\n",
      "Predicting variants for PAS = WDR78.2 (164900)\n",
      "Predicting variants for PAS = WDR92.17 (165000)\n",
      "Predicting variants for PAS = WFDC1.3 (165100)\n",
      "Predicting variants for PAS = WIPI1.8 (165200)\n",
      "Predicting variants for PAS = WLS.15 (165300)\n",
      "Predicting variants for PAS = WNT2B.10 (165400)\n",
      "Predicting variants for PAS = WRB-SH3BGR.3 (165500)\n",
      "Predicting variants for PAS = WSB2.27 (165600)\n",
      "Predicting variants for PAS = WWC2.23 (165700)\n",
      "Predicting variants for PAS = WWP1.32 (165800)\n",
      "Predicting variants for PAS = XKR5.6 (165900)\n",
      "Predicting variants for PAS = XPNPEP3.12 (166000)\n",
      "Predicting variants for PAS = XPO7.6 (166100)\n",
      "Predicting variants for PAS = XRCC3.12 (166200)\n",
      "Predicting variants for PAS = XXYLT1.3 (166300)\n",
      "Predicting variants for PAS = YAF2.16 (166400)\n",
      "Predicting variants for PAS = YBX3.8 (166500)\n",
      "Predicting variants for PAS = YIPF5.5 (166600)\n",
      "Predicting variants for PAS = YME1L1.15 (166700)\n",
      "Predicting variants for PAS = YTHDF3.9 (166800)\n",
      "Predicting variants for PAS = YWHAQ.7 (166900)\n",
      "Predicting variants for PAS = ZADH2.22 (167000)\n",
      "Predicting variants for PAS = ZBTB1.3 (167100)\n",
      "Predicting variants for PAS = ZBTB20.17 (167200)\n",
      "Predicting variants for PAS = ZBTB34.4 (167300)\n",
      "Predicting variants for PAS = ZBTB44.4 (167400)\n",
      "Predicting variants for PAS = ZBTB8A.10 (167500)\n",
      "Predicting variants for PAS = ZC3H12C.14 (167600)\n",
      "Predicting variants for PAS = ZC3H6.4 (167700)\n",
      "Predicting variants for PAS = ZCCHC10.8 (167800)\n",
      "Predicting variants for PAS = ZCCHC4.13 (167900)\n",
      "Predicting variants for PAS = ZDBF2.3 (168000)\n",
      "Predicting variants for PAS = ZDHHC17.10 (168100)\n",
      "Predicting variants for PAS = ZDHHC21.34 (168200)\n",
      "Predicting variants for PAS = ZDHHC9.3 (168300)\n",
      "Predicting variants for PAS = ZFAND4.2 (168400)\n",
      "Predicting variants for PAS = ZFHX4.7 (168500)\n",
      "Predicting variants for PAS = ZFP41.5 (168600)\n",
      "Predicting variants for PAS = ZFP91.7 (168700)\n",
      "Predicting variants for PAS = ZFYVE16.13 (168800)\n",
      "Predicting variants for PAS = ZGPAT.2 (168900)\n",
      "Predicting variants for PAS = ZIC4.2 (169000)\n",
      "Predicting variants for PAS = ZKSCAN7.6 (169100)\n",
      "Predicting variants for PAS = ZMPSTE24.5 (169200)\n",
      "Predicting variants for PAS = ZMYM4.14 (169300)\n",
      "Predicting variants for PAS = ZNF100.9 (169400)\n",
      "Predicting variants for PAS = ZNF124.11 (169500)\n",
      "Predicting variants for PAS = ZNF141.13 (169600)\n",
      "Predicting variants for PAS = ZNF157.4 (169700)\n",
      "Predicting variants for PAS = ZNF19.1 (169800)\n",
      "Predicting variants for PAS = ZNF207.17 (169900)\n",
      "Predicting variants for PAS = ZNF223.1 (170000)\n",
      "Predicting variants for PAS = ZNF234.4 (170100)\n",
      "Predicting variants for PAS = ZNF251.6 (170200)\n",
      "Predicting variants for PAS = ZNF254.43 (170300)\n",
      "Predicting variants for PAS = ZNF273.2 (170400)\n",
      "Predicting variants for PAS = ZNF280D.14 (170500)\n",
      "Predicting variants for PAS = ZNF292.26 (170600)\n",
      "Predicting variants for PAS = ZNF320.18 (170700)\n",
      "Predicting variants for PAS = ZNF335.1 (170800)\n",
      "Predicting variants for PAS = ZNF346.3 (170900)\n",
      "Predicting variants for PAS = ZNF362.8 (171000)\n",
      "Predicting variants for PAS = ZNF383.13 (171100)\n",
      "Predicting variants for PAS = ZNF397.11 (171200)\n",
      "Predicting variants for PAS = ZNF417.14 (171300)\n",
      "Predicting variants for PAS = ZNF430.6 (171400)\n",
      "Predicting variants for PAS = ZNF445.2 (171500)\n",
      "Predicting variants for PAS = ZNF460.8 (171600)\n",
      "Predicting variants for PAS = ZNF483.6 (171700)\n",
      "Predicting variants for PAS = ZNF496.7 (171800)\n",
      "Predicting variants for PAS = ZNF510.10 (171900)\n",
      "Predicting variants for PAS = ZNF519.7 (172000)\n",
      "Predicting variants for PAS = ZNF530.8 (172100)\n",
      "Predicting variants for PAS = ZNF546.2 (172200)\n",
      "Predicting variants for PAS = ZNF555.1 (172300)\n",
      "Predicting variants for PAS = ZNF562.17 (172400)\n",
      "Predicting variants for PAS = ZNF573.5 (172500)\n",
      "Predicting variants for PAS = ZNF585A.6 (172600)\n",
      "Predicting variants for PAS = ZNF595.10 (172700)\n",
      "Predicting variants for PAS = ZNF608.4 (172800)\n",
      "Predicting variants for PAS = ZNF620.6 (172900)\n",
      "Predicting variants for PAS = ZNF638.23 (173000)\n",
      "Predicting variants for PAS = ZNF653.2 (173100)\n",
      "Predicting variants for PAS = ZNF664-RFLNA.25 (173200)\n",
      "Predicting variants for PAS = ZNF678.18 (173300)\n",
      "Predicting variants for PAS = ZNF696.5 (173400)\n",
      "Predicting variants for PAS = ZNF704.17 (173500)\n",
      "Predicting variants for PAS = ZNF713.1 (173600)\n",
      "Predicting variants for PAS = ZNF720.4 (173700)\n",
      "Predicting variants for PAS = ZNF732.4 (173800)\n",
      "Predicting variants for PAS = ZNF747.3 (173900)\n",
      "Predicting variants for PAS = ZNF765-ZNF761.27 (174000)\n",
      "Predicting variants for PAS = ZNF778.8 (174100)\n",
      "Predicting variants for PAS = ZNF784.1 (174200)\n",
      "Predicting variants for PAS = ZNF791.13 (174300)\n",
      "Predicting variants for PAS = ZNF81.7 (174400)\n",
      "Predicting variants for PAS = ZNF829.1 (174500)\n",
      "Predicting variants for PAS = ZNF84.6 (174600)\n",
      "Predicting variants for PAS = ZNF862.7 (174700)\n",
      "Predicting variants for PAS = ZNF91.4 (174800)\n",
      "Predicting variants for PAS = ZNRF1.6 (174900)\n",
      "Predicting variants for PAS = ZRANB2.12 (175000)\n",
      "Predicting variants for PAS = ZSCAN20.2 (175100)\n",
      "Predicting variants for PAS = ZSCAN32.7 (175200)\n",
      "Predicting variants for PAS = ZW10.5 (175300)\n",
      "Predicting variants for PAS = ZYX.2 (175400)\n"
     ]
    }
   ],
   "source": [
    "#Perform in-silico saturation mutagenesis\n",
    "\n",
    "aparent_encoder = get_aparent_encoder()\n",
    "\n",
    "mutagenesis_end = 146\n",
    "\n",
    "min_abs_logodds = 0.0\n",
    "\n",
    "variant_dict = {\n",
    "    'gene' : [],\n",
    "    'gene_id' : [],\n",
    "    #'ref_seq' : [],\n",
    "    'chrom' : [],\n",
    "    'strand' : [],\n",
    "    'site_type' : [],\n",
    "    'native_usage' : [],\n",
    "    #'var_seq' : [],\n",
    "    'var_position' : [],\n",
    "    'ref_nucleotide' : [],\n",
    "    'var_nucleotide' : [],\n",
    "    'delta_logodds_77_127' : [],\n",
    "    'delta_logodds_0_205' : []\n",
    "}\n",
    "\n",
    "for pd_index, row in polyadb_df.iterrows() :\n",
    "    gene = row['gene']\n",
    "    gene_id = row['gene_id']\n",
    "    \n",
    "    if pd_index % 100 == 0 :\n",
    "        print(\"Predicting variants for PAS = \" + str(gene_id) + \" (\" + str(pd_index) + \")\")\n",
    "    \n",
    "    ref_seq = row['wide_seq_ext'][175-70:175-70+205]\n",
    "    \n",
    "    chrom = row['chrom']\n",
    "    strand = row['strand']\n",
    "    site_type = row['site_type']\n",
    "    ref_usage = row['ratio']\n",
    "    \n",
    "    pas_pos = row['pas_pos']\n",
    "    \n",
    "    seq_start = 0\n",
    "    if strand == '+' :\n",
    "        seq_start = pas_pos - 70 + 1\n",
    "    else :\n",
    "        seq_start = pas_pos - (205 - 70)\n",
    "    \n",
    "    seq_end = seq_start + 205\n",
    "    \n",
    "    #Predict reference sequence with APARENT model\n",
    "    _, ref_cut_pred = aparent_model.predict(x=aparent_encoder([ref_seq]), batch_size=1)\n",
    "    \n",
    "    ref_iso_pred_77_127 = np.clip(np.sum(ref_cut_pred[0, 77:127]), 1e-7, 1. - 1e-7)\n",
    "    ref_iso_pred_0_205 = np.clip(np.sum(ref_cut_pred[0, 0:205]), 1e-7, 1. - 1e-7)\n",
    "    \n",
    "    #Predict all variants\n",
    "    var_seqs = []\n",
    "    for pos in range(mutagenesis_end) :\n",
    "        for base in ['A', 'C', 'G', 'T'] :\n",
    "            var_seq = ref_seq[:pos] + base + ref_seq[pos+1:]\n",
    "            if var_seq == ref_seq :\n",
    "                continue\n",
    "            \n",
    "            var_seqs.append(var_seq)\n",
    "    \n",
    "    _, var_cut_preds = aparent_model.predict(x=aparent_encoder(var_seqs), batch_size=len(var_seqs))\n",
    "    \n",
    "    #Predict all variants\n",
    "    ii = 0\n",
    "    for pos in range(mutagenesis_end) :\n",
    "        for base in ['A', 'C', 'G', 'T'] :\n",
    "            var_seq = ref_seq[:pos] + base + ref_seq[pos+1:]\n",
    "            if var_seq == ref_seq :\n",
    "                continue\n",
    "            \n",
    "            #Predict variant sequence with APARENT model\n",
    "            var_cut_pred = var_cut_preds[ii, :]\n",
    "            \n",
    "            var_iso_pred_77_127 = np.clip(np.sum(var_cut_pred[77:127]), 1e-7, 1. - 1e-7)\n",
    "            delta_logodds_77_127 = np.log(var_iso_pred_77_127 / (1. - var_iso_pred_77_127)) - np.log(ref_iso_pred_77_127 / (1. - ref_iso_pred_77_127))\n",
    "            \n",
    "            var_iso_pred_0_205 = np.clip(np.sum(var_cut_pred[0:205]), 1e-7, 1. - 1e-7)\n",
    "            delta_logodds_0_205 = np.log(var_iso_pred_0_205 / (1. - var_iso_pred_0_205)) - np.log(ref_iso_pred_0_205 / (1. - ref_iso_pred_0_205))\n",
    "            \n",
    "            var_position = 0\n",
    "            var_nucleotide = 'A'\n",
    "            if strand == '+' :\n",
    "                var_position = seq_start + pos\n",
    "                var_nucleotide = base\n",
    "            else :\n",
    "                var_position = seq_end - pos\n",
    "                if base == 'A' :\n",
    "                    var_nucleotide = 'T'\n",
    "                elif base == 'C' :\n",
    "                    var_nucleotide = 'G'\n",
    "                elif base == 'G' :\n",
    "                    var_nucleotide = 'C'\n",
    "                elif base == 'T' :\n",
    "                    var_nucleotide = 'A'\n",
    "            \n",
    "            ref_nucleotide = 'A'\n",
    "            if strand == '+' :\n",
    "                ref_nucleotide = ref_seq[pos]\n",
    "            else :\n",
    "                if ref_seq[pos] == 'A' :\n",
    "                    ref_nucleotide = 'T'\n",
    "                elif ref_seq[pos] == 'C' :\n",
    "                    ref_nucleotide = 'G'\n",
    "                elif ref_seq[pos] == 'G' :\n",
    "                    ref_nucleotide = 'C'\n",
    "                elif ref_seq[pos] == 'T' :\n",
    "                    ref_nucleotide = 'A'\n",
    "            \n",
    "            if np.abs(delta_logodds_77_127) >= min_abs_logodds :\n",
    "                variant_dict['gene'].append(gene)\n",
    "                variant_dict['gene_id'].append(gene_id)\n",
    "                #variant_dict['ref_seq'].append(ref_seq)\n",
    "                variant_dict['chrom'].append(chrom)\n",
    "                variant_dict['strand'].append(strand)\n",
    "                variant_dict['site_type'].append(site_type)\n",
    "                variant_dict['native_usage'].append(np.round(ref_usage, 5))\n",
    "                #variant_dict['var_seq'].append(var_seq)\n",
    "                variant_dict['var_position'].append(var_position)\n",
    "                variant_dict['ref_nucleotide'].append(ref_nucleotide)\n",
    "                variant_dict['var_nucleotide'].append(var_nucleotide)\n",
    "                variant_dict['delta_logodds_77_127'].append(np.round(delta_logodds_77_127, 5))\n",
    "                variant_dict['delta_logodds_0_205'].append(np.round(delta_logodds_0_205, 5))\n",
    "            \n",
    "            ii += 1\n",
    "\n",
    "variant_df = pd.DataFrame(variant_dict)\n",
    "variant_df = variant_df[['gene','gene_id','chrom','strand','site_type','native_usage','var_position','ref_nucleotide','var_nucleotide','delta_logodds_77_127','delta_logodds_0_205']]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Store variant prediction dataframe\n",
    "\n",
    "variant_df.to_csv('aparent_resnet_variant_predictions_polyadb_no_sequences_no_cutoff.csv', sep='\\t')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>chrom</th>\n",
       "      <th>strand</th>\n",
       "      <th>site_type</th>\n",
       "      <th>native_usage</th>\n",
       "      <th>var_position</th>\n",
       "      <th>ref_nucleotide</th>\n",
       "      <th>var_nucleotide</th>\n",
       "      <th>delta_logodds_77_127</th>\n",
       "      <th>delta_logodds_0_205</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A1BG</td>\n",
       "      <td>A1BG.1</td>\n",
       "      <td>chr19</td>\n",
       "      <td>-</td>\n",
       "      <td>Intron</td>\n",
       "      <td>0.06413</td>\n",
       "      <td>58859922</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>0.01185</td>\n",
       "      <td>0.00747</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A1BG</td>\n",
       "      <td>A1BG.1</td>\n",
       "      <td>chr19</td>\n",
       "      <td>-</td>\n",
       "      <td>Intron</td>\n",
       "      <td>0.06413</td>\n",
       "      <td>58859922</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>-0.00505</td>\n",
       "      <td>-0.00600</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A1BG</td>\n",
       "      <td>A1BG.1</td>\n",
       "      <td>chr19</td>\n",
       "      <td>-</td>\n",
       "      <td>Intron</td>\n",
       "      <td>0.06413</td>\n",
       "      <td>58859922</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>0.09807</td>\n",
       "      <td>0.09027</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>A1BG</td>\n",
       "      <td>A1BG.1</td>\n",
       "      <td>chr19</td>\n",
       "      <td>-</td>\n",
       "      <td>Intron</td>\n",
       "      <td>0.06413</td>\n",
       "      <td>58859921</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>-0.02839</td>\n",
       "      <td>-0.01798</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>A1BG</td>\n",
       "      <td>A1BG.1</td>\n",
       "      <td>chr19</td>\n",
       "      <td>-</td>\n",
       "      <td>Intron</td>\n",
       "      <td>0.06413</td>\n",
       "      <td>58859921</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>-0.02564</td>\n",
       "      <td>-0.02576</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>76847533</th>\n",
       "      <td>ZZZ3</td>\n",
       "      <td>ZZZ3.26</td>\n",
       "      <td>chr1</td>\n",
       "      <td>-</td>\n",
       "      <td>3_most_exon</td>\n",
       "      <td>0.00257</td>\n",
       "      <td>78022097</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>-0.04053</td>\n",
       "      <td>-0.01775</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>76847534</th>\n",
       "      <td>ZZZ3</td>\n",
       "      <td>ZZZ3.26</td>\n",
       "      <td>chr1</td>\n",
       "      <td>-</td>\n",
       "      <td>3_most_exon</td>\n",
       "      <td>0.00257</td>\n",
       "      <td>78022097</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.00094</td>\n",
       "      <td>0.00233</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>76847535</th>\n",
       "      <td>ZZZ3</td>\n",
       "      <td>ZZZ3.26</td>\n",
       "      <td>chr1</td>\n",
       "      <td>-</td>\n",
       "      <td>3_most_exon</td>\n",
       "      <td>0.00257</td>\n",
       "      <td>78022096</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>0.01273</td>\n",
       "      <td>0.01195</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>76847536</th>\n",
       "      <td>ZZZ3</td>\n",
       "      <td>ZZZ3.26</td>\n",
       "      <td>chr1</td>\n",
       "      <td>-</td>\n",
       "      <td>3_most_exon</td>\n",
       "      <td>0.00257</td>\n",
       "      <td>78022096</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.03467</td>\n",
       "      <td>0.01759</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>76847537</th>\n",
       "      <td>ZZZ3</td>\n",
       "      <td>ZZZ3.26</td>\n",
       "      <td>chr1</td>\n",
       "      <td>-</td>\n",
       "      <td>3_most_exon</td>\n",
       "      <td>0.00257</td>\n",
       "      <td>78022096</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>0.01696</td>\n",
       "      <td>0.01061</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>76847538 rows × 11 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          gene  gene_id  chrom strand    site_type  native_usage  \\\n",
       "0         A1BG   A1BG.1  chr19      -       Intron       0.06413   \n",
       "1         A1BG   A1BG.1  chr19      -       Intron       0.06413   \n",
       "2         A1BG   A1BG.1  chr19      -       Intron       0.06413   \n",
       "3         A1BG   A1BG.1  chr19      -       Intron       0.06413   \n",
       "4         A1BG   A1BG.1  chr19      -       Intron       0.06413   \n",
       "...        ...      ...    ...    ...          ...           ...   \n",
       "76847533  ZZZ3  ZZZ3.26   chr1      -  3_most_exon       0.00257   \n",
       "76847534  ZZZ3  ZZZ3.26   chr1      -  3_most_exon       0.00257   \n",
       "76847535  ZZZ3  ZZZ3.26   chr1      -  3_most_exon       0.00257   \n",
       "76847536  ZZZ3  ZZZ3.26   chr1      -  3_most_exon       0.00257   \n",
       "76847537  ZZZ3  ZZZ3.26   chr1      -  3_most_exon       0.00257   \n",
       "\n",
       "          var_position ref_nucleotide var_nucleotide  delta_logodds_77_127  \\\n",
       "0             58859922              T              G               0.01185   \n",
       "1             58859922              T              C              -0.00505   \n",
       "2             58859922              T              A               0.09807   \n",
       "3             58859921              A              T              -0.02839   \n",
       "4             58859921              A              G              -0.02564   \n",
       "...                ...            ...            ...                   ...   \n",
       "76847533      78022097              G              C              -0.04053   \n",
       "76847534      78022097              G              A               0.00094   \n",
       "76847535      78022096              A              T               0.01273   \n",
       "76847536      78022096              A              G               0.03467   \n",
       "76847537      78022096              A              C               0.01696   \n",
       "\n",
       "          delta_logodds_0_205  \n",
       "0                     0.00747  \n",
       "1                    -0.00600  \n",
       "2                     0.09027  \n",
       "3                    -0.01798  \n",
       "4                    -0.02576  \n",
       "...                       ...  \n",
       "76847533             -0.01775  \n",
       "76847534              0.00233  \n",
       "76847535              0.01195  \n",
       "76847536              0.01759  \n",
       "76847537              0.01061  \n",
       "\n",
       "[76847538 rows x 11 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\n",
    "variant_df\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Environment (conda_tensorflow_p36)",
   "language": "python",
   "name": "conda_tensorflow_p36"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
