{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Using TensorFlow backend.\n"
     ]
    }
   ],
   "source": [
    "from __future__ import print_function\n",
    "import keras\n",
    "from keras.models import Sequential, Model, load_model\n",
    "from keras import backend as K\n",
    "\n",
    "import tensorflow as tf\n",
    "\n",
    "import os\n",
    "import pandas as pd\n",
    "\n",
    "import numpy as np\n",
    "\n",
    "import matplotlib.pyplot as plt\n",
    "import matplotlib.cm as cm\n",
    "\n",
    "import aparent.visualization as vis\n",
    "\n",
    "from aparent.predictor import *\n",
    "\n",
    "import urllib\n",
    "import urllib.request\n",
    "import pickle\n",
    "from time import sleep\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "len(polyadb_df) = 190117\n"
     ]
    }
   ],
   "source": [
    "#Load APADB Data and filter on targeted genes\n",
    "\n",
    "#genes = ['RUNX1', 'CEBPA', 'GATA2', 'ANKRD26', 'DDX41', 'ETV6', 'PTEN', 'BRCA1', 'BRCA2', 'TP53', 'APC', 'ATM', 'PALB2', 'MSH2', 'MLH1', 'MSH6', 'PMS2', 'MUTYH']\n",
    "\n",
    "polyadb_df = pd.read_csv('../../data/native_data/polyadb/processed_data/polyadb_processed.csv', sep=',')\n",
    "\n",
    "#polyadb_df = polyadb_df.loc[polyadb_df['gene'].isin(genes)].reset_index(drop=True).copy()\n",
    "polyadb_df = polyadb_df.loc[~polyadb_df['gene'].isna()].reset_index(drop=True).copy()\n",
    "polyadb_df = polyadb_df.loc[polyadb_df['gene'] != 'na'].reset_index(drop=True).copy()\n",
    "\n",
    "print('len(polyadb_df) = ' + str(len(polyadb_df)))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "polyadb_df_minus = polyadb_df.query(\"strand == '-'\").copy().reset_index(drop=True)\n",
    "polyadb_df_plus = polyadb_df.query(\"strand == '+'\").copy().reset_index(drop=True)\n",
    "\n",
    "polyadb_df_minus = polyadb_df_minus.sort_values(by='pas_pos', ascending=False).copy().reset_index(drop=True)\n",
    "polyadb_df_plus = polyadb_df_plus.sort_values(by='pas_pos', ascending=True).copy().reset_index(drop=True)\n",
    "\n",
    "new_gene_id_list_plus = []\n",
    "sitenum_list_plus = []\n",
    "gene_id_dict = {}\n",
    "for _, row in polyadb_df_plus.iterrows() :\n",
    "\n",
    "    gene = str(row['gene'])\n",
    "\n",
    "    if gene not in gene_id_dict :\n",
    "        gene_id_dict[gene] = 0\n",
    "\n",
    "    gene_id_dict[gene] += 1\n",
    "\n",
    "    new_gene_id_list_plus.append(gene + \".\" + str(gene_id_dict[gene]))\n",
    "    sitenum_list_plus.append(gene_id_dict[gene])\n",
    "\n",
    "polyadb_df_plus['gene_id'] = new_gene_id_list_plus\n",
    "polyadb_df_plus['sitenum'] = sitenum_list_plus\n",
    "\n",
    "new_gene_id_list_minus = []\n",
    "sitenum_list_minus = []\n",
    "gene_id_dict = {}\n",
    "for _, row in polyadb_df_minus.iterrows() :\n",
    "\n",
    "    gene = str(row['gene'])\n",
    "\n",
    "    if gene not in gene_id_dict :\n",
    "        gene_id_dict[gene] = 0\n",
    "\n",
    "    gene_id_dict[gene] += 1\n",
    "\n",
    "    new_gene_id_list_minus.append(gene + \".\" + str(gene_id_dict[gene]))\n",
    "    sitenum_list_minus.append(gene_id_dict[gene])\n",
    "\n",
    "polyadb_df_minus['gene_id'] = new_gene_id_list_minus\n",
    "polyadb_df_minus['sitenum'] = sitenum_list_minus\n",
    "\n",
    "polyadb_df = pd.concat([polyadb_df_plus, polyadb_df_minus])\n",
    "\n",
    "polyadb_df = polyadb_df.sort_values(by=['gene', 'sitenum'], ascending=True).reset_index(drop=True).copy()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>sitenum</th>\n",
       "      <th>num_sites</th>\n",
       "      <th>pas</th>\n",
       "      <th>seq</th>\n",
       "      <th>seq_ext</th>\n",
       "      <th>wide_seq</th>\n",
       "      <th>wide_seq_ext</th>\n",
       "      <th>rpm</th>\n",
       "      <th>total_rpm</th>\n",
       "      <th>ratio</th>\n",
       "      <th>pas_pos</th>\n",
       "      <th>cut_mode</th>\n",
       "      <th>chrom</th>\n",
       "      <th>strand</th>\n",
       "      <th>site_type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A1BG</td>\n",
       "      <td>A1BG.1</td>\n",
       "      <td>1</td>\n",
       "      <td>11</td>\n",
       "      <td>0</td>\n",
       "      <td>TTCTGCTAGGCTGGGCTAGTATCTTCCATGGCAAGATACTCAAACT...</td>\n",
       "      <td>TAGGGAATATAACTTTTAAAATGCGAAAGTATATTATACATATTGC...</td>\n",
       "      <td>AGTTTGGCTAGCATTTCAAAATGGATAGGGAATATAACTTTTAAAA...</td>\n",
       "      <td>GCTTTCAAAATTCCTGATTAGCTTACAATTAATTGCAAATAACTTC...</td>\n",
       "      <td>1.564234</td>\n",
       "      <td>24.390222</td>\n",
       "      <td>0.064134</td>\n",
       "      <td>58859852</td>\n",
       "      <td>58859830</td>\n",
       "      <td>chr19</td>\n",
       "      <td>-</td>\n",
       "      <td>Intron</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A1BG</td>\n",
       "      <td>A1BG.2</td>\n",
       "      <td>2</td>\n",
       "      <td>11</td>\n",
       "      <td>-1</td>\n",
       "      <td>AGGTGCCGCTACCGCTCCTGGGTGCCCCACACCTTCGAATCGGAGC...</td>\n",
       "      <td>CGGCGAACCTCGAGCTGATCTTCGTGGGGCCCCAGCACGCCGGCAA...</td>\n",
       "      <td>AAGACGGTCCGCACCCCCGGGGCCGCGGCGAACCTCGAGCTGATCT...</td>\n",
       "      <td>TCCCCGACGTCACCTTCGAGCTGCTGCGCGAGGGCGAGACGAAGGC...</td>\n",
       "      <td>1.740036</td>\n",
       "      <td>24.390222</td>\n",
       "      <td>0.071342</td>\n",
       "      <td>58858744</td>\n",
       "      <td>58858719</td>\n",
       "      <td>chr19</td>\n",
       "      <td>-</td>\n",
       "      <td>Internal_exon</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A1BG</td>\n",
       "      <td>A1BG.3</td>\n",
       "      <td>3</td>\n",
       "      <td>11</td>\n",
       "      <td>0</td>\n",
       "      <td>CCGGGGTCTCTGGGGGGCTGGAGAAGCCTCCCTCATTCCTCCCAGG...</td>\n",
       "      <td>AGTGCCGGGGATTCTGGACTGGCTCCCTCCCCTCCTGTTGCAGCAC...</td>\n",
       "      <td>CCAGGGTGCTGTTGGTGTCCTCAGAAGTGCCGGGGATTCTGGACTG...</td>\n",
       "      <td>CACCATTTCTCTTTTCTCCCGTTGTCTAGAAAGCTGATGCAGCCGC...</td>\n",
       "      <td>11.529986</td>\n",
       "      <td>24.390222</td>\n",
       "      <td>0.472730</td>\n",
       "      <td>58858249</td>\n",
       "      <td>58858220</td>\n",
       "      <td>chr19</td>\n",
       "      <td>-</td>\n",
       "      <td>3_most_exon</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>A1BG</td>\n",
       "      <td>A1BG.4</td>\n",
       "      <td>4</td>\n",
       "      <td>11</td>\n",
       "      <td>4</td>\n",
       "      <td>GAAGATGCGTATCAGTATTATCAACCAATAGTTGATGAATTGTGTT...</td>\n",
       "      <td>ACCACTGCCGAGTTTCTGCCTGGAGCAGGGCCTCAAGGCCCTGGCA...</td>\n",
       "      <td>CCATCTGACATCTGATGCCTTCCCAACCACTGCCGAGTTTCTGCCT...</td>\n",
       "      <td>GATTCCTCTGTGTTCTTTCCCATTATCCCTCAGCTCTGAGTCCTCT...</td>\n",
       "      <td>1.386686</td>\n",
       "      <td>24.390222</td>\n",
       "      <td>0.056854</td>\n",
       "      <td>58857215</td>\n",
       "      <td>58857210</td>\n",
       "      <td>chr19</td>\n",
       "      <td>-</td>\n",
       "      <td>3_most_exon</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>A1BG</td>\n",
       "      <td>A1BG.5</td>\n",
       "      <td>5</td>\n",
       "      <td>11</td>\n",
       "      <td>4</td>\n",
       "      <td>AACGAATTTGCTAGTGATCTGGTTTACTGCCTTAGTAATATCTAGT...</td>\n",
       "      <td>GAAGATGCGTATCAGTATTATCAACCAATAGTTGATGAATTGTGTT...</td>\n",
       "      <td>CAGGGCCTCAAGGCCCTGGCACACAGAAGATGCGTATCAGTATTAT...</td>\n",
       "      <td>CCATCTGACATCTGATGCCTTCCCAACCACTGCCGAGTTTCTGCCT...</td>\n",
       "      <td>1.174863</td>\n",
       "      <td>24.390222</td>\n",
       "      <td>0.048169</td>\n",
       "      <td>58857165</td>\n",
       "      <td>58857142</td>\n",
       "      <td>chr19</td>\n",
       "      <td>-</td>\n",
       "      <td>3_most_exon</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   gene gene_id  sitenum  num_sites  pas  \\\n",
       "0  A1BG  A1BG.1        1         11    0   \n",
       "1  A1BG  A1BG.2        2         11   -1   \n",
       "2  A1BG  A1BG.3        3         11    0   \n",
       "3  A1BG  A1BG.4        4         11    4   \n",
       "4  A1BG  A1BG.5        5         11    4   \n",
       "\n",
       "                                                 seq  \\\n",
       "0  TTCTGCTAGGCTGGGCTAGTATCTTCCATGGCAAGATACTCAAACT...   \n",
       "1  AGGTGCCGCTACCGCTCCTGGGTGCCCCACACCTTCGAATCGGAGC...   \n",
       "2  CCGGGGTCTCTGGGGGGCTGGAGAAGCCTCCCTCATTCCTCCCAGG...   \n",
       "3  GAAGATGCGTATCAGTATTATCAACCAATAGTTGATGAATTGTGTT...   \n",
       "4  AACGAATTTGCTAGTGATCTGGTTTACTGCCTTAGTAATATCTAGT...   \n",
       "\n",
       "                                             seq_ext  \\\n",
       "0  TAGGGAATATAACTTTTAAAATGCGAAAGTATATTATACATATTGC...   \n",
       "1  CGGCGAACCTCGAGCTGATCTTCGTGGGGCCCCAGCACGCCGGCAA...   \n",
       "2  AGTGCCGGGGATTCTGGACTGGCTCCCTCCCCTCCTGTTGCAGCAC...   \n",
       "3  ACCACTGCCGAGTTTCTGCCTGGAGCAGGGCCTCAAGGCCCTGGCA...   \n",
       "4  GAAGATGCGTATCAGTATTATCAACCAATAGTTGATGAATTGTGTT...   \n",
       "\n",
       "                                            wide_seq  \\\n",
       "0  AGTTTGGCTAGCATTTCAAAATGGATAGGGAATATAACTTTTAAAA...   \n",
       "1  AAGACGGTCCGCACCCCCGGGGCCGCGGCGAACCTCGAGCTGATCT...   \n",
       "2  CCAGGGTGCTGTTGGTGTCCTCAGAAGTGCCGGGGATTCTGGACTG...   \n",
       "3  CCATCTGACATCTGATGCCTTCCCAACCACTGCCGAGTTTCTGCCT...   \n",
       "4  CAGGGCCTCAAGGCCCTGGCACACAGAAGATGCGTATCAGTATTAT...   \n",
       "\n",
       "                                        wide_seq_ext        rpm  total_rpm  \\\n",
       "0  GCTTTCAAAATTCCTGATTAGCTTACAATTAATTGCAAATAACTTC...   1.564234  24.390222   \n",
       "1  TCCCCGACGTCACCTTCGAGCTGCTGCGCGAGGGCGAGACGAAGGC...   1.740036  24.390222   \n",
       "2  CACCATTTCTCTTTTCTCCCGTTGTCTAGAAAGCTGATGCAGCCGC...  11.529986  24.390222   \n",
       "3  GATTCCTCTGTGTTCTTTCCCATTATCCCTCAGCTCTGAGTCCTCT...   1.386686  24.390222   \n",
       "4  CCATCTGACATCTGATGCCTTCCCAACCACTGCCGAGTTTCTGCCT...   1.174863  24.390222   \n",
       "\n",
       "      ratio   pas_pos  cut_mode  chrom strand      site_type  \n",
       "0  0.064134  58859852  58859830  chr19      -         Intron  \n",
       "1  0.071342  58858744  58858719  chr19      -  Internal_exon  \n",
       "2  0.472730  58858249  58858220  chr19      -    3_most_exon  \n",
       "3  0.056854  58857215  58857210  chr19      -    3_most_exon  \n",
       "4  0.048169  58857165  58857142  chr19      -    3_most_exon  "
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "polyadb_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "polyadb_df['start'] = polyadb_df['pas_pos'] - 150\n",
    "polyadb_df['end'] = polyadb_df['pas_pos'] + 150\n",
    "\n",
    "polyadb_bed = polyadb_df[['chrom', 'start', 'end', 'gene', 'gene_id', 'strand']].copy().reset_index(drop=True)\n",
    "\n",
    "polyadb_bed.head()\n",
    "\n",
    "polyadb_bed.to_csv(\"polyadb_coordinates.bed\", sep='\\t', header=False, index=False)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Tissue types to compile data for\n",
    "\n",
    "tissue_types = [\n",
    "    'Adipose_Subcutaneous',\n",
    "    'Adipose_Visceral_Omentum',\n",
    "    'Adrenal_Gland',\n",
    "    'Artery_Aorta',\n",
    "    'Artery_Coronary',\n",
    "    'Artery_Tibial',\n",
    "    'Brain_Amygdala',\n",
    "    'Brain_Anterior_cingulate_cortex_BA24',\n",
    "    'Brain_Caudate_basal_ganglia',\n",
    "    'Brain_Cerebellar_Hemisphere',\n",
    "    #'Brain_Cerebellum',\n",
    "    'Brain_Cortex',\n",
    "    'Brain_Frontal_Cortex_BA9',\n",
    "    'Brain_Hippocampus',\n",
    "    'Brain_Hypothalamus',\n",
    "    'Brain_Nucleus_accumbens_basal_ganglia',\n",
    "    'Brain_Putamen_basal_ganglia',\n",
    "    'Brain_Spinal_cord_cervical_c-1',\n",
    "    'Breast_Mammary_Tissue',\n",
    "    'Cells_EBV-transformed_lymphocytes',\n",
    "    'Cells_Transformed_fibroblasts',\n",
    "    'Colon_Sigmoid',\n",
    "    'Colon_Transverse',\n",
    "    'Esophagus_Gastroesophageal_Junction',\n",
    "    'Esophagus_Mucosa',\n",
    "    'Esophagus_Muscularis',\n",
    "    #'Heart_Atrial_Appendage',\n",
    "    'Heart_Left_Ventricle',\n",
    "    'Liver',\n",
    "    'Lung',\n",
    "    'Muscle_Skeletal',\n",
    "    'Nerve_Tibial',\n",
    "    'Ovary',\n",
    "    'Pancreas',\n",
    "    'Pituitary',\n",
    "    'Prostate',\n",
    "    'Skin_Not_Sun_Exposed_Suprapubic',\n",
    "    'Skin_Sun_Exposed_Lower_leg',\n",
    "    'Small_Intestine_Terminal_Ileum',\n",
    "    'Spleen',\n",
    "    'Stomach',\n",
    "    'Testis',\n",
    "    'Thyroid',\n",
    "    'Uterus',\n",
    "    'Vagina',\n",
    "    'Whole_Blood'\n",
    "]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "version_suffix = '_lead'\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Processing data for tissue = 'Adipose_Subcutaneous'.\n",
      "Processing data for tissue = 'Adipose_Visceral_Omentum'.\n",
      "Processing data for tissue = 'Adrenal_Gland'.\n",
      "Processing data for tissue = 'Artery_Aorta'.\n",
      "Processing data for tissue = 'Artery_Coronary'.\n",
      "Processing data for tissue = 'Artery_Tibial'.\n",
      "Processing data for tissue = 'Brain_Amygdala'.\n",
      "Processing data for tissue = 'Brain_Anterior_cingulate_cortex_BA24'.\n",
      "Processing data for tissue = 'Brain_Caudate_basal_ganglia'.\n",
      "Processing data for tissue = 'Brain_Cerebellar_Hemisphere'.\n",
      "Processing data for tissue = 'Brain_Cortex'.\n",
      "Processing data for tissue = 'Brain_Frontal_Cortex_BA9'.\n",
      "Processing data for tissue = 'Brain_Hippocampus'.\n",
      "Processing data for tissue = 'Brain_Hypothalamus'.\n",
      "Processing data for tissue = 'Brain_Nucleus_accumbens_basal_ganglia'.\n",
      "Processing data for tissue = 'Brain_Putamen_basal_ganglia'.\n",
      "Processing data for tissue = 'Brain_Spinal_cord_cervical_c-1'.\n",
      "Processing data for tissue = 'Breast_Mammary_Tissue'.\n",
      "Processing data for tissue = 'Cells_EBV-transformed_lymphocytes'.\n",
      "Processing data for tissue = 'Cells_Transformed_fibroblasts'.\n",
      "Processing data for tissue = 'Colon_Sigmoid'.\n",
      "Processing data for tissue = 'Colon_Transverse'.\n",
      "Processing data for tissue = 'Esophagus_Gastroesophageal_Junction'.\n",
      "Processing data for tissue = 'Esophagus_Mucosa'.\n",
      "Processing data for tissue = 'Esophagus_Muscularis'.\n",
      "Processing data for tissue = 'Heart_Left_Ventricle'.\n",
      "Processing data for tissue = 'Liver'.\n",
      "Processing data for tissue = 'Lung'.\n",
      "Processing data for tissue = 'Muscle_Skeletal'.\n",
      "Processing data for tissue = 'Nerve_Tibial'.\n",
      "Processing data for tissue = 'Ovary'.\n",
      "Processing data for tissue = 'Pancreas'.\n",
      "Processing data for tissue = 'Pituitary'.\n",
      "Processing data for tissue = 'Prostate'.\n",
      "Processing data for tissue = 'Skin_Not_Sun_Exposed_Suprapubic'.\n",
      "Processing data for tissue = 'Skin_Sun_Exposed_Lower_leg'.\n",
      "Processing data for tissue = 'Small_Intestine_Terminal_Ileum'.\n",
      "Processing data for tissue = 'Spleen'.\n",
      "Processing data for tissue = 'Stomach'.\n",
      "Processing data for tissue = 'Testis'.\n",
      "Processing data for tissue = 'Thyroid'.\n",
      "Processing data for tissue = 'Uterus'.\n",
      "Processing data for tissue = 'Vagina'.\n",
      "Processing data for tissue = 'Whole_Blood'.\n"
     ]
    }
   ],
   "source": [
    "#Compile apaQTL data from GTEx\n",
    "\n",
    "def get_snp_pos(row) :\n",
    "    if row['strand'] == '+' :\n",
    "        return row['snp_pos'] - row['pas_pos'] + 175 - 1\n",
    "    else :\n",
    "        return row['pas_pos'] - row['snp_pos'] + 175\n",
    "\n",
    "def get_snp_ref(row) :\n",
    "    if row['strand'] == '+' :\n",
    "        return row['snp_ref']\n",
    "    else :\n",
    "        if row['snp_ref'] == 'A' :\n",
    "            return 'T'\n",
    "        if row['snp_ref'] == 'C' :\n",
    "            return 'G'\n",
    "        if row['snp_ref'] == 'G' :\n",
    "            return 'C'\n",
    "        if row['snp_ref'] == 'T' :\n",
    "            return 'A'\n",
    "\n",
    "def get_snp_var(row) :\n",
    "    if row['strand'] == '+' :\n",
    "        return row['snp_var']\n",
    "    else :\n",
    "        if row['snp_var'] == 'A' :\n",
    "            return 'T'\n",
    "        if row['snp_var'] == 'C' :\n",
    "            return 'G'\n",
    "        if row['snp_var'] == 'G' :\n",
    "            return 'C'\n",
    "        if row['snp_var'] == 'T' :\n",
    "            return 'A'\n",
    "\n",
    "def get_var_seq(row) :\n",
    "    ref_seq = row['wide_seq_ext']\n",
    "    var_seq = ref_seq[:row['rel_snp_pos']] + row['rel_snp_var'] + ref_seq[row['rel_snp_pos']+1:]\n",
    "    \n",
    "    return var_seq\n",
    "\n",
    "for tissue_type in tissue_types :\n",
    "    \n",
    "    print(\"Processing data for tissue = '\" + str(tissue_type) + \"'.\")\n",
    "    \n",
    "    snp_df = pd.read_csv(tissue_type + '.cis.3aQTL.txt', sep='\\t')\n",
    "\n",
    "    snp_df['chr'] = snp_df['SNP'].apply(lambda x: x.split(\"_\")[0])\n",
    "    snp_df['start'] = snp_df['SNP'].apply(lambda x: x.split(\"_\")[1])\n",
    "    snp_df['end'] = snp_df['SNP'].apply(lambda x: str(int(x.split(\"_\")[1]) + 1))\n",
    "    snp_df['gene'] = snp_df['transcript'].apply(lambda x: x.split(\"|\")[1])\n",
    "\n",
    "    snp_df = snp_df[['chr', 'start', 'end', 'gene', 'SNP', 'transcript', 'beta', 't.stat', 'p.value']]\n",
    "\n",
    "    snp_df.to_csv(tissue_type + version_suffix + \".cis.3aQTL.coordinates.bed\", sep='\\t', header=False, index=False)\n",
    "\n",
    "    !bedtools intersect -a polyadb_coordinates.bed -b {tissue_type + version_suffix}.cis.3aQTL.coordinates.bed -wa -wb > polyadb_{tissue_type + version_suffix}_intersect.bed\n",
    "\n",
    "    snp_bed_hg19 = pd.read_csv(\"polyadb_\" + tissue_type + \"_intersect.bed\", sep='\\t', error_bad_lines=False, names=['chrom', 'start', 'end', 'gene', 'gene_id', 'strand', 'chr_2', 'snp_pos', 'end_2', 'gene_2', 'snp_id', 'transcript_id', 'effect_size', 'test_statistic', 'p_val'])\n",
    "    snp_bed_hg19 = snp_bed_hg19.query(\"gene == gene_2\").copy().reset_index(drop=True)\n",
    "\n",
    "    snp_bed_hg19 = snp_bed_hg19[['chrom', 'start', 'end', 'gene', 'gene_id', 'strand', 'chr_2', 'snp_pos', 'end_2', 'gene_2', 'snp_id', 'transcript_id', 'effect_size', 'test_statistic', 'p_val']]\n",
    "\n",
    "    snp_bed_hg19['snp_ref'] = snp_bed_hg19['snp_id'].apply(lambda s: s.split(\"_\")[2])\n",
    "    snp_bed_hg19['snp_var'] = snp_bed_hg19['snp_id'].apply(lambda s: s.split(\"_\")[3])\n",
    "    \n",
    "    polyadb_df_sel = polyadb_df[['wide_seq_ext', 'gene_id', 'pas', 'pas_pos', 'cut_mode', 'site_type', 'rpm', 'total_rpm', 'ratio']].copy().reset_index(drop=True)\n",
    "\n",
    "    snp_df = snp_bed_hg19.join(polyadb_df_sel.set_index('gene_id'), on='gene_id').copy().reset_index(drop=True)\n",
    "    \n",
    "    snp_df['rel_snp_pos'] = snp_df.apply(get_snp_pos, axis=1)\n",
    "    snp_df['rel_snp_ref'] = snp_df.apply(get_snp_ref, axis=1)\n",
    "    snp_df['rel_snp_var'] = snp_df.apply(get_snp_var, axis=1)\n",
    "\n",
    "    snp_df['wide_seq_ext_var'] = snp_df.apply(get_var_seq, axis=1)\n",
    "    \n",
    "    #Filter by lead SNPs\n",
    "    lead_df = pd.read_csv(\"lead/\" + tissue_type + \".lead.3aQTL.txt\", sep='\\t')\n",
    "    snp_df = snp_df.join(lead_df[['SNP']].set_index(\"SNP\"), on='snp_id', how='inner')\n",
    "    \n",
    "    #Calculate mean PDUIs\n",
    "    pdui_df = pd.read_csv(tissue_type + \"_combined_All_PDUIs_clean.txt\", sep='\\t')\n",
    "    pdui_df = pdui_df.join(snp_df[['transcript_id']].set_index('transcript_id'), on='event_id', how='inner').copy().reset_index(drop=True)\n",
    "\n",
    "    gtex_samples = pdui_df.columns.values.tolist()[2:]\n",
    "\n",
    "    def calc_mean_PDUI(row) :\n",
    "\n",
    "        n_samples = 0.\n",
    "        PDUI = 0.\n",
    "\n",
    "        for gtex_sample in gtex_samples :\n",
    "            if not np.isnan(row[gtex_sample]) :\n",
    "\n",
    "                PDUI += row[gtex_sample]\n",
    "                n_samples += 1.\n",
    "\n",
    "        return PDUI / n_samples\n",
    "\n",
    "    pdui_df['mean_PDUI'] = pdui_df.apply(calc_mean_PDUI, axis=1)\n",
    "\n",
    "    snp_df = snp_df.join(pdui_df[['event_id', 'mean_PDUI']].set_index('event_id'), on='transcript_id', how='inner').copy().reset_index(drop=True)\n",
    "\n",
    "    snp_df['transcript_id'] = snp_df['transcript_id'].apply(lambda x: x.split(\"|\")[0])\n",
    "    snp_df['snp_transcript_id'] = snp_df['snp_id'] + \"_\" + snp_df['transcript_id']\n",
    "    \n",
    "    transcript_end_dict = { row['transcript_id'] : -1 for _, row in snp_df.iterrows() }\n",
    "\n",
    "    with open(\"/home/jlinder2/Downloads/GRCh37_latest_genomic.gtf\", 'rt') as f :\n",
    "        for line_raw in f.readlines() :\n",
    "\n",
    "            a = line_raw.strip()\n",
    "\n",
    "            if len(a.split(\"\\t\")) > 8 :\n",
    "                start = a.strip().split(\"\\t\")[3]\n",
    "                end = a.strip().split(\"\\t\")[4]\n",
    "\n",
    "                strand = a.strip().split(\"\\t\")[6]\n",
    "\n",
    "                t_id = \"\"\n",
    "                if strand in ['+', '-'] :\n",
    "                    t_id = a.strip().split(\"\\t\")[8].split(\";\")[1].split(\"\\\"\")[1].split(\".\")[0]\n",
    "\n",
    "                if t_id in transcript_end_dict :\n",
    "\n",
    "                    if transcript_end_dict[t_id] == -1 :\n",
    "                        if strand == '+' :\n",
    "                            transcript_end_dict[t_id] = int(end)\n",
    "                        else :\n",
    "                            transcript_end_dict[t_id] = int(start)\n",
    "                    else :\n",
    "                        if strand == '+' :\n",
    "                            transcript_end_dict[t_id] = max(transcript_end_dict[t_id], int(end))\n",
    "                        else :\n",
    "                            transcript_end_dict[t_id] = min(transcript_end_dict[t_id], int(start))\n",
    "\n",
    "    qtl_cut_poses = []\n",
    "    for _, row in snp_df.iterrows() :\n",
    "\n",
    "        qtl_cut_poses.append(transcript_end_dict[row['transcript_id']])\n",
    "\n",
    "    snp_df['qtl_cut_pos'] = qtl_cut_poses\n",
    "    snp_df['snp_distance'] = np.abs(snp_df['qtl_cut_pos'] - snp_df['snp_pos'])\n",
    "\n",
    "    #Drop duplicates\n",
    "    snp_df['row_id'] = snp_df['gene_id'] + \"_\" + snp_df['snp_id'] + \"_\" + snp_df['transcript_id']\n",
    "    snp_df = snp_df.sort_values(by='snp_id').drop_duplicates(subset=['row_id'], keep='first')\n",
    "\n",
    "    #Save final apaQTL dataframe\n",
    "    snp_df.to_csv(\"polyadb_\" + tissue_type + version_suffix + \"_SNPs.csv\", sep='\\t')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Processing tissue = 'Adipose_Subcutaneous'\n",
      "Processing tissue = 'Adipose_Visceral_Omentum'\n",
      "Processing tissue = 'Adrenal_Gland'\n",
      "Processing tissue = 'Artery_Aorta'\n",
      "Processing tissue = 'Artery_Coronary'\n",
      "Processing tissue = 'Artery_Tibial'\n",
      "Processing tissue = 'Brain_Amygdala'\n",
      "Processing tissue = 'Brain_Anterior_cingulate_cortex_BA24'\n",
      "Processing tissue = 'Brain_Caudate_basal_ganglia'\n",
      "Processing tissue = 'Brain_Cerebellar_Hemisphere'\n",
      "Processing tissue = 'Brain_Cortex'\n",
      "Processing tissue = 'Brain_Frontal_Cortex_BA9'\n",
      "Processing tissue = 'Brain_Hippocampus'\n",
      "Processing tissue = 'Brain_Hypothalamus'\n",
      "Processing tissue = 'Brain_Nucleus_accumbens_basal_ganglia'\n",
      "Processing tissue = 'Brain_Putamen_basal_ganglia'\n",
      "Processing tissue = 'Brain_Spinal_cord_cervical_c-1'\n",
      "Processing tissue = 'Breast_Mammary_Tissue'\n",
      "Processing tissue = 'Cells_EBV-transformed_lymphocytes'\n",
      "Processing tissue = 'Cells_Transformed_fibroblasts'\n",
      "Processing tissue = 'Colon_Sigmoid'\n",
      "Processing tissue = 'Colon_Transverse'\n",
      "Processing tissue = 'Esophagus_Gastroesophageal_Junction'\n",
      "Processing tissue = 'Esophagus_Mucosa'\n",
      "Processing tissue = 'Esophagus_Muscularis'\n",
      "Processing tissue = 'Heart_Left_Ventricle'\n",
      "Processing tissue = 'Liver'\n",
      "Processing tissue = 'Lung'\n",
      "Processing tissue = 'Muscle_Skeletal'\n",
      "Processing tissue = 'Nerve_Tibial'\n",
      "Processing tissue = 'Ovary'\n",
      "Processing tissue = 'Pancreas'\n",
      "Processing tissue = 'Pituitary'\n",
      "Processing tissue = 'Prostate'\n",
      "Processing tissue = 'Skin_Not_Sun_Exposed_Suprapubic'\n",
      "Processing tissue = 'Skin_Sun_Exposed_Lower_leg'\n",
      "Processing tissue = 'Small_Intestine_Terminal_Ileum'\n",
      "Processing tissue = 'Spleen'\n",
      "Processing tissue = 'Stomach'\n",
      "Processing tissue = 'Testis'\n",
      "Processing tissue = 'Thyroid'\n",
      "Processing tissue = 'Uterus'\n",
      "Processing tissue = 'Vagina'\n",
      "Processing tissue = 'Whole_Blood'\n"
     ]
    }
   ],
   "source": [
    "#Concatenate tissue-specific dataframes\n",
    "\n",
    "snp_dfs = []\n",
    "\n",
    "for tissue_type in tissue_types :\n",
    "    \n",
    "    print(\"Processing tissue = '\" + tissue_type + \"'\")\n",
    "    \n",
    "    snp_df = pd.read_csv(\"polyadb_\" + tissue_type + version_suffix + \"_SNPs.csv\", sep='\\t')\n",
    "    snp_df['tissue_type'] = tissue_type\n",
    "    \n",
    "    snp_dfs.append(snp_df)\n",
    "\n",
    "snp_df = pd.concat(snp_dfs).copy().reset_index(drop=True)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "merged_df = snp_df\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "snp_dict = {}\n",
    "\n",
    "for i, tissue_type in enumerate(tissue_types) :\n",
    "    \n",
    "    snp_df = pd.read_csv(\"polyadb_\" + tissue_type + version_suffix + \"_SNPs.csv\", sep='\\t')\n",
    "    \n",
    "    for _, row in snp_df.iterrows() :\n",
    "        \n",
    "        snp_id = row['snp_id'] + \"__\" + row['gene_id'] + \"__\" + row['transcript_id']\n",
    "        \n",
    "        effect_size = row['effect_size']\n",
    "        \n",
    "        if snp_id not in snp_dict :\n",
    "            snp_dict[snp_id] = {\n",
    "                'effect_size' : []\n",
    "            }\n",
    "        \n",
    "        snp_dict[snp_id]['effect_size'].append(effect_size)\n",
    "\n",
    "snp_ids = []\n",
    "effect_sizes = []\n",
    "n_tissues = []\n",
    "\n",
    "for snp_id in snp_dict :\n",
    "    \n",
    "    snp_ids.append(snp_id)\n",
    "    effect_sizes.append(np.median(snp_dict[snp_id]['effect_size']))\n",
    "    \n",
    "    n_tissues.append(len(snp_dict[snp_id]['effect_size']))\n",
    "\n",
    "median_df = pd.DataFrame({\n",
    "    'extended_snp_id' : snp_ids,\n",
    "    'effect_size' : effect_sizes,\n",
    "    'n_tissues' : n_tissues\n",
    "})\n",
    "median_df = median_df[['extended_snp_id', 'effect_size', 'n_tissues']]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "merged_df['extended_snp_id'] = merged_df['snp_id'] + \"__\" + merged_df['gene_id'] + \"__\" + merged_df['transcript_id']\n",
    "\n",
    "merged_df = merged_df.join(median_df.set_index('extended_snp_id'), on=\"extended_snp_id\", how='inner', rsuffix='_median').copy().reset_index(drop=True)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "merged_df.to_csv(\"polyadb_merged\" + version_suffix + \"_SNPs.csv\", sep='\\t')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Excluding []...\n",
      "Processing tissue = 'Adipose_Subcutaneous'\n",
      "Processing tissue = 'Adipose_Visceral_Omentum'\n",
      "Processing tissue = 'Adrenal_Gland'\n",
      "Processing tissue = 'Artery_Aorta'\n",
      "Processing tissue = 'Artery_Coronary'\n",
      "Processing tissue = 'Artery_Tibial'\n",
      "Processing tissue = 'Brain_Amygdala'\n",
      "Processing tissue = 'Brain_Anterior_cingulate_cortex_BA24'\n",
      "Processing tissue = 'Brain_Caudate_basal_ganglia'\n",
      "Processing tissue = 'Brain_Cerebellar_Hemisphere'\n",
      "Processing tissue = 'Brain_Cortex'\n",
      "Processing tissue = 'Brain_Frontal_Cortex_BA9'\n",
      "Processing tissue = 'Brain_Hippocampus'\n",
      "Processing tissue = 'Brain_Hypothalamus'\n",
      "Processing tissue = 'Brain_Nucleus_accumbens_basal_ganglia'\n",
      "Processing tissue = 'Brain_Putamen_basal_ganglia'\n",
      "Processing tissue = 'Brain_Spinal_cord_cervical_c-1'\n",
      "Processing tissue = 'Breast_Mammary_Tissue'\n",
      "Processing tissue = 'Cells_EBV-transformed_lymphocytes'\n",
      "Processing tissue = 'Cells_Transformed_fibroblasts'\n",
      "Processing tissue = 'Colon_Sigmoid'\n",
      "Processing tissue = 'Colon_Transverse'\n",
      "Processing tissue = 'Esophagus_Gastroesophageal_Junction'\n",
      "Processing tissue = 'Esophagus_Mucosa'\n",
      "Processing tissue = 'Esophagus_Muscularis'\n",
      "Processing tissue = 'Heart_Left_Ventricle'\n",
      "Processing tissue = 'Liver'\n",
      "Processing tissue = 'Lung'\n",
      "Processing tissue = 'Muscle_Skeletal'\n",
      "Processing tissue = 'Nerve_Tibial'\n",
      "Processing tissue = 'Ovary'\n",
      "Processing tissue = 'Pancreas'\n",
      "Processing tissue = 'Pituitary'\n",
      "Processing tissue = 'Prostate'\n",
      "Processing tissue = 'Skin_Not_Sun_Exposed_Suprapubic'\n",
      "Processing tissue = 'Skin_Sun_Exposed_Lower_leg'\n",
      "Processing tissue = 'Small_Intestine_Terminal_Ileum'\n",
      "Processing tissue = 'Spleen'\n",
      "Processing tissue = 'Stomach'\n",
      "Processing tissue = 'Testis'\n",
      "Processing tissue = 'Thyroid'\n",
      "Processing tissue = 'Uterus'\n",
      "Processing tissue = 'Vagina'\n",
      "Processing tissue = 'Whole_Blood'\n"
     ]
    }
   ],
   "source": [
    "#Compute median effect sizes\n",
    "\n",
    "exclude_tissues = [[]]\n",
    "exclude_suffixes = ['']\n",
    "\n",
    "#exclude_tissues = [[], ['Brain_Cortex', 'Brain_Frontal_Cortex_BA9'], ['Testis']]\n",
    "#exclude_suffixes = ['', '_Exclude_Brain', '_Exclude_Testis']\n",
    "\n",
    "for exclude_tissue, exclude_suffix in zip(exclude_tissues, exclude_suffixes) :\n",
    "\n",
    "    print(\"Excluding \" + str(exclude_tissue) + \"...\")\n",
    "    \n",
    "    snp_dict = {}\n",
    "\n",
    "    for i, tissue_type in enumerate(tissue_types) :\n",
    "        \n",
    "        if tissue_type in exclude_tissue :\n",
    "            continue\n",
    "\n",
    "        print(\"Processing tissue = '\" + tissue_type + \"'\")\n",
    "\n",
    "        snp_df = pd.read_csv(\"polyadb_\" + tissue_type + version_suffix + \"_SNPs.csv\", sep='\\t')\n",
    "\n",
    "        for _, row in snp_df.iterrows() :\n",
    "\n",
    "            snp_id = row['snp_id'] + \"__\" + row['gene_id'] + \"__\" + row['transcript_id']\n",
    "\n",
    "            effect_size = row['effect_size']\n",
    "            p_val = row['p_val']\n",
    "\n",
    "            if snp_id not in snp_dict :\n",
    "                snp_dict[snp_id] = {\n",
    "                    'effect_size' : [],\n",
    "                    'p_val' : [],\n",
    "                    'snp_id' : row['snp_id'],\n",
    "                    'gene' : row['gene'],\n",
    "                    'gene_id' : row['gene_id'],\n",
    "                    'pas' : row['pas'],\n",
    "                    'wide_seq_ext' : row['wide_seq_ext'],\n",
    "                    'wide_seq_ext_var' : row['wide_seq_ext_var'],\n",
    "                    'rel_snp_pos' : row['rel_snp_pos'],\n",
    "                    'snp_pos' : row['rel_snp_pos'] - 105,\n",
    "                    'mean_PDUI' : row['mean_PDUI'],\n",
    "                    'site_type' : row['site_type'],\n",
    "                }\n",
    "\n",
    "            snp_dict[snp_id]['effect_size'].append(effect_size)\n",
    "            snp_dict[snp_id]['p_val'].append(p_val)\n",
    "\n",
    "    extended_snp_ids = []\n",
    "    snp_ids = []\n",
    "    genes = []\n",
    "    gene_ids = []\n",
    "    pases = []\n",
    "    seqs = []\n",
    "    seq_vars = []\n",
    "    rel_snp_poses = []\n",
    "    snp_poses = []\n",
    "    effect_sizes = []\n",
    "    p_vals = []\n",
    "    n_tissues = []\n",
    "    mean_PDUIs = []\n",
    "    site_types = []\n",
    "\n",
    "    for snp_id in snp_dict :\n",
    "\n",
    "        extended_snp_ids.append(snp_id)\n",
    "        snp_ids.append(snp_dict[snp_id]['snp_id'])\n",
    "        effect_sizes.append(np.median(snp_dict[snp_id]['effect_size']))\n",
    "        p_vals.append(np.min(snp_dict[snp_id]['p_val']))\n",
    "        \n",
    "        genes.append(snp_dict[snp_id]['gene'])\n",
    "        gene_ids.append(snp_dict[snp_id]['gene_id'])\n",
    "        pases.append(snp_dict[snp_id]['pas'])\n",
    "        seqs.append(snp_dict[snp_id]['wide_seq_ext'])\n",
    "        seq_vars.append(snp_dict[snp_id]['wide_seq_ext_var'])\n",
    "        rel_snp_poses.append(snp_dict[snp_id]['rel_snp_pos'])\n",
    "        snp_poses.append(snp_dict[snp_id]['snp_pos'])\n",
    "        mean_PDUIs.append(snp_dict[snp_id]['mean_PDUI'])\n",
    "        site_types.append(snp_dict[snp_id]['site_type'])\n",
    "\n",
    "        n_tissues.append(len(snp_dict[snp_id]['effect_size']))\n",
    "\n",
    "    median_df = pd.DataFrame({\n",
    "        'extended_snp_id' : extended_snp_ids,\n",
    "        'snp_id' : snp_ids,\n",
    "        'gene' : genes,\n",
    "        'gene_id' : gene_ids,\n",
    "        'site_type' : site_types,\n",
    "        'pas' : pases,\n",
    "        'wide_seq_ext' : seqs,\n",
    "        'wide_seq_ext_var' : seq_vars,\n",
    "        'rel_snp_pos' : rel_snp_poses,\n",
    "        'snp_pos' : snp_poses,\n",
    "        'effect_size' : effect_sizes,\n",
    "        'mean_PDUI' : mean_PDUIs,\n",
    "        'p_val' : p_vals,\n",
    "        'n_tissues' : n_tissues\n",
    "    })\n",
    "    median_df = median_df[['extended_snp_id', 'snp_id', 'gene', 'gene_id', 'site_type', 'pas', 'wide_seq_ext', 'wide_seq_ext_var', 'rel_snp_pos', 'snp_pos', 'effect_size', 'mean_PDUI', 'p_val', 'n_tissues']]\n",
    "\n",
    "    median_df.to_csv(\"polyadb_median\" + version_suffix + exclude_suffix + \"_SNPs.csv\", sep='\\t')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>extended_snp_id</th>\n",
       "      <th>snp_id</th>\n",
       "      <th>gene</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>site_type</th>\n",
       "      <th>pas</th>\n",
       "      <th>wide_seq_ext</th>\n",
       "      <th>wide_seq_ext_var</th>\n",
       "      <th>rel_snp_pos</th>\n",
       "      <th>snp_pos</th>\n",
       "      <th>effect_size</th>\n",
       "      <th>mean_PDUI</th>\n",
       "      <th>p_val</th>\n",
       "      <th>n_tissues</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr10_100176869_C_T__HPS1.12__NM_000195</td>\n",
       "      <td>chr10_100176869_C_T</td>\n",
       "      <td>HPS1</td>\n",
       "      <td>HPS1.12</td>\n",
       "      <td>3_most_exon</td>\n",
       "      <td>0</td>\n",
       "      <td>ACCCAGCAAGCTCTCTCCTACCCCAGGTAAAAGGTGCTCCTTTGCC...</td>\n",
       "      <td>ACCCAGCAAGCTCTCTCCTACCCCAGGTAAAAGGTGCTCCTTTGCC...</td>\n",
       "      <td>112</td>\n",
       "      <td>7</td>\n",
       "      <td>-0.020032</td>\n",
       "      <td>0.141994</td>\n",
       "      <td>1.360477e-23</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr10_16555528_A_T__PTER.25__NM_030664</td>\n",
       "      <td>chr10_16555528_A_T</td>\n",
       "      <td>PTER</td>\n",
       "      <td>PTER.25</td>\n",
       "      <td>3_most_exon</td>\n",
       "      <td>0</td>\n",
       "      <td>TGGATAACTCTTAAAGAGAAGTCATTTTAACTGTTTTTGCTACTCC...</td>\n",
       "      <td>TGGATAACTCTTAAAGAGAAGTCATTTTAACTGTTTTTGCTACTCC...</td>\n",
       "      <td>320</td>\n",
       "      <td>215</td>\n",
       "      <td>-0.108717</td>\n",
       "      <td>0.827883</td>\n",
       "      <td>1.388327e-26</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr10_16555528_A_T__PTER.26__NM_030664</td>\n",
       "      <td>chr10_16555528_A_T</td>\n",
       "      <td>PTER</td>\n",
       "      <td>PTER.26</td>\n",
       "      <td>3_most_exon</td>\n",
       "      <td>0</td>\n",
       "      <td>ATATTGTCATTCAAAATATATTTTAACCCAAAATAAGTTAAATAAT...</td>\n",
       "      <td>ATATTGTCATTCAAAATATATTTTAACCCAAAATAAGTTAAATAAT...</td>\n",
       "      <td>272</td>\n",
       "      <td>167</td>\n",
       "      <td>-0.108717</td>\n",
       "      <td>0.827883</td>\n",
       "      <td>1.388327e-26</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr10_16555528_A_T__PTER.27__NM_030664</td>\n",
       "      <td>chr10_16555528_A_T</td>\n",
       "      <td>PTER</td>\n",
       "      <td>PTER.27</td>\n",
       "      <td>3_most_exon</td>\n",
       "      <td>2</td>\n",
       "      <td>TACACTTTTTTATATTAAAATTTTGAGGCTATACAGCCACTGTGCC...</td>\n",
       "      <td>TACACTTTTTTATATTAAAATTTTGAGGCTATACAGCCACTGTGCC...</td>\n",
       "      <td>197</td>\n",
       "      <td>92</td>\n",
       "      <td>-0.108717</td>\n",
       "      <td>0.827883</td>\n",
       "      <td>1.388327e-26</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr10_16555528_A_T__PTER.28__NM_030664</td>\n",
       "      <td>chr10_16555528_A_T</td>\n",
       "      <td>PTER</td>\n",
       "      <td>PTER.28</td>\n",
       "      <td>3_most_exon</td>\n",
       "      <td>0</td>\n",
       "      <td>TGTGTGAGTATATGTGTGCATGTTTAGCAGATATTTGTATAAAATA...</td>\n",
       "      <td>TGTGTGAGTATATGTGTGCATGTTTAGCAGATATTTGTATAAAATA...</td>\n",
       "      <td>66</td>\n",
       "      <td>-39</td>\n",
       "      <td>-0.108717</td>\n",
       "      <td>0.827883</td>\n",
       "      <td>1.388327e-26</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr10_53457124_G_A__CSTF2T.1__NM_015235</td>\n",
       "      <td>chr10_53457124_G_A</td>\n",
       "      <td>CSTF2T</td>\n",
       "      <td>CSTF2T.1</td>\n",
       "      <td>3_most_exon</td>\n",
       "      <td>2</td>\n",
       "      <td>TTCTGTTTTCCTTTTTTTTTTAATTGAGGGTGGGGGGAGGAGGGAG...</td>\n",
       "      <td>TTCTGTTTTCCTTTTTTTTTTAATTGAGGGTGGGGGGAGGAGGGAG...</td>\n",
       "      <td>185</td>\n",
       "      <td>80</td>\n",
       "      <td>-0.035068</td>\n",
       "      <td>0.342879</td>\n",
       "      <td>2.501164e-11</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr10_53457124_G_A__CSTF2T.2__NM_015235</td>\n",
       "      <td>chr10_53457124_G_A</td>\n",
       "      <td>CSTF2T</td>\n",
       "      <td>CSTF2T.2</td>\n",
       "      <td>3_most_exon</td>\n",
       "      <td>0</td>\n",
       "      <td>TATGCCAAATAAGATTACAAAGAATAAGCAGCAATATTGAAGCATC...</td>\n",
       "      <td>TATGCCAAATAAGATTACAAAGAATAAGCAGCAATATTGAAGCATC...</td>\n",
       "      <td>88</td>\n",
       "      <td>-17</td>\n",
       "      <td>-0.035068</td>\n",
       "      <td>0.342879</td>\n",
       "      <td>2.501164e-11</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr10_53457124_G_A__CSTF2T.3__NM_015235</td>\n",
       "      <td>chr10_53457124_G_A</td>\n",
       "      <td>CSTF2T</td>\n",
       "      <td>CSTF2T.3</td>\n",
       "      <td>3_most_exon</td>\n",
       "      <td>3</td>\n",
       "      <td>TATGTTAACTACATTTTTTAAATGTCGAGTAAAACTTCGTGAAAAC...</td>\n",
       "      <td>TATGTTAACTACATTTTTTAAATGTCGAGTAAAACTTTGTGAAAAC...</td>\n",
       "      <td>37</td>\n",
       "      <td>-68</td>\n",
       "      <td>-0.035068</td>\n",
       "      <td>0.342879</td>\n",
       "      <td>2.501164e-11</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr10_90044789_C_T__RNLS.6__NM_001031709</td>\n",
       "      <td>chr10_90044789_C_T</td>\n",
       "      <td>RNLS</td>\n",
       "      <td>RNLS.6</td>\n",
       "      <td>Intron</td>\n",
       "      <td>0</td>\n",
       "      <td>ATTTTTCATGTGGAGTATAAAATCAATTTTGTAATTTTGATAGTTA...</td>\n",
       "      <td>ATTTTTCATGTGGAGTATAAAATCAATTTTGTAATTTTGATAGTTA...</td>\n",
       "      <td>198</td>\n",
       "      <td>93</td>\n",
       "      <td>-0.014959</td>\n",
       "      <td>0.147291</td>\n",
       "      <td>1.492930e-04</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr10_90044789_C_T__RNLS.7__NM_001031709</td>\n",
       "      <td>chr10_90044789_C_T</td>\n",
       "      <td>RNLS</td>\n",
       "      <td>RNLS.7</td>\n",
       "      <td>Intron</td>\n",
       "      <td>0</td>\n",
       "      <td>AAATTCCTCACACCTTGCACCTTCCCTACTTTTCTGAATTGCTATG...</td>\n",
       "      <td>AAATTCCTCACACCTTGCACCTTCCCTACTTTTCTGAATTGCTATG...</td>\n",
       "      <td>134</td>\n",
       "      <td>29</td>\n",
       "      <td>-0.014959</td>\n",
       "      <td>0.147291</td>\n",
       "      <td>1.492930e-04</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>chr10_96997435_T_C__PDLIM1.8__NM_020992</td>\n",
       "      <td>chr10_96997435_T_C</td>\n",
       "      <td>PDLIM1</td>\n",
       "      <td>PDLIM1.8</td>\n",
       "      <td>3_most_exon</td>\n",
       "      <td>0</td>\n",
       "      <td>CACCAGCTGTGCGATGCCCGCTTTTACAATTAAAGGAAAACTGTTT...</td>\n",
       "      <td>CACCAGCTGTGCGATGCCCGCTTTTACAATTAAAGGAAAACTGTTT...</td>\n",
       "      <td>89</td>\n",
       "      <td>-16</td>\n",
       "      <td>-0.032086</td>\n",
       "      <td>0.783768</td>\n",
       "      <td>1.783002e-12</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>chr10_96997435_T_C__PDLIM1.7__NM_020992</td>\n",
       "      <td>chr10_96997435_T_C</td>\n",
       "      <td>PDLIM1</td>\n",
       "      <td>PDLIM1.7</td>\n",
       "      <td>3_most_exon</td>\n",
       "      <td>3</td>\n",
       "      <td>CATTGAGTCCCCTCCCTGCCTTGGTTAATTGACTCACACCAGCTGT...</td>\n",
       "      <td>CATTGAGTCCCCTCCCTGCCTTGGTTAATTGACTCACACCAGCTGT...</td>\n",
       "      <td>125</td>\n",
       "      <td>20</td>\n",
       "      <td>-0.032086</td>\n",
       "      <td>0.783768</td>\n",
       "      <td>1.783002e-12</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>chr10_96997435_T_C__PDLIM1.6__NM_020992</td>\n",
       "      <td>chr10_96997435_T_C</td>\n",
       "      <td>PDLIM1</td>\n",
       "      <td>PDLIM1.6</td>\n",
       "      <td>3_most_exon</td>\n",
       "      <td>-1</td>\n",
       "      <td>TGGCCCTCTCCTCTCTTGAAAGTTCTCTGCTTACTTTGGTTTTCCC...</td>\n",
       "      <td>TGGCCCTCTCCTCTCTTGAAAGTTCTCTGCTTACTTTGGTTTTCCC...</td>\n",
       "      <td>184</td>\n",
       "      <td>79</td>\n",
       "      <td>-0.032086</td>\n",
       "      <td>0.783768</td>\n",
       "      <td>1.783002e-12</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>chr10_96997435_T_C__PDLIM1.4__NM_020992</td>\n",
       "      <td>chr10_96997435_T_C</td>\n",
       "      <td>PDLIM1</td>\n",
       "      <td>PDLIM1.4</td>\n",
       "      <td>3_most_exon</td>\n",
       "      <td>4</td>\n",
       "      <td>TGAAGTGGTCACTGTGTTCCCCAAGTGAGCCAGCAGATCTGACCAC...</td>\n",
       "      <td>TGAAGTGGTCACTGTGTTCCCCAAGTGAGCCAGCAGATCTGACCAC...</td>\n",
       "      <td>274</td>\n",
       "      <td>169</td>\n",
       "      <td>-0.032086</td>\n",
       "      <td>0.783768</td>\n",
       "      <td>1.783002e-12</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>chr10_96997435_T_C__PDLIM1.5__NM_020992</td>\n",
       "      <td>chr10_96997435_T_C</td>\n",
       "      <td>PDLIM1</td>\n",
       "      <td>PDLIM1.5</td>\n",
       "      <td>3_most_exon</td>\n",
       "      <td>0</td>\n",
       "      <td>CTGACCACTGTTCTCCAGCAGGCCTCTGCTGCAGCTTTTTCTCTCA...</td>\n",
       "      <td>CTGACCACTGTTCTCCAGCAGGCCTCTGCTGCAGCTTTTTCTCTCA...</td>\n",
       "      <td>236</td>\n",
       "      <td>131</td>\n",
       "      <td>-0.032086</td>\n",
       "      <td>0.783768</td>\n",
       "      <td>1.783002e-12</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>chr11_1774136_T_G__CTSD.19__NM_001909</td>\n",
       "      <td>chr11_1774136_T_G</td>\n",
       "      <td>CTSD</td>\n",
       "      <td>CTSD.19</td>\n",
       "      <td>3_most_exon</td>\n",
       "      <td>-1</td>\n",
       "      <td>GGCCCTCCCTTGGAAACCTGCCCTGCCTGAGGGCCCCTCTGCCCAG...</td>\n",
       "      <td>GGCCCTCCCTTGGAAACCTGCCCTGCCTGAGGGCCCCTCTGCCCAG...</td>\n",
       "      <td>198</td>\n",
       "      <td>93</td>\n",
       "      <td>0.024994</td>\n",
       "      <td>0.895170</td>\n",
       "      <td>2.422780e-06</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>chr11_1774136_T_G__CTSD.22__NM_001909</td>\n",
       "      <td>chr11_1774136_T_G</td>\n",
       "      <td>CTSD</td>\n",
       "      <td>CTSD.22</td>\n",
       "      <td>3_most_exon</td>\n",
       "      <td>3</td>\n",
       "      <td>AGGGGACAAAACCCACCTTGTTGGAGCCTGCAGGGTGGTGCTGGGA...</td>\n",
       "      <td>AGGGGACAAAACCCACCTTGTTGGAGCCTGCAGGGTGGTGCTGGGA...</td>\n",
       "      <td>53</td>\n",
       "      <td>-52</td>\n",
       "      <td>0.024994</td>\n",
       "      <td>0.895170</td>\n",
       "      <td>2.422780e-06</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>chr11_1774136_T_G__CTSD.21__NM_001909</td>\n",
       "      <td>chr11_1774136_T_G</td>\n",
       "      <td>CTSD</td>\n",
       "      <td>CTSD.21</td>\n",
       "      <td>3_most_exon</td>\n",
       "      <td>3</td>\n",
       "      <td>AGGGGACAAAACCCACCTTGTTGGAGCCTGCAGGGTGGTGCTGGGA...</td>\n",
       "      <td>AGGGGACAAAACCCACCTTGTTGGAGCCTGCAGGGTGGTGCTGGGA...</td>\n",
       "      <td>53</td>\n",
       "      <td>-52</td>\n",
       "      <td>0.024994</td>\n",
       "      <td>0.895170</td>\n",
       "      <td>2.422780e-06</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>chr11_1774136_T_G__CTSD.20__NM_001909</td>\n",
       "      <td>chr11_1774136_T_G</td>\n",
       "      <td>CTSD</td>\n",
       "      <td>CTSD.20</td>\n",
       "      <td>3_most_exon</td>\n",
       "      <td>-1</td>\n",
       "      <td>CTGCCACCCTACCTGTTCAGTGTCCCGGGCCCGTTGAGGATGAGGC...</td>\n",
       "      <td>CTGCCACCCTACCTGTTCAGTGTCCCGGGCCCGTTGAGGATGAGGC...</td>\n",
       "      <td>134</td>\n",
       "      <td>29</td>\n",
       "      <td>0.024994</td>\n",
       "      <td>0.895170</td>\n",
       "      <td>2.422780e-06</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>chr11_1774136_T_G__CTSD.18__NM_001909</td>\n",
       "      <td>chr11_1774136_T_G</td>\n",
       "      <td>CTSD</td>\n",
       "      <td>CTSD.18</td>\n",
       "      <td>3_most_exon</td>\n",
       "      <td>4</td>\n",
       "      <td>TACCCCACACCAGGCCGTCCTCCCGGGCCCTCCCTTGGAAACCTGC...</td>\n",
       "      <td>TACCCCACACCAGGCCGTCCTCCCGGGCCCTCCCTTGGAAACCTGC...</td>\n",
       "      <td>223</td>\n",
       "      <td>118</td>\n",
       "      <td>0.024994</td>\n",
       "      <td>0.895170</td>\n",
       "      <td>2.422780e-06</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>chr11_45832935_G_A__SLC35C1.1__NM_001145266</td>\n",
       "      <td>chr11_45832935_G_A</td>\n",
       "      <td>SLC35C1</td>\n",
       "      <td>SLC35C1.1</td>\n",
       "      <td>3_most_exon</td>\n",
       "      <td>4</td>\n",
       "      <td>CACCCTGGATGGCCCGGCCCCGGGGCCCGTACACAGGCGGGGCCAG...</td>\n",
       "      <td>CACCCTGGATGGCCCGGCCCCGGGGCCCGTACACAGGCAGGGCCAG...</td>\n",
       "      <td>38</td>\n",
       "      <td>-67</td>\n",
       "      <td>0.048006</td>\n",
       "      <td>0.841029</td>\n",
       "      <td>4.738651e-10</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>chr11_57295864_A_G__TIMM10.9__NM_012456</td>\n",
       "      <td>chr11_57295864_A_G</td>\n",
       "      <td>TIMM10</td>\n",
       "      <td>TIMM10.9</td>\n",
       "      <td>3_most_exon</td>\n",
       "      <td>2</td>\n",
       "      <td>AGAGTCTTCAAGATCCCGGAGTGGTAGCGCTGTCTCCTGGTGAAGG...</td>\n",
       "      <td>AGAGTCTTCAAGATCCCGGAGTGGTAGCGCTGTCTCCTGGTGAAGG...</td>\n",
       "      <td>215</td>\n",
       "      <td>110</td>\n",
       "      <td>0.056948</td>\n",
       "      <td>0.378714</td>\n",
       "      <td>6.179427e-14</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>chr11_57295864_A_G__TIMM10.8__NM_012456</td>\n",
       "      <td>chr11_57295864_A_G</td>\n",
       "      <td>TIMM10</td>\n",
       "      <td>TIMM10.8</td>\n",
       "      <td>3_most_exon</td>\n",
       "      <td>2</td>\n",
       "      <td>ACGTGCTCCCTGTTGGGTGTCATCTGTGAAGACTGCCAGGCCTAGG...</td>\n",
       "      <td>ACGTGCTCCCTGTTGGGTGTCATCTGTGAAGACTGCCAGGCCTAGG...</td>\n",
       "      <td>271</td>\n",
       "      <td>166</td>\n",
       "      <td>0.056948</td>\n",
       "      <td>0.378714</td>\n",
       "      <td>6.179427e-14</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>chr11_612148_T_C__PHRF1.8__NM_020901</td>\n",
       "      <td>chr11_612148_T_C</td>\n",
       "      <td>PHRF1</td>\n",
       "      <td>PHRF1.8</td>\n",
       "      <td>3_most_exon</td>\n",
       "      <td>4</td>\n",
       "      <td>AGTGGCGGGAATCGGGGCCATGCCCGGGGAGCTGTCGGGAGTGGCG...</td>\n",
       "      <td>AGTGGCGGGAATCGGGGCCATGCCCGGGGAGCTGTCGGGAGTGGCG...</td>\n",
       "      <td>292</td>\n",
       "      <td>187</td>\n",
       "      <td>-0.019693</td>\n",
       "      <td>0.230171</td>\n",
       "      <td>3.420028e-05</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>chr11_612148_T_C__PHRF1.12__NM_020901</td>\n",
       "      <td>chr11_612148_T_C</td>\n",
       "      <td>PHRF1</td>\n",
       "      <td>PHRF1.12</td>\n",
       "      <td>3_most_exon</td>\n",
       "      <td>4</td>\n",
       "      <td>GATTATCTTTAGAAACCTCTTGATTGACTTACTACTTGGAAGATAA...</td>\n",
       "      <td>GATTATCTTTAGAAACCTCTTGATTGACTTACTACTTGGAAGATAA...</td>\n",
       "      <td>102</td>\n",
       "      <td>-3</td>\n",
       "      <td>-0.019693</td>\n",
       "      <td>0.230171</td>\n",
       "      <td>3.420028e-05</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>chr11_612148_T_C__PHRF1.11__NM_020901</td>\n",
       "      <td>chr11_612148_T_C</td>\n",
       "      <td>PHRF1</td>\n",
       "      <td>PHRF1.11</td>\n",
       "      <td>3_most_exon</td>\n",
       "      <td>2</td>\n",
       "      <td>CCTGTGTTGCTCACAGTTGAAAACTGGACACTTTTGTATGTATATT...</td>\n",
       "      <td>CCTGTGTTGCTCACAGTTGAAAACTGGACACTTTTGTATGTATATT...</td>\n",
       "      <td>185</td>\n",
       "      <td>80</td>\n",
       "      <td>-0.019693</td>\n",
       "      <td>0.230171</td>\n",
       "      <td>3.420028e-05</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>chr11_612148_T_C__PHRF1.10__NM_020901</td>\n",
       "      <td>chr11_612148_T_C</td>\n",
       "      <td>PHRF1</td>\n",
       "      <td>PHRF1.10</td>\n",
       "      <td>3_most_exon</td>\n",
       "      <td>2</td>\n",
       "      <td>GGCATCACCATGCCTGCCGTCGGGTTCCTGCGCTGACACCTGGTCT...</td>\n",
       "      <td>GGCATCACCATGCCTGCCGTCGGGTTCCTGCGCTGACACCTGGTCT...</td>\n",
       "      <td>236</td>\n",
       "      <td>131</td>\n",
       "      <td>-0.019693</td>\n",
       "      <td>0.230171</td>\n",
       "      <td>3.420028e-05</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>chr11_612148_T_C__PHRF1.9__NM_020901</td>\n",
       "      <td>chr11_612148_T_C</td>\n",
       "      <td>PHRF1</td>\n",
       "      <td>PHRF1.9</td>\n",
       "      <td>3_most_exon</td>\n",
       "      <td>4</td>\n",
       "      <td>CGGGGAGCTGTCGGGAGTGGCGGGAAATGGGGGGCATCACCATGCC...</td>\n",
       "      <td>CGGGGAGCTGTCGGGAGTGGCGGGAAATGGGGGGCATCACCATGCC...</td>\n",
       "      <td>268</td>\n",
       "      <td>163</td>\n",
       "      <td>-0.019693</td>\n",
       "      <td>0.230171</td>\n",
       "      <td>3.420028e-05</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>chr11_75282882_G_A__SERPINH1.4__NM_001207014</td>\n",
       "      <td>chr11_75282882_G_A</td>\n",
       "      <td>SERPINH1</td>\n",
       "      <td>SERPINH1.4</td>\n",
       "      <td>3_most_exon</td>\n",
       "      <td>-1</td>\n",
       "      <td>AGCCATGGCCTCACCTGGCACACCTCCTTGTTCCCACAGAAACACC...</td>\n",
       "      <td>AGCCATGGCCTCACCTGGCACACCTCCTTGTTCCCACAGAAACACC...</td>\n",
       "      <td>95</td>\n",
       "      <td>-10</td>\n",
       "      <td>0.036067</td>\n",
       "      <td>0.834915</td>\n",
       "      <td>1.182143e-16</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>chr11_838634_C_G__CD151.6__NM_001039490</td>\n",
       "      <td>chr11_838634_C_G</td>\n",
       "      <td>CD151</td>\n",
       "      <td>CD151.6</td>\n",
       "      <td>3_most_exon</td>\n",
       "      <td>-1</td>\n",
       "      <td>GGCCCCATCCCAGTTGGGGAAGCCAGGTGAGCTCTGACCCTTGGGC...</td>\n",
       "      <td>GGCCCCATCCCAGTTGGGGAAGCCAGGTGAGCTCTGACCCTTGGGC...</td>\n",
       "      <td>74</td>\n",
       "      <td>-31</td>\n",
       "      <td>0.035169</td>\n",
       "      <td>0.640312</td>\n",
       "      <td>5.367018e-11</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1505</th>\n",
       "      <td>chr12_132623225_C_T__DDX51.3__NM_175066</td>\n",
       "      <td>chr12_132623225_C_T</td>\n",
       "      <td>DDX51</td>\n",
       "      <td>DDX51.3</td>\n",
       "      <td>3_most_exon</td>\n",
       "      <td>3</td>\n",
       "      <td>TGCTGAGAGTTGGGGGCATTAACCTTGAGCTAATGTTAAAAATCAA...</td>\n",
       "      <td>TGCTGAGAGTTGGGGGCATTAACCTTGAGCTAATGTTAAAAATCAA...</td>\n",
       "      <td>289</td>\n",
       "      <td>184</td>\n",
       "      <td>0.052330</td>\n",
       "      <td>0.631144</td>\n",
       "      <td>5.107935e-05</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1506</th>\n",
       "      <td>chr15_77357312_G_C__TSPAN3.8__NM_198902</td>\n",
       "      <td>chr15_77357312_G_C</td>\n",
       "      <td>TSPAN3</td>\n",
       "      <td>TSPAN3.8</td>\n",
       "      <td>Intron</td>\n",
       "      <td>0</td>\n",
       "      <td>TATGTTGGCAAAAAAAATCTGGAGTATAAGAAGTTGCTGACTTGGG...</td>\n",
       "      <td>TATGTTGGCAAAAAAAATCTGGAGTATAAGAAGTTGCTGACTTGGG...</td>\n",
       "      <td>200</td>\n",
       "      <td>95</td>\n",
       "      <td>0.043844</td>\n",
       "      <td>0.120767</td>\n",
       "      <td>4.540639e-21</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1507</th>\n",
       "      <td>chr16_633706_C_T__PIGQ.4__NM_004204</td>\n",
       "      <td>chr16_633706_C_T</td>\n",
       "      <td>PIGQ</td>\n",
       "      <td>PIGQ.4</td>\n",
       "      <td>3_most_exon</td>\n",
       "      <td>3</td>\n",
       "      <td>CTTGCCAGGATGAACCCCCCAGTCCCAGGCACCCTCTAGCTCCCTC...</td>\n",
       "      <td>CTTGCCAGGATGAACCCCCCAGTCCCAGGCACCCTCTAGCTCCCTC...</td>\n",
       "      <td>247</td>\n",
       "      <td>142</td>\n",
       "      <td>-0.043637</td>\n",
       "      <td>0.872803</td>\n",
       "      <td>8.074476e-05</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1508</th>\n",
       "      <td>chr17_1801592_A_G__RPA1.11__NM_002945</td>\n",
       "      <td>chr17_1801592_A_G</td>\n",
       "      <td>RPA1</td>\n",
       "      <td>RPA1.11</td>\n",
       "      <td>3_most_exon</td>\n",
       "      <td>2</td>\n",
       "      <td>GCTATGTTACATGGAAAGGATTTTGAAGCCTTTTGTTTCCCTTGCT...</td>\n",
       "      <td>GCTATGTTACATGGAAAGGATTTTGAAGCCTTTTGTTTCCCTTGCT...</td>\n",
       "      <td>307</td>\n",
       "      <td>202</td>\n",
       "      <td>0.005495</td>\n",
       "      <td>0.022966</td>\n",
       "      <td>9.598851e-09</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1509</th>\n",
       "      <td>chr17_81052423_G_A__METRNL.2__NM_001004431</td>\n",
       "      <td>chr17_81052423_G_A</td>\n",
       "      <td>METRNL</td>\n",
       "      <td>METRNL.2</td>\n",
       "      <td>3_most_exon</td>\n",
       "      <td>4</td>\n",
       "      <td>CAGGCTGCGGTGGGCGCTGCGGTCCTGGTGGGGCCGTGCGGTGAGG...</td>\n",
       "      <td>CAGGCTGCGGTGGGCGCTGCGGTCCTGGTGGGGCCGTGCGGTGAGG...</td>\n",
       "      <td>65</td>\n",
       "      <td>-40</td>\n",
       "      <td>0.028365</td>\n",
       "      <td>0.699418</td>\n",
       "      <td>1.678454e-05</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1510</th>\n",
       "      <td>chr19_39884917_G_A__MED29.4__NM_017592</td>\n",
       "      <td>chr19_39884917_G_A</td>\n",
       "      <td>MED29</td>\n",
       "      <td>MED29.4</td>\n",
       "      <td>Intron</td>\n",
       "      <td>4</td>\n",
       "      <td>TAGACAGCACATCTGACCAAGTCTGAAACAAAGAAGAGAAGAGGGG...</td>\n",
       "      <td>TAGACAGCACATCTGACCAAGTCTGAAACAAAGAAGAGAAGAGGGG...</td>\n",
       "      <td>264</td>\n",
       "      <td>159</td>\n",
       "      <td>-0.051837</td>\n",
       "      <td>0.744850</td>\n",
       "      <td>4.547551e-07</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1511</th>\n",
       "      <td>chr1_20977221_A_T__PINK1.6__NM_032409</td>\n",
       "      <td>chr1_20977221_A_T</td>\n",
       "      <td>PINK1</td>\n",
       "      <td>PINK1.6</td>\n",
       "      <td>3_most_exon</td>\n",
       "      <td>4</td>\n",
       "      <td>CCTCTGCTCATGGAGGGCAGCCCTGTGATGTCCCTGCATGGAGCTG...</td>\n",
       "      <td>CCTCTGCTCATGGAGGGCAGCCCTGTGATGTCCCTGCATGGAGCTG...</td>\n",
       "      <td>64</td>\n",
       "      <td>-41</td>\n",
       "      <td>-0.026149</td>\n",
       "      <td>0.617702</td>\n",
       "      <td>3.229706e-05</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1512</th>\n",
       "      <td>chr1_20977221_A_T__PINK1.5__NM_032409</td>\n",
       "      <td>chr1_20977221_A_T</td>\n",
       "      <td>PINK1</td>\n",
       "      <td>PINK1.5</td>\n",
       "      <td>3_most_exon</td>\n",
       "      <td>3</td>\n",
       "      <td>CAACAATCGGCCGCCACTTTGTTGGCCAACAGGCTCACAGAGAAGT...</td>\n",
       "      <td>CAACAATCGGCCGCCACTTTGTTGGCCAACAGGCTCACAGAGAAGT...</td>\n",
       "      <td>186</td>\n",
       "      <td>81</td>\n",
       "      <td>-0.026149</td>\n",
       "      <td>0.617702</td>\n",
       "      <td>3.229706e-05</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1513</th>\n",
       "      <td>chr1_40213890_C_T__PPIE.2__NM_006112</td>\n",
       "      <td>chr1_40213890_C_T</td>\n",
       "      <td>PPIE</td>\n",
       "      <td>PPIE.2</td>\n",
       "      <td>Intron</td>\n",
       "      <td>0</td>\n",
       "      <td>ACAGGGTCTTGCTCTGACTCCCAGGCTGGAGTGCAGTGGTGAGAAC...</td>\n",
       "      <td>ACAGGGTCTTGCTCTGACTCCCAGGCTGGAGTGCAGTGGTGAGAAC...</td>\n",
       "      <td>264</td>\n",
       "      <td>159</td>\n",
       "      <td>-0.008553</td>\n",
       "      <td>0.057059</td>\n",
       "      <td>2.468077e-10</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1514</th>\n",
       "      <td>chr20_24940521_A_G__CST7.1__NM_003650</td>\n",
       "      <td>chr20_24940521_A_G</td>\n",
       "      <td>CST7</td>\n",
       "      <td>CST7.1</td>\n",
       "      <td>3_most_exon</td>\n",
       "      <td>0</td>\n",
       "      <td>CTTCAGCAAGACCACAGCCATGACAAACACCAGGATGCATGCTCCT...</td>\n",
       "      <td>CTTCAGCAAGACCACAGCCATGACAAACACCAGGATGCATGCTCCT...</td>\n",
       "      <td>163</td>\n",
       "      <td>58</td>\n",
       "      <td>-0.023371</td>\n",
       "      <td>0.570025</td>\n",
       "      <td>2.222029e-08</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1515</th>\n",
       "      <td>chr20_419067_C_T__TBC1D20.6__NM_144628</td>\n",
       "      <td>chr20_419067_C_T</td>\n",
       "      <td>TBC1D20</td>\n",
       "      <td>TBC1D20.6</td>\n",
       "      <td>Intron</td>\n",
       "      <td>0</td>\n",
       "      <td>GGGGCTGACAGTGGCACTTGGAGCGGCTGCACTGGCTGTGGTGAAA...</td>\n",
       "      <td>GGGGCTGACAGTGGCACTTGGAGCGGCTGCACTGGCTGTGGTGAAA...</td>\n",
       "      <td>256</td>\n",
       "      <td>151</td>\n",
       "      <td>-0.057123</td>\n",
       "      <td>0.801096</td>\n",
       "      <td>1.231877e-07</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1516</th>\n",
       "      <td>chr22_50830044_G_A__PPP6R2.4__NM_014678</td>\n",
       "      <td>chr22_50830044_G_A</td>\n",
       "      <td>PPP6R2</td>\n",
       "      <td>PPP6R2.4</td>\n",
       "      <td>Intron</td>\n",
       "      <td>4</td>\n",
       "      <td>TTGGAAGGCTGAGATGGGAGGACTGCTTGAGCCTGCAGTGAGCTAT...</td>\n",
       "      <td>TTGGAAGGCTGAGATGGGAGGACTGCTTGAGCCTGCAGTGAGCTAT...</td>\n",
       "      <td>110</td>\n",
       "      <td>5</td>\n",
       "      <td>-0.045340</td>\n",
       "      <td>0.497051</td>\n",
       "      <td>6.601039e-05</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1517</th>\n",
       "      <td>chr22_50830044_G_A__PPP6R2.3__NM_014678</td>\n",
       "      <td>chr22_50830044_G_A</td>\n",
       "      <td>PPP6R2</td>\n",
       "      <td>PPP6R2.3</td>\n",
       "      <td>Intron</td>\n",
       "      <td>0</td>\n",
       "      <td>TCTCAAAAAAAAAAAAAAAAAAAAAAAATGTAAACATTAGCTGGTC...</td>\n",
       "      <td>TCTCAAAAAAAAAAAAAAAAAAAAAAAATGTAAACATTAGCTGGTC...</td>\n",
       "      <td>186</td>\n",
       "      <td>81</td>\n",
       "      <td>-0.045340</td>\n",
       "      <td>0.497051</td>\n",
       "      <td>6.601039e-05</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1518</th>\n",
       "      <td>chr22_50830044_G_A__PPP6R2.3__NM_001242900</td>\n",
       "      <td>chr22_50830044_G_A</td>\n",
       "      <td>PPP6R2</td>\n",
       "      <td>PPP6R2.3</td>\n",
       "      <td>Intron</td>\n",
       "      <td>0</td>\n",
       "      <td>TCTCAAAAAAAAAAAAAAAAAAAAAAAATGTAAACATTAGCTGGTC...</td>\n",
       "      <td>TCTCAAAAAAAAAAAAAAAAAAAAAAAATGTAAACATTAGCTGGTC...</td>\n",
       "      <td>186</td>\n",
       "      <td>81</td>\n",
       "      <td>-0.045810</td>\n",
       "      <td>0.510872</td>\n",
       "      <td>1.007116e-04</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1519</th>\n",
       "      <td>chr22_50830044_G_A__PPP6R2.4__NM_001242900</td>\n",
       "      <td>chr22_50830044_G_A</td>\n",
       "      <td>PPP6R2</td>\n",
       "      <td>PPP6R2.4</td>\n",
       "      <td>Intron</td>\n",
       "      <td>4</td>\n",
       "      <td>TTGGAAGGCTGAGATGGGAGGACTGCTTGAGCCTGCAGTGAGCTAT...</td>\n",
       "      <td>TTGGAAGGCTGAGATGGGAGGACTGCTTGAGCCTGCAGTGAGCTAT...</td>\n",
       "      <td>110</td>\n",
       "      <td>5</td>\n",
       "      <td>-0.045810</td>\n",
       "      <td>0.510872</td>\n",
       "      <td>1.007116e-04</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1520</th>\n",
       "      <td>chr3_40503809_G_A__RPL14.8__NM_003973</td>\n",
       "      <td>chr3_40503809_G_A</td>\n",
       "      <td>RPL14</td>\n",
       "      <td>RPL14.8</td>\n",
       "      <td>3_most_exon</td>\n",
       "      <td>4</td>\n",
       "      <td>CCAGAAAGCCACAGGCCAGAAAGCAGCGCCTGCTCCAAAAGCTCAG...</td>\n",
       "      <td>CCAGAAAGCCACAGGCCAGAAAGCAGCGCCTGCTCCAAAAGCTCAG...</td>\n",
       "      <td>194</td>\n",
       "      <td>89</td>\n",
       "      <td>-0.006097</td>\n",
       "      <td>0.242273</td>\n",
       "      <td>1.538800e-04</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1521</th>\n",
       "      <td>chr3_40503809_G_A__RPL14.9__NM_003973</td>\n",
       "      <td>chr3_40503809_G_A</td>\n",
       "      <td>RPL14</td>\n",
       "      <td>RPL14.9</td>\n",
       "      <td>3_most_exon</td>\n",
       "      <td>0</td>\n",
       "      <td>TCAAAAAGCTCCAGCCCAGAAAGCACCTGCTCCAAAGGCATCTGGC...</td>\n",
       "      <td>TCAAAAAGCTCCAGCCCAGAAAGCACCTGCTCCAAAGGCATCTGGC...</td>\n",
       "      <td>143</td>\n",
       "      <td>38</td>\n",
       "      <td>-0.006097</td>\n",
       "      <td>0.242273</td>\n",
       "      <td>1.538800e-04</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1522</th>\n",
       "      <td>chr3_40503809_G_A__RPL14.5__NM_003973</td>\n",
       "      <td>chr3_40503809_G_A</td>\n",
       "      <td>RPL14</td>\n",
       "      <td>RPL14.5</td>\n",
       "      <td>3_most_exon</td>\n",
       "      <td>4</td>\n",
       "      <td>TCCCAAAAAAGCACCTGGTACTAAGGGTACTGCTGCTGCTGCTGCT...</td>\n",
       "      <td>TCCCAAAAAAGCACCTGGTACTAAGGGTACTGCTGCTGCTGCTGCT...</td>\n",
       "      <td>317</td>\n",
       "      <td>212</td>\n",
       "      <td>-0.006097</td>\n",
       "      <td>0.242273</td>\n",
       "      <td>1.538800e-04</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1523</th>\n",
       "      <td>chr3_40503809_G_A__RPL14.6__NM_003973</td>\n",
       "      <td>chr3_40503809_G_A</td>\n",
       "      <td>RPL14</td>\n",
       "      <td>RPL14.6</td>\n",
       "      <td>3_most_exon</td>\n",
       "      <td>4</td>\n",
       "      <td>TGGTACTAAGGGTACTGCTGCTGCTGCTGCTGCTGCTGCTGCTGCT...</td>\n",
       "      <td>TGGTACTAAGGGTACTGCTGCTGCTGCTGCTGCTGCTGCTGCTGCT...</td>\n",
       "      <td>302</td>\n",
       "      <td>197</td>\n",
       "      <td>-0.006097</td>\n",
       "      <td>0.242273</td>\n",
       "      <td>1.538800e-04</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1524</th>\n",
       "      <td>chr3_40503809_G_A__RPL14.7__NM_003973</td>\n",
       "      <td>chr3_40503809_G_A</td>\n",
       "      <td>RPL14</td>\n",
       "      <td>RPL14.7</td>\n",
       "      <td>3_most_exon</td>\n",
       "      <td>0</td>\n",
       "      <td>AAAGATCACCGCCGCGAGTAAAAAGGCTCCAGCCCAGAAGGTTCCT...</td>\n",
       "      <td>AAAGATCACCGCCGCGAGTAAAAAGGCTCCAGCCCAGAAGGTTCCT...</td>\n",
       "      <td>242</td>\n",
       "      <td>137</td>\n",
       "      <td>-0.006097</td>\n",
       "      <td>0.242273</td>\n",
       "      <td>1.538800e-04</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1525</th>\n",
       "      <td>chr5_159679636_C_T__CCNJL.10__NM_024565</td>\n",
       "      <td>chr5_159679636_C_T</td>\n",
       "      <td>CCNJL</td>\n",
       "      <td>CCNJL.10</td>\n",
       "      <td>3_most_exon</td>\n",
       "      <td>-1</td>\n",
       "      <td>TTTTCTTTCCCTGGCTCTATGCAGAGGGGCCTGAGTTGGTGTGTAT...</td>\n",
       "      <td>TTTTCTTTCCCTGGCTCTATGCAGAGGGGCCTGAGTTGGTGTGTAT...</td>\n",
       "      <td>291</td>\n",
       "      <td>186</td>\n",
       "      <td>-0.048638</td>\n",
       "      <td>0.857751</td>\n",
       "      <td>2.480168e-06</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1526</th>\n",
       "      <td>chr5_159679636_C_T__CCNJL.11__NM_024565</td>\n",
       "      <td>chr5_159679636_C_T</td>\n",
       "      <td>CCNJL</td>\n",
       "      <td>CCNJL.11</td>\n",
       "      <td>3_most_exon</td>\n",
       "      <td>-1</td>\n",
       "      <td>GCCTTGCGAGGCACAGTGGGTGGCTGTCTTGCCTTTGTTTTCGAAC...</td>\n",
       "      <td>GCCTTGCGAGGCACAGTGGGTGGCTGTCTTGCCTTTGTTTTCGAAC...</td>\n",
       "      <td>233</td>\n",
       "      <td>128</td>\n",
       "      <td>-0.048638</td>\n",
       "      <td>0.857751</td>\n",
       "      <td>2.480168e-06</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1527</th>\n",
       "      <td>chr5_169675165_G_C__LCP2.1__NM_005565</td>\n",
       "      <td>chr5_169675165_G_C</td>\n",
       "      <td>LCP2</td>\n",
       "      <td>LCP2.1</td>\n",
       "      <td>3_most_exon</td>\n",
       "      <td>0</td>\n",
       "      <td>GCATTCATTAGTTTTATACATGTAATTTGAAAATCACTAAACCTCG...</td>\n",
       "      <td>GCATTCATTAGTTTTATACATGTAATTTGAAAATCACTAAACCTCG...</td>\n",
       "      <td>119</td>\n",
       "      <td>14</td>\n",
       "      <td>0.020263</td>\n",
       "      <td>0.486987</td>\n",
       "      <td>7.841949e-07</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1528</th>\n",
       "      <td>chr6_108025416_G_A__SCML4.2__NM_198081</td>\n",
       "      <td>chr6_108025416_G_A</td>\n",
       "      <td>SCML4</td>\n",
       "      <td>SCML4.2</td>\n",
       "      <td>3_most_exon</td>\n",
       "      <td>3</td>\n",
       "      <td>TTATTGACTGACTGACTGATACTCAACTAGCAAAGGCAAGAATTTG...</td>\n",
       "      <td>TTATTGACTGACTGACTGATACTCAACTAGCAAAGGCAAGAATTTG...</td>\n",
       "      <td>190</td>\n",
       "      <td>85</td>\n",
       "      <td>-0.024215</td>\n",
       "      <td>0.210733</td>\n",
       "      <td>5.127755e-07</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1529</th>\n",
       "      <td>chr6_108025416_G_A__SCML4.3__NM_198081</td>\n",
       "      <td>chr6_108025416_G_A</td>\n",
       "      <td>SCML4</td>\n",
       "      <td>SCML4.3</td>\n",
       "      <td>3_most_exon</td>\n",
       "      <td>3</td>\n",
       "      <td>AGGTGTTGAAGTGTTGGCATGCCGAAGAATCTTAGTGACATTTAGC...</td>\n",
       "      <td>AGGTGTTGAAGTGTTGGCATGCCGAAGAATCTTAGTGACATTTAGC...</td>\n",
       "      <td>86</td>\n",
       "      <td>-19</td>\n",
       "      <td>-0.024215</td>\n",
       "      <td>0.210733</td>\n",
       "      <td>5.127755e-07</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1530</th>\n",
       "      <td>chr6_108364106_A_G__OSTM1.16__NM_014028</td>\n",
       "      <td>chr6_108364106_A_G</td>\n",
       "      <td>OSTM1</td>\n",
       "      <td>OSTM1.16</td>\n",
       "      <td>3_most_exon</td>\n",
       "      <td>3</td>\n",
       "      <td>TTAAGCGCTACCCTAAAGTGTGTAAGTTTTTCATTTTGTCATATTG...</td>\n",
       "      <td>TTAAGCGCTACCCTAAAGTGTGTAAGTTTTTCATTTTGTCATATTG...</td>\n",
       "      <td>72</td>\n",
       "      <td>-33</td>\n",
       "      <td>0.046161</td>\n",
       "      <td>0.295860</td>\n",
       "      <td>4.027238e-15</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1531</th>\n",
       "      <td>chr6_108364106_A_G__OSTM1.15__NM_014028</td>\n",
       "      <td>chr6_108364106_A_G</td>\n",
       "      <td>OSTM1</td>\n",
       "      <td>OSTM1.15</td>\n",
       "      <td>3_most_exon</td>\n",
       "      <td>0</td>\n",
       "      <td>GCTCTTTAGACCTAGCATCAGCAATAGATTTCAAAGATAAGTATTA...</td>\n",
       "      <td>GCTCTTTAGACCTAGCATCAGCAATAGATTTCAAAGATAAGTATTA...</td>\n",
       "      <td>115</td>\n",
       "      <td>10</td>\n",
       "      <td>0.046161</td>\n",
       "      <td>0.295860</td>\n",
       "      <td>4.027238e-15</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1532</th>\n",
       "      <td>chr7_151071839_T_A__NUB1.14__NM_016118</td>\n",
       "      <td>chr7_151071839_T_A</td>\n",
       "      <td>NUB1</td>\n",
       "      <td>NUB1.14</td>\n",
       "      <td>Intron</td>\n",
       "      <td>0</td>\n",
       "      <td>TGCTCAGTGCTTGGCCCAGGCCTGGCTCCAAAACGTGGTGCCTGTC...</td>\n",
       "      <td>TGCTCAGTGCTTGGCCCAGGCCTGGCTCCAAAACGTGGTGCCTGTC...</td>\n",
       "      <td>322</td>\n",
       "      <td>217</td>\n",
       "      <td>-0.047723</td>\n",
       "      <td>0.865469</td>\n",
       "      <td>1.980405e-10</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1533</th>\n",
       "      <td>chr9_116023979_A_C__SLC31A1.10__NM_001859</td>\n",
       "      <td>chr9_116023979_A_C</td>\n",
       "      <td>SLC31A1</td>\n",
       "      <td>SLC31A1.10</td>\n",
       "      <td>3_most_exon</td>\n",
       "      <td>3</td>\n",
       "      <td>GCTGATGCATCGACTTGCATTCCCACTTAACACTTTGATTAGCATG...</td>\n",
       "      <td>GCTGATGCATCGACTTGCATTCCCACTTAACACTTTGATTAGCATG...</td>\n",
       "      <td>273</td>\n",
       "      <td>168</td>\n",
       "      <td>0.062840</td>\n",
       "      <td>0.643205</td>\n",
       "      <td>4.253433e-05</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1534</th>\n",
       "      <td>chr9_132569172_A_G__TOR1A.19__NM_000113</td>\n",
       "      <td>chr9_132569172_A_G</td>\n",
       "      <td>TOR1A</td>\n",
       "      <td>TOR1A.19</td>\n",
       "      <td>3_most_exon</td>\n",
       "      <td>0</td>\n",
       "      <td>CGCGCAGGCTCCATCTGCTCCTCATGCACTGAAGGACGAGGTCAGA...</td>\n",
       "      <td>CGCGCAGGCTCCATCTGCTCCTCATGCACTGAAGGACGAGGTCAGA...</td>\n",
       "      <td>237</td>\n",
       "      <td>132</td>\n",
       "      <td>0.039304</td>\n",
       "      <td>0.642646</td>\n",
       "      <td>7.056888e-08</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1535 rows  14 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                   extended_snp_id               snp_id  \\\n",
       "0          chr10_100176869_C_T__HPS1.12__NM_000195  chr10_100176869_C_T   \n",
       "1           chr10_16555528_A_T__PTER.25__NM_030664   chr10_16555528_A_T   \n",
       "2           chr10_16555528_A_T__PTER.26__NM_030664   chr10_16555528_A_T   \n",
       "3           chr10_16555528_A_T__PTER.27__NM_030664   chr10_16555528_A_T   \n",
       "4           chr10_16555528_A_T__PTER.28__NM_030664   chr10_16555528_A_T   \n",
       "5          chr10_53457124_G_A__CSTF2T.1__NM_015235   chr10_53457124_G_A   \n",
       "6          chr10_53457124_G_A__CSTF2T.2__NM_015235   chr10_53457124_G_A   \n",
       "7          chr10_53457124_G_A__CSTF2T.3__NM_015235   chr10_53457124_G_A   \n",
       "8         chr10_90044789_C_T__RNLS.6__NM_001031709   chr10_90044789_C_T   \n",
       "9         chr10_90044789_C_T__RNLS.7__NM_001031709   chr10_90044789_C_T   \n",
       "10         chr10_96997435_T_C__PDLIM1.8__NM_020992   chr10_96997435_T_C   \n",
       "11         chr10_96997435_T_C__PDLIM1.7__NM_020992   chr10_96997435_T_C   \n",
       "12         chr10_96997435_T_C__PDLIM1.6__NM_020992   chr10_96997435_T_C   \n",
       "13         chr10_96997435_T_C__PDLIM1.4__NM_020992   chr10_96997435_T_C   \n",
       "14         chr10_96997435_T_C__PDLIM1.5__NM_020992   chr10_96997435_T_C   \n",
       "15           chr11_1774136_T_G__CTSD.19__NM_001909    chr11_1774136_T_G   \n",
       "16           chr11_1774136_T_G__CTSD.22__NM_001909    chr11_1774136_T_G   \n",
       "17           chr11_1774136_T_G__CTSD.21__NM_001909    chr11_1774136_T_G   \n",
       "18           chr11_1774136_T_G__CTSD.20__NM_001909    chr11_1774136_T_G   \n",
       "19           chr11_1774136_T_G__CTSD.18__NM_001909    chr11_1774136_T_G   \n",
       "20     chr11_45832935_G_A__SLC35C1.1__NM_001145266   chr11_45832935_G_A   \n",
       "21         chr11_57295864_A_G__TIMM10.9__NM_012456   chr11_57295864_A_G   \n",
       "22         chr11_57295864_A_G__TIMM10.8__NM_012456   chr11_57295864_A_G   \n",
       "23            chr11_612148_T_C__PHRF1.8__NM_020901     chr11_612148_T_C   \n",
       "24           chr11_612148_T_C__PHRF1.12__NM_020901     chr11_612148_T_C   \n",
       "25           chr11_612148_T_C__PHRF1.11__NM_020901     chr11_612148_T_C   \n",
       "26           chr11_612148_T_C__PHRF1.10__NM_020901     chr11_612148_T_C   \n",
       "27            chr11_612148_T_C__PHRF1.9__NM_020901     chr11_612148_T_C   \n",
       "28    chr11_75282882_G_A__SERPINH1.4__NM_001207014   chr11_75282882_G_A   \n",
       "29         chr11_838634_C_G__CD151.6__NM_001039490     chr11_838634_C_G   \n",
       "...                                            ...                  ...   \n",
       "1505       chr12_132623225_C_T__DDX51.3__NM_175066  chr12_132623225_C_T   \n",
       "1506       chr15_77357312_G_C__TSPAN3.8__NM_198902   chr15_77357312_G_C   \n",
       "1507           chr16_633706_C_T__PIGQ.4__NM_004204     chr16_633706_C_T   \n",
       "1508         chr17_1801592_A_G__RPA1.11__NM_002945    chr17_1801592_A_G   \n",
       "1509    chr17_81052423_G_A__METRNL.2__NM_001004431   chr17_81052423_G_A   \n",
       "1510        chr19_39884917_G_A__MED29.4__NM_017592   chr19_39884917_G_A   \n",
       "1511         chr1_20977221_A_T__PINK1.6__NM_032409    chr1_20977221_A_T   \n",
       "1512         chr1_20977221_A_T__PINK1.5__NM_032409    chr1_20977221_A_T   \n",
       "1513          chr1_40213890_C_T__PPIE.2__NM_006112    chr1_40213890_C_T   \n",
       "1514         chr20_24940521_A_G__CST7.1__NM_003650   chr20_24940521_A_G   \n",
       "1515        chr20_419067_C_T__TBC1D20.6__NM_144628     chr20_419067_C_T   \n",
       "1516       chr22_50830044_G_A__PPP6R2.4__NM_014678   chr22_50830044_G_A   \n",
       "1517       chr22_50830044_G_A__PPP6R2.3__NM_014678   chr22_50830044_G_A   \n",
       "1518    chr22_50830044_G_A__PPP6R2.3__NM_001242900   chr22_50830044_G_A   \n",
       "1519    chr22_50830044_G_A__PPP6R2.4__NM_001242900   chr22_50830044_G_A   \n",
       "1520         chr3_40503809_G_A__RPL14.8__NM_003973    chr3_40503809_G_A   \n",
       "1521         chr3_40503809_G_A__RPL14.9__NM_003973    chr3_40503809_G_A   \n",
       "1522         chr3_40503809_G_A__RPL14.5__NM_003973    chr3_40503809_G_A   \n",
       "1523         chr3_40503809_G_A__RPL14.6__NM_003973    chr3_40503809_G_A   \n",
       "1524         chr3_40503809_G_A__RPL14.7__NM_003973    chr3_40503809_G_A   \n",
       "1525       chr5_159679636_C_T__CCNJL.10__NM_024565   chr5_159679636_C_T   \n",
       "1526       chr5_159679636_C_T__CCNJL.11__NM_024565   chr5_159679636_C_T   \n",
       "1527         chr5_169675165_G_C__LCP2.1__NM_005565   chr5_169675165_G_C   \n",
       "1528        chr6_108025416_G_A__SCML4.2__NM_198081   chr6_108025416_G_A   \n",
       "1529        chr6_108025416_G_A__SCML4.3__NM_198081   chr6_108025416_G_A   \n",
       "1530       chr6_108364106_A_G__OSTM1.16__NM_014028   chr6_108364106_A_G   \n",
       "1531       chr6_108364106_A_G__OSTM1.15__NM_014028   chr6_108364106_A_G   \n",
       "1532        chr7_151071839_T_A__NUB1.14__NM_016118   chr7_151071839_T_A   \n",
       "1533     chr9_116023979_A_C__SLC31A1.10__NM_001859   chr9_116023979_A_C   \n",
       "1534       chr9_132569172_A_G__TOR1A.19__NM_000113   chr9_132569172_A_G   \n",
       "\n",
       "          gene     gene_id    site_type  pas  \\\n",
       "0         HPS1     HPS1.12  3_most_exon    0   \n",
       "1         PTER     PTER.25  3_most_exon    0   \n",
       "2         PTER     PTER.26  3_most_exon    0   \n",
       "3         PTER     PTER.27  3_most_exon    2   \n",
       "4         PTER     PTER.28  3_most_exon    0   \n",
       "5       CSTF2T    CSTF2T.1  3_most_exon    2   \n",
       "6       CSTF2T    CSTF2T.2  3_most_exon    0   \n",
       "7       CSTF2T    CSTF2T.3  3_most_exon    3   \n",
       "8         RNLS      RNLS.6       Intron    0   \n",
       "9         RNLS      RNLS.7       Intron    0   \n",
       "10      PDLIM1    PDLIM1.8  3_most_exon    0   \n",
       "11      PDLIM1    PDLIM1.7  3_most_exon    3   \n",
       "12      PDLIM1    PDLIM1.6  3_most_exon   -1   \n",
       "13      PDLIM1    PDLIM1.4  3_most_exon    4   \n",
       "14      PDLIM1    PDLIM1.5  3_most_exon    0   \n",
       "15        CTSD     CTSD.19  3_most_exon   -1   \n",
       "16        CTSD     CTSD.22  3_most_exon    3   \n",
       "17        CTSD     CTSD.21  3_most_exon    3   \n",
       "18        CTSD     CTSD.20  3_most_exon   -1   \n",
       "19        CTSD     CTSD.18  3_most_exon    4   \n",
       "20     SLC35C1   SLC35C1.1  3_most_exon    4   \n",
       "21      TIMM10    TIMM10.9  3_most_exon    2   \n",
       "22      TIMM10    TIMM10.8  3_most_exon    2   \n",
       "23       PHRF1     PHRF1.8  3_most_exon    4   \n",
       "24       PHRF1    PHRF1.12  3_most_exon    4   \n",
       "25       PHRF1    PHRF1.11  3_most_exon    2   \n",
       "26       PHRF1    PHRF1.10  3_most_exon    2   \n",
       "27       PHRF1     PHRF1.9  3_most_exon    4   \n",
       "28    SERPINH1  SERPINH1.4  3_most_exon   -1   \n",
       "29       CD151     CD151.6  3_most_exon   -1   \n",
       "...        ...         ...          ...  ...   \n",
       "1505     DDX51     DDX51.3  3_most_exon    3   \n",
       "1506    TSPAN3    TSPAN3.8       Intron    0   \n",
       "1507      PIGQ      PIGQ.4  3_most_exon    3   \n",
       "1508      RPA1     RPA1.11  3_most_exon    2   \n",
       "1509    METRNL    METRNL.2  3_most_exon    4   \n",
       "1510     MED29     MED29.4       Intron    4   \n",
       "1511     PINK1     PINK1.6  3_most_exon    4   \n",
       "1512     PINK1     PINK1.5  3_most_exon    3   \n",
       "1513      PPIE      PPIE.2       Intron    0   \n",
       "1514      CST7      CST7.1  3_most_exon    0   \n",
       "1515   TBC1D20   TBC1D20.6       Intron    0   \n",
       "1516    PPP6R2    PPP6R2.4       Intron    4   \n",
       "1517    PPP6R2    PPP6R2.3       Intron    0   \n",
       "1518    PPP6R2    PPP6R2.3       Intron    0   \n",
       "1519    PPP6R2    PPP6R2.4       Intron    4   \n",
       "1520     RPL14     RPL14.8  3_most_exon    4   \n",
       "1521     RPL14     RPL14.9  3_most_exon    0   \n",
       "1522     RPL14     RPL14.5  3_most_exon    4   \n",
       "1523     RPL14     RPL14.6  3_most_exon    4   \n",
       "1524     RPL14     RPL14.7  3_most_exon    0   \n",
       "1525     CCNJL    CCNJL.10  3_most_exon   -1   \n",
       "1526     CCNJL    CCNJL.11  3_most_exon   -1   \n",
       "1527      LCP2      LCP2.1  3_most_exon    0   \n",
       "1528     SCML4     SCML4.2  3_most_exon    3   \n",
       "1529     SCML4     SCML4.3  3_most_exon    3   \n",
       "1530     OSTM1    OSTM1.16  3_most_exon    3   \n",
       "1531     OSTM1    OSTM1.15  3_most_exon    0   \n",
       "1532      NUB1     NUB1.14       Intron    0   \n",
       "1533   SLC31A1  SLC31A1.10  3_most_exon    3   \n",
       "1534     TOR1A    TOR1A.19  3_most_exon    0   \n",
       "\n",
       "                                           wide_seq_ext  \\\n",
       "0     ACCCAGCAAGCTCTCTCCTACCCCAGGTAAAAGGTGCTCCTTTGCC...   \n",
       "1     TGGATAACTCTTAAAGAGAAGTCATTTTAACTGTTTTTGCTACTCC...   \n",
       "2     ATATTGTCATTCAAAATATATTTTAACCCAAAATAAGTTAAATAAT...   \n",
       "3     TACACTTTTTTATATTAAAATTTTGAGGCTATACAGCCACTGTGCC...   \n",
       "4     TGTGTGAGTATATGTGTGCATGTTTAGCAGATATTTGTATAAAATA...   \n",
       "5     TTCTGTTTTCCTTTTTTTTTTAATTGAGGGTGGGGGGAGGAGGGAG...   \n",
       "6     TATGCCAAATAAGATTACAAAGAATAAGCAGCAATATTGAAGCATC...   \n",
       "7     TATGTTAACTACATTTTTTAAATGTCGAGTAAAACTTCGTGAAAAC...   \n",
       "8     ATTTTTCATGTGGAGTATAAAATCAATTTTGTAATTTTGATAGTTA...   \n",
       "9     AAATTCCTCACACCTTGCACCTTCCCTACTTTTCTGAATTGCTATG...   \n",
       "10    CACCAGCTGTGCGATGCCCGCTTTTACAATTAAAGGAAAACTGTTT...   \n",
       "11    CATTGAGTCCCCTCCCTGCCTTGGTTAATTGACTCACACCAGCTGT...   \n",
       "12    TGGCCCTCTCCTCTCTTGAAAGTTCTCTGCTTACTTTGGTTTTCCC...   \n",
       "13    TGAAGTGGTCACTGTGTTCCCCAAGTGAGCCAGCAGATCTGACCAC...   \n",
       "14    CTGACCACTGTTCTCCAGCAGGCCTCTGCTGCAGCTTTTTCTCTCA...   \n",
       "15    GGCCCTCCCTTGGAAACCTGCCCTGCCTGAGGGCCCCTCTGCCCAG...   \n",
       "16    AGGGGACAAAACCCACCTTGTTGGAGCCTGCAGGGTGGTGCTGGGA...   \n",
       "17    AGGGGACAAAACCCACCTTGTTGGAGCCTGCAGGGTGGTGCTGGGA...   \n",
       "18    CTGCCACCCTACCTGTTCAGTGTCCCGGGCCCGTTGAGGATGAGGC...   \n",
       "19    TACCCCACACCAGGCCGTCCTCCCGGGCCCTCCCTTGGAAACCTGC...   \n",
       "20    CACCCTGGATGGCCCGGCCCCGGGGCCCGTACACAGGCGGGGCCAG...   \n",
       "21    AGAGTCTTCAAGATCCCGGAGTGGTAGCGCTGTCTCCTGGTGAAGG...   \n",
       "22    ACGTGCTCCCTGTTGGGTGTCATCTGTGAAGACTGCCAGGCCTAGG...   \n",
       "23    AGTGGCGGGAATCGGGGCCATGCCCGGGGAGCTGTCGGGAGTGGCG...   \n",
       "24    GATTATCTTTAGAAACCTCTTGATTGACTTACTACTTGGAAGATAA...   \n",
       "25    CCTGTGTTGCTCACAGTTGAAAACTGGACACTTTTGTATGTATATT...   \n",
       "26    GGCATCACCATGCCTGCCGTCGGGTTCCTGCGCTGACACCTGGTCT...   \n",
       "27    CGGGGAGCTGTCGGGAGTGGCGGGAAATGGGGGGCATCACCATGCC...   \n",
       "28    AGCCATGGCCTCACCTGGCACACCTCCTTGTTCCCACAGAAACACC...   \n",
       "29    GGCCCCATCCCAGTTGGGGAAGCCAGGTGAGCTCTGACCCTTGGGC...   \n",
       "...                                                 ...   \n",
       "1505  TGCTGAGAGTTGGGGGCATTAACCTTGAGCTAATGTTAAAAATCAA...   \n",
       "1506  TATGTTGGCAAAAAAAATCTGGAGTATAAGAAGTTGCTGACTTGGG...   \n",
       "1507  CTTGCCAGGATGAACCCCCCAGTCCCAGGCACCCTCTAGCTCCCTC...   \n",
       "1508  GCTATGTTACATGGAAAGGATTTTGAAGCCTTTTGTTTCCCTTGCT...   \n",
       "1509  CAGGCTGCGGTGGGCGCTGCGGTCCTGGTGGGGCCGTGCGGTGAGG...   \n",
       "1510  TAGACAGCACATCTGACCAAGTCTGAAACAAAGAAGAGAAGAGGGG...   \n",
       "1511  CCTCTGCTCATGGAGGGCAGCCCTGTGATGTCCCTGCATGGAGCTG...   \n",
       "1512  CAACAATCGGCCGCCACTTTGTTGGCCAACAGGCTCACAGAGAAGT...   \n",
       "1513  ACAGGGTCTTGCTCTGACTCCCAGGCTGGAGTGCAGTGGTGAGAAC...   \n",
       "1514  CTTCAGCAAGACCACAGCCATGACAAACACCAGGATGCATGCTCCT...   \n",
       "1515  GGGGCTGACAGTGGCACTTGGAGCGGCTGCACTGGCTGTGGTGAAA...   \n",
       "1516  TTGGAAGGCTGAGATGGGAGGACTGCTTGAGCCTGCAGTGAGCTAT...   \n",
       "1517  TCTCAAAAAAAAAAAAAAAAAAAAAAAATGTAAACATTAGCTGGTC...   \n",
       "1518  TCTCAAAAAAAAAAAAAAAAAAAAAAAATGTAAACATTAGCTGGTC...   \n",
       "1519  TTGGAAGGCTGAGATGGGAGGACTGCTTGAGCCTGCAGTGAGCTAT...   \n",
       "1520  CCAGAAAGCCACAGGCCAGAAAGCAGCGCCTGCTCCAAAAGCTCAG...   \n",
       "1521  TCAAAAAGCTCCAGCCCAGAAAGCACCTGCTCCAAAGGCATCTGGC...   \n",
       "1522  TCCCAAAAAAGCACCTGGTACTAAGGGTACTGCTGCTGCTGCTGCT...   \n",
       "1523  TGGTACTAAGGGTACTGCTGCTGCTGCTGCTGCTGCTGCTGCTGCT...   \n",
       "1524  AAAGATCACCGCCGCGAGTAAAAAGGCTCCAGCCCAGAAGGTTCCT...   \n",
       "1525  TTTTCTTTCCCTGGCTCTATGCAGAGGGGCCTGAGTTGGTGTGTAT...   \n",
       "1526  GCCTTGCGAGGCACAGTGGGTGGCTGTCTTGCCTTTGTTTTCGAAC...   \n",
       "1527  GCATTCATTAGTTTTATACATGTAATTTGAAAATCACTAAACCTCG...   \n",
       "1528  TTATTGACTGACTGACTGATACTCAACTAGCAAAGGCAAGAATTTG...   \n",
       "1529  AGGTGTTGAAGTGTTGGCATGCCGAAGAATCTTAGTGACATTTAGC...   \n",
       "1530  TTAAGCGCTACCCTAAAGTGTGTAAGTTTTTCATTTTGTCATATTG...   \n",
       "1531  GCTCTTTAGACCTAGCATCAGCAATAGATTTCAAAGATAAGTATTA...   \n",
       "1532  TGCTCAGTGCTTGGCCCAGGCCTGGCTCCAAAACGTGGTGCCTGTC...   \n",
       "1533  GCTGATGCATCGACTTGCATTCCCACTTAACACTTTGATTAGCATG...   \n",
       "1534  CGCGCAGGCTCCATCTGCTCCTCATGCACTGAAGGACGAGGTCAGA...   \n",
       "\n",
       "                                       wide_seq_ext_var  rel_snp_pos  snp_pos  \\\n",
       "0     ACCCAGCAAGCTCTCTCCTACCCCAGGTAAAAGGTGCTCCTTTGCC...          112        7   \n",
       "1     TGGATAACTCTTAAAGAGAAGTCATTTTAACTGTTTTTGCTACTCC...          320      215   \n",
       "2     ATATTGTCATTCAAAATATATTTTAACCCAAAATAAGTTAAATAAT...          272      167   \n",
       "3     TACACTTTTTTATATTAAAATTTTGAGGCTATACAGCCACTGTGCC...          197       92   \n",
       "4     TGTGTGAGTATATGTGTGCATGTTTAGCAGATATTTGTATAAAATA...           66      -39   \n",
       "5     TTCTGTTTTCCTTTTTTTTTTAATTGAGGGTGGGGGGAGGAGGGAG...          185       80   \n",
       "6     TATGCCAAATAAGATTACAAAGAATAAGCAGCAATATTGAAGCATC...           88      -17   \n",
       "7     TATGTTAACTACATTTTTTAAATGTCGAGTAAAACTTTGTGAAAAC...           37      -68   \n",
       "8     ATTTTTCATGTGGAGTATAAAATCAATTTTGTAATTTTGATAGTTA...          198       93   \n",
       "9     AAATTCCTCACACCTTGCACCTTCCCTACTTTTCTGAATTGCTATG...          134       29   \n",
       "10    CACCAGCTGTGCGATGCCCGCTTTTACAATTAAAGGAAAACTGTTT...           89      -16   \n",
       "11    CATTGAGTCCCCTCCCTGCCTTGGTTAATTGACTCACACCAGCTGT...          125       20   \n",
       "12    TGGCCCTCTCCTCTCTTGAAAGTTCTCTGCTTACTTTGGTTTTCCC...          184       79   \n",
       "13    TGAAGTGGTCACTGTGTTCCCCAAGTGAGCCAGCAGATCTGACCAC...          274      169   \n",
       "14    CTGACCACTGTTCTCCAGCAGGCCTCTGCTGCAGCTTTTTCTCTCA...          236      131   \n",
       "15    GGCCCTCCCTTGGAAACCTGCCCTGCCTGAGGGCCCCTCTGCCCAG...          198       93   \n",
       "16    AGGGGACAAAACCCACCTTGTTGGAGCCTGCAGGGTGGTGCTGGGA...           53      -52   \n",
       "17    AGGGGACAAAACCCACCTTGTTGGAGCCTGCAGGGTGGTGCTGGGA...           53      -52   \n",
       "18    CTGCCACCCTACCTGTTCAGTGTCCCGGGCCCGTTGAGGATGAGGC...          134       29   \n",
       "19    TACCCCACACCAGGCCGTCCTCCCGGGCCCTCCCTTGGAAACCTGC...          223      118   \n",
       "20    CACCCTGGATGGCCCGGCCCCGGGGCCCGTACACAGGCAGGGCCAG...           38      -67   \n",
       "21    AGAGTCTTCAAGATCCCGGAGTGGTAGCGCTGTCTCCTGGTGAAGG...          215      110   \n",
       "22    ACGTGCTCCCTGTTGGGTGTCATCTGTGAAGACTGCCAGGCCTAGG...          271      166   \n",
       "23    AGTGGCGGGAATCGGGGCCATGCCCGGGGAGCTGTCGGGAGTGGCG...          292      187   \n",
       "24    GATTATCTTTAGAAACCTCTTGATTGACTTACTACTTGGAAGATAA...          102       -3   \n",
       "25    CCTGTGTTGCTCACAGTTGAAAACTGGACACTTTTGTATGTATATT...          185       80   \n",
       "26    GGCATCACCATGCCTGCCGTCGGGTTCCTGCGCTGACACCTGGTCT...          236      131   \n",
       "27    CGGGGAGCTGTCGGGAGTGGCGGGAAATGGGGGGCATCACCATGCC...          268      163   \n",
       "28    AGCCATGGCCTCACCTGGCACACCTCCTTGTTCCCACAGAAACACC...           95      -10   \n",
       "29    GGCCCCATCCCAGTTGGGGAAGCCAGGTGAGCTCTGACCCTTGGGC...           74      -31   \n",
       "...                                                 ...          ...      ...   \n",
       "1505  TGCTGAGAGTTGGGGGCATTAACCTTGAGCTAATGTTAAAAATCAA...          289      184   \n",
       "1506  TATGTTGGCAAAAAAAATCTGGAGTATAAGAAGTTGCTGACTTGGG...          200       95   \n",
       "1507  CTTGCCAGGATGAACCCCCCAGTCCCAGGCACCCTCTAGCTCCCTC...          247      142   \n",
       "1508  GCTATGTTACATGGAAAGGATTTTGAAGCCTTTTGTTTCCCTTGCT...          307      202   \n",
       "1509  CAGGCTGCGGTGGGCGCTGCGGTCCTGGTGGGGCCGTGCGGTGAGG...           65      -40   \n",
       "1510  TAGACAGCACATCTGACCAAGTCTGAAACAAAGAAGAGAAGAGGGG...          264      159   \n",
       "1511  CCTCTGCTCATGGAGGGCAGCCCTGTGATGTCCCTGCATGGAGCTG...           64      -41   \n",
       "1512  CAACAATCGGCCGCCACTTTGTTGGCCAACAGGCTCACAGAGAAGT...          186       81   \n",
       "1513  ACAGGGTCTTGCTCTGACTCCCAGGCTGGAGTGCAGTGGTGAGAAC...          264      159   \n",
       "1514  CTTCAGCAAGACCACAGCCATGACAAACACCAGGATGCATGCTCCT...          163       58   \n",
       "1515  GGGGCTGACAGTGGCACTTGGAGCGGCTGCACTGGCTGTGGTGAAA...          256      151   \n",
       "1516  TTGGAAGGCTGAGATGGGAGGACTGCTTGAGCCTGCAGTGAGCTAT...          110        5   \n",
       "1517  TCTCAAAAAAAAAAAAAAAAAAAAAAAATGTAAACATTAGCTGGTC...          186       81   \n",
       "1518  TCTCAAAAAAAAAAAAAAAAAAAAAAAATGTAAACATTAGCTGGTC...          186       81   \n",
       "1519  TTGGAAGGCTGAGATGGGAGGACTGCTTGAGCCTGCAGTGAGCTAT...          110        5   \n",
       "1520  CCAGAAAGCCACAGGCCAGAAAGCAGCGCCTGCTCCAAAAGCTCAG...          194       89   \n",
       "1521  TCAAAAAGCTCCAGCCCAGAAAGCACCTGCTCCAAAGGCATCTGGC...          143       38   \n",
       "1522  TCCCAAAAAAGCACCTGGTACTAAGGGTACTGCTGCTGCTGCTGCT...          317      212   \n",
       "1523  TGGTACTAAGGGTACTGCTGCTGCTGCTGCTGCTGCTGCTGCTGCT...          302      197   \n",
       "1524  AAAGATCACCGCCGCGAGTAAAAAGGCTCCAGCCCAGAAGGTTCCT...          242      137   \n",
       "1525  TTTTCTTTCCCTGGCTCTATGCAGAGGGGCCTGAGTTGGTGTGTAT...          291      186   \n",
       "1526  GCCTTGCGAGGCACAGTGGGTGGCTGTCTTGCCTTTGTTTTCGAAC...          233      128   \n",
       "1527  GCATTCATTAGTTTTATACATGTAATTTGAAAATCACTAAACCTCG...          119       14   \n",
       "1528  TTATTGACTGACTGACTGATACTCAACTAGCAAAGGCAAGAATTTG...          190       85   \n",
       "1529  AGGTGTTGAAGTGTTGGCATGCCGAAGAATCTTAGTGACATTTAGC...           86      -19   \n",
       "1530  TTAAGCGCTACCCTAAAGTGTGTAAGTTTTTCATTTTGTCATATTG...           72      -33   \n",
       "1531  GCTCTTTAGACCTAGCATCAGCAATAGATTTCAAAGATAAGTATTA...          115       10   \n",
       "1532  TGCTCAGTGCTTGGCCCAGGCCTGGCTCCAAAACGTGGTGCCTGTC...          322      217   \n",
       "1533  GCTGATGCATCGACTTGCATTCCCACTTAACACTTTGATTAGCATG...          273      168   \n",
       "1534  CGCGCAGGCTCCATCTGCTCCTCATGCACTGAAGGACGAGGTCAGA...          237      132   \n",
       "\n",
       "      effect_size  mean_PDUI         p_val  n_tissues  \n",
       "0       -0.020032   0.141994  1.360477e-23          6  \n",
       "1       -0.108717   0.827883  1.388327e-26          8  \n",
       "2       -0.108717   0.827883  1.388327e-26          8  \n",
       "3       -0.108717   0.827883  1.388327e-26          8  \n",
       "4       -0.108717   0.827883  1.388327e-26          8  \n",
       "5       -0.035068   0.342879  2.501164e-11         10  \n",
       "6       -0.035068   0.342879  2.501164e-11         10  \n",
       "7       -0.035068   0.342879  2.501164e-11         10  \n",
       "8       -0.014959   0.147291  1.492930e-04          1  \n",
       "9       -0.014959   0.147291  1.492930e-04          1  \n",
       "10      -0.032086   0.783768  1.783002e-12         11  \n",
       "11      -0.032086   0.783768  1.783002e-12         11  \n",
       "12      -0.032086   0.783768  1.783002e-12         11  \n",
       "13      -0.032086   0.783768  1.783002e-12         11  \n",
       "14      -0.032086   0.783768  1.783002e-12         11  \n",
       "15       0.024994   0.895170  2.422780e-06          3  \n",
       "16       0.024994   0.895170  2.422780e-06          3  \n",
       "17       0.024994   0.895170  2.422780e-06          3  \n",
       "18       0.024994   0.895170  2.422780e-06          3  \n",
       "19       0.024994   0.895170  2.422780e-06          3  \n",
       "20       0.048006   0.841029  4.738651e-10          7  \n",
       "21       0.056948   0.378714  6.179427e-14          1  \n",
       "22       0.056948   0.378714  6.179427e-14          1  \n",
       "23      -0.019693   0.230171  3.420028e-05          1  \n",
       "24      -0.019693   0.230171  3.420028e-05          1  \n",
       "25      -0.019693   0.230171  3.420028e-05          1  \n",
       "26      -0.019693   0.230171  3.420028e-05          1  \n",
       "27      -0.019693   0.230171  3.420028e-05          1  \n",
       "28       0.036067   0.834915  1.182143e-16          2  \n",
       "29       0.035169   0.640312  5.367018e-11          6  \n",
       "...           ...        ...           ...        ...  \n",
       "1505     0.052330   0.631144  5.107935e-05          1  \n",
       "1506     0.043844   0.120767  4.540639e-21          1  \n",
       "1507    -0.043637   0.872803  8.074476e-05          1  \n",
       "1508     0.005495   0.022966  9.598851e-09          1  \n",
       "1509     0.028365   0.699418  1.678454e-05          1  \n",
       "1510    -0.051837   0.744850  4.547551e-07          1  \n",
       "1511    -0.026149   0.617702  3.229706e-05          1  \n",
       "1512    -0.026149   0.617702  3.229706e-05          1  \n",
       "1513    -0.008553   0.057059  2.468077e-10          1  \n",
       "1514    -0.023371   0.570025  2.222029e-08          1  \n",
       "1515    -0.057123   0.801096  1.231877e-07          1  \n",
       "1516    -0.045340   0.497051  6.601039e-05          1  \n",
       "1517    -0.045340   0.497051  6.601039e-05          1  \n",
       "1518    -0.045810   0.510872  1.007116e-04          1  \n",
       "1519    -0.045810   0.510872  1.007116e-04          1  \n",
       "1520    -0.006097   0.242273  1.538800e-04          1  \n",
       "1521    -0.006097   0.242273  1.538800e-04          1  \n",
       "1522    -0.006097   0.242273  1.538800e-04          1  \n",
       "1523    -0.006097   0.242273  1.538800e-04          1  \n",
       "1524    -0.006097   0.242273  1.538800e-04          1  \n",
       "1525    -0.048638   0.857751  2.480168e-06          1  \n",
       "1526    -0.048638   0.857751  2.480168e-06          1  \n",
       "1527     0.020263   0.486987  7.841949e-07          1  \n",
       "1528    -0.024215   0.210733  5.127755e-07          1  \n",
       "1529    -0.024215   0.210733  5.127755e-07          1  \n",
       "1530     0.046161   0.295860  4.027238e-15          1  \n",
       "1531     0.046161   0.295860  4.027238e-15          1  \n",
       "1532    -0.047723   0.865469  1.980405e-10          1  \n",
       "1533     0.062840   0.643205  4.253433e-05          1  \n",
       "1534     0.039304   0.642646  7.056888e-08          1  \n",
       "\n",
       "[1535 rows x 14 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "median_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "816"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\n",
    "len(median_df['snp_id'].unique())\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "44"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(tissue_types)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "366"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\n",
    "lead_median_df = median_df.copy()\n",
    "\n",
    "lead_median_df['cut_to_snp'] = np.abs(lead_median_df['snp_pos'] - 100)\n",
    "lead_median_df = lead_median_df.sort_values(by='cut_to_snp', ascending=True).drop_duplicates(subset=['snp_id'], keep='first')\n",
    "\n",
    "blacklist_ids = [\n",
    "    'chr5_76728085_C_T'\n",
    "]\n",
    "\n",
    "lead_median_df = lead_median_df.loc[~lead_median_df['snp_id'].isin(blacklist_ids)]\n",
    "\n",
    "len(lead_median_df.query(\"snp_pos >= 20 and snp_pos < 126 and pas != -1\")['snp_id'].unique())\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python [conda env:tensorflow]",
   "language": "python",
   "name": "conda-env-tensorflow-py"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
