{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Using TensorFlow backend.\n"
     ]
    }
   ],
   "source": [
    "from __future__ import print_function\n",
    "import keras\n",
    "from keras.models import Sequential, Model, load_model\n",
    "from keras import backend as K\n",
    "\n",
    "import tensorflow as tf\n",
    "\n",
    "import os\n",
    "import pandas as pd\n",
    "\n",
    "import numpy as np\n",
    "\n",
    "import matplotlib.pyplot as plt\n",
    "import matplotlib.cm as cm\n",
    "\n",
    "import aparent.visualization as vis\n",
    "\n",
    "from aparent.predictor import *\n",
    "\n",
    "import urllib\n",
    "import urllib.request\n",
    "import pickle\n",
    "from time import sleep\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "len(polyadb_df) = 84723\n"
     ]
    }
   ],
   "source": [
    "#Load PolyADB data\n",
    "\n",
    "df = pd.read_csv('../../data/native_data/polyadb_processed_v3.csv', delimiter=',')\n",
    "df = df.query(\"site_type == '3_most_exon'\").copy().reset_index(drop=True)\n",
    "#df = df.loc[~df['wide_seq_ext'].str.slice(175 - 70, 175 - 70 + 205).str.contains(\"AAAAAAA|AAAGAAAA|AAACAAAA|AAAAGAAA|AAAACAAA\")].copy().reset_index(drop=True)\n",
    "\n",
    "isoform_df = pd.read_csv(\"../../data/native_data/polyadb_processed_v3_utr3_isoforms_and_coords.csv\", sep='\\t')\n",
    "\n",
    "polyadb_df = df.loc[df['gene_id'].isin(isoform_df['gene_id'].values.tolist())].copy().reset_index(drop=True)\n",
    "\n",
    "print('len(polyadb_df) = ' + str(len(polyadb_df)))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "polyadb_df['start'] = polyadb_df['pas_pos'] - 150\n",
    "polyadb_df['end'] = polyadb_df['pas_pos'] + 150\n",
    "\n",
    "polyadb_bed = polyadb_df[['chrom', 'start', 'end', 'gene', 'gene_id', 'strand']].copy().reset_index(drop=True)\n",
    "\n",
    "polyadb_bed.to_csv(\"polyadb_utr3_only_coordinates.bed\", sep='\\t', header=False, index=False)\n",
    "\n",
    "isoform_bed = isoform_df[['chrom', 'start', 'end', 'gene', 'gene_id', 'strand']].copy().reset_index(drop=True)\n",
    "\n",
    "isoform_bed.to_csv(\"isoform_utr3_only_coordinates.bed\", sep='\\t', header=False, index=False)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Tissue types to compile data for\n",
    "\n",
    "tissue_types = [\n",
    "    'Adipose_Subcutaneous',\n",
    "    'Adipose_Visceral_Omentum',\n",
    "    'Adrenal_Gland',\n",
    "    'Artery_Aorta',\n",
    "    'Artery_Coronary',\n",
    "    'Artery_Tibial',\n",
    "    'Brain_Amygdala',\n",
    "    'Brain_Anterior_cingulate_cortex_BA24',\n",
    "    'Brain_Caudate_basal_ganglia',\n",
    "    'Brain_Cerebellar_Hemisphere',\n",
    "    #'Brain_Cerebellum',\n",
    "    'Brain_Cortex',\n",
    "    'Brain_Frontal_Cortex_BA9',\n",
    "    'Brain_Hippocampus',\n",
    "    'Brain_Hypothalamus',\n",
    "    'Brain_Nucleus_accumbens_basal_ganglia',\n",
    "    'Brain_Putamen_basal_ganglia',\n",
    "    'Brain_Spinal_cord_cervical_c-1',\n",
    "    'Breast_Mammary_Tissue',\n",
    "    'Cells_EBV-transformed_lymphocytes',\n",
    "    'Cells_Transformed_fibroblasts',\n",
    "    'Colon_Sigmoid',\n",
    "    'Colon_Transverse',\n",
    "    'Esophagus_Gastroesophageal_Junction',\n",
    "    'Esophagus_Mucosa',\n",
    "    'Esophagus_Muscularis',\n",
    "    #'Heart_Atrial_Appendage',\n",
    "    'Heart_Left_Ventricle',\n",
    "    'Liver',\n",
    "    'Lung',\n",
    "    'Muscle_Skeletal',\n",
    "    'Nerve_Tibial',\n",
    "    'Ovary',\n",
    "    'Pancreas',\n",
    "    'Pituitary',\n",
    "    'Prostate',\n",
    "    'Skin_Not_Sun_Exposed_Suprapubic',\n",
    "    'Skin_Sun_Exposed_Lower_leg',\n",
    "    'Small_Intestine_Terminal_Ileum',\n",
    "    'Spleen',\n",
    "    'Stomach',\n",
    "    'Testis',\n",
    "    'Thyroid',\n",
    "    'Uterus',\n",
    "    'Vagina',\n",
    "    'Whole_Blood'\n",
    "]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "version_suffix = '_utr3_only_iso_nonlead_sizematched_p_thresh'\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Processing data for tissue = 'Adipose_Subcutaneous'.\n",
      "Processing data for tissue = 'Adipose_Visceral_Omentum'.\n",
      "Processing data for tissue = 'Adrenal_Gland'.\n",
      "Processing data for tissue = 'Artery_Aorta'.\n",
      "Processing data for tissue = 'Artery_Coronary'.\n",
      "Processing data for tissue = 'Artery_Tibial'.\n",
      "Processing data for tissue = 'Brain_Amygdala'.\n",
      "Processing data for tissue = 'Brain_Anterior_cingulate_cortex_BA24'.\n",
      "Processing data for tissue = 'Brain_Caudate_basal_ganglia'.\n",
      "Processing data for tissue = 'Brain_Cerebellar_Hemisphere'.\n",
      "Processing data for tissue = 'Brain_Cortex'.\n",
      "Processing data for tissue = 'Brain_Frontal_Cortex_BA9'.\n",
      "Processing data for tissue = 'Brain_Hippocampus'.\n",
      "Processing data for tissue = 'Brain_Hypothalamus'.\n",
      "Processing data for tissue = 'Brain_Nucleus_accumbens_basal_ganglia'.\n",
      "Processing data for tissue = 'Brain_Putamen_basal_ganglia'.\n",
      "Processing data for tissue = 'Brain_Spinal_cord_cervical_c-1'.\n",
      "Processing data for tissue = 'Breast_Mammary_Tissue'.\n",
      "Processing data for tissue = 'Cells_EBV-transformed_lymphocytes'.\n",
      "Processing data for tissue = 'Cells_Transformed_fibroblasts'.\n",
      "Processing data for tissue = 'Colon_Sigmoid'.\n",
      "Processing data for tissue = 'Colon_Transverse'.\n",
      "Processing data for tissue = 'Esophagus_Gastroesophageal_Junction'.\n",
      "Processing data for tissue = 'Esophagus_Mucosa'.\n",
      "Processing data for tissue = 'Esophagus_Muscularis'.\n",
      "Processing data for tissue = 'Heart_Left_Ventricle'.\n",
      "Processing data for tissue = 'Liver'.\n",
      "Processing data for tissue = 'Lung'.\n",
      "Processing data for tissue = 'Muscle_Skeletal'.\n",
      "Processing data for tissue = 'Nerve_Tibial'.\n",
      "Processing data for tissue = 'Ovary'.\n",
      "Processing data for tissue = 'Pancreas'.\n",
      "Processing data for tissue = 'Pituitary'.\n",
      "Processing data for tissue = 'Prostate'.\n",
      "Processing data for tissue = 'Skin_Not_Sun_Exposed_Suprapubic'.\n",
      "Processing data for tissue = 'Skin_Sun_Exposed_Lower_leg'.\n",
      "Processing data for tissue = 'Small_Intestine_Terminal_Ileum'.\n",
      "Processing data for tissue = 'Spleen'.\n",
      "Processing data for tissue = 'Stomach'.\n",
      "Processing data for tissue = 'Testis'.\n",
      "Processing data for tissue = 'Thyroid'.\n",
      "Processing data for tissue = 'Uterus'.\n",
      "Processing data for tissue = 'Vagina'.\n",
      "Processing data for tissue = 'Whole_Blood'.\n"
     ]
    }
   ],
   "source": [
    "#Compile apaQTL data from GTEx\n",
    "\n",
    "p_val_thresh = 1e-10\n",
    "\n",
    "def get_snp_pos(row) :\n",
    "    if row['strand'] == '+' :\n",
    "        return row['snp_pos'] - row['pas_pos'] + 175 - 1\n",
    "    else :\n",
    "        return row['pas_pos'] - row['snp_pos'] + 175\n",
    "\n",
    "def get_snp_ref(row) :\n",
    "    if row['strand'] == '+' :\n",
    "        return row['snp_ref']\n",
    "    else :\n",
    "        if row['snp_ref'] == 'A' :\n",
    "            return 'T'\n",
    "        if row['snp_ref'] == 'C' :\n",
    "            return 'G'\n",
    "        if row['snp_ref'] == 'G' :\n",
    "            return 'C'\n",
    "        if row['snp_ref'] == 'T' :\n",
    "            return 'A'\n",
    "\n",
    "def get_snp_var(row) :\n",
    "    if row['strand'] == '+' :\n",
    "        return row['snp_var']\n",
    "    else :\n",
    "        if row['snp_var'] == 'A' :\n",
    "            return 'T'\n",
    "        if row['snp_var'] == 'C' :\n",
    "            return 'G'\n",
    "        if row['snp_var'] == 'G' :\n",
    "            return 'C'\n",
    "        if row['snp_var'] == 'T' :\n",
    "            return 'A'\n",
    "\n",
    "def get_var_seq(row) :\n",
    "    ref_seq = row['wide_seq_ext']\n",
    "    var_seq = ref_seq[:row['rel_snp_pos']] + row['rel_snp_var'] + ref_seq[row['rel_snp_pos']+1:]\n",
    "    \n",
    "    return var_seq\n",
    "\n",
    "for tissue_type in tissue_types :\n",
    "    \n",
    "    print(\"Processing data for tissue = '\" + str(tissue_type) + \"'.\")\n",
    "    \n",
    "    snp_df = pd.read_csv(tissue_type + '.cis.3aQTL.txt', sep='\\t')\n",
    "\n",
    "    snp_df['chr'] = snp_df['SNP'].apply(lambda x: x.split(\"_\")[0])\n",
    "    snp_df['start'] = snp_df['SNP'].apply(lambda x: x.split(\"_\")[1])\n",
    "    snp_df['end'] = snp_df['SNP'].apply(lambda x: str(int(x.split(\"_\")[1]) + 1))\n",
    "    snp_df['gene'] = snp_df['transcript'].apply(lambda x: x.split(\"|\")[1])\n",
    "\n",
    "    snp_df = snp_df[['chr', 'start', 'end', 'gene', 'SNP', 'transcript', 'beta', 't.stat', 'p.value']]\n",
    "\n",
    "    snp_df.to_csv(tissue_type + version_suffix + \".cis.3aQTL.coordinates.bed\", sep='\\t', header=False, index=False)\n",
    "\n",
    "    !bedtools intersect -a polyadb_utr3_only_coordinates.bed -b {tissue_type + version_suffix}.cis.3aQTL.coordinates.bed -wa -wb > polyadb_{tissue_type + version_suffix}_intersect.bed\n",
    "    \n",
    "    snp_bed_hg19 = pd.read_csv(\"polyadb_\" + tissue_type + version_suffix + \"_intersect.bed\", sep='\\t', error_bad_lines=False, names=['chrom', 'start', 'end', 'gene', 'gene_id', 'strand', 'chr_2', 'snp_pos', 'end_2', 'gene_2', 'snp_id', 'transcript_id', 'effect_size', 'test_statistic', 'p_val'])\n",
    "    snp_bed_hg19 = snp_bed_hg19.query(\"gene == gene_2\").copy().reset_index(drop=True)\n",
    "\n",
    "    snp_bed_hg19 = snp_bed_hg19[['chrom', 'start', 'end', 'gene', 'gene_id', 'strand', 'chr_2', 'snp_pos', 'end_2', 'gene_2', 'snp_id', 'transcript_id', 'effect_size', 'test_statistic', 'p_val']]\n",
    "\n",
    "    snp_bed_hg19['snp_ref'] = snp_bed_hg19['snp_id'].apply(lambda s: s.split(\"_\")[2])\n",
    "    snp_bed_hg19['snp_var'] = snp_bed_hg19['snp_id'].apply(lambda s: s.split(\"_\")[3])\n",
    "    \n",
    "    polyadb_df_sel = polyadb_df[['wide_seq_ext', 'gene_id', 'pas', 'pas_pos', 'cut_mode', 'site_type', 'rpm']].copy().reset_index(drop=True)\n",
    "\n",
    "    snp_df = snp_bed_hg19.join(polyadb_df_sel.set_index('gene_id'), on='gene_id', how='inner').copy().reset_index(drop=True)\n",
    "    \n",
    "    snp_df['rel_snp_pos'] = snp_df.apply(get_snp_pos, axis=1)\n",
    "    snp_df['rel_snp_ref'] = snp_df.apply(get_snp_ref, axis=1)\n",
    "    snp_df['rel_snp_var'] = snp_df.apply(get_snp_var, axis=1)\n",
    "\n",
    "    snp_df['wide_seq_ext_var'] = snp_df.apply(get_var_seq, axis=1)\n",
    "    \n",
    "    #Filter by lead SNPs (remove lead SNPs)\n",
    "    lead_df = pd.read_csv(\"lead/\" + tissue_type + \".lead.3aQTL.txt\", sep='\\t')\n",
    "    lead_df['is_lead'] = True\n",
    "    \n",
    "    snp_df = snp_df.join(lead_df[['SNP', 'is_lead']].set_index(\"SNP\"), on='snp_id', how='left')\n",
    "    snp_df = snp_df.loc[snp_df['is_lead'].isnull()].copy()\n",
    "    \n",
    "    #Filter on p-value\n",
    "    snp_df = snp_df.query(\"p_val <= \" + str(p_val_thresh)).copy()\n",
    "    \n",
    "    #Calculate mean PDUIs\n",
    "    pdui_df = pd.read_csv(tissue_type + \"_combined_All_PDUIs_clean.txt\", sep='\\t')\n",
    "    pdui_df = pdui_df.join(snp_df[['transcript_id']].set_index('transcript_id'), on='event_id', how='inner').copy().reset_index(drop=True)\n",
    "\n",
    "    gtex_samples = pdui_df.columns.values.tolist()[2:]\n",
    "\n",
    "    def calc_mean_PDUI(row) :\n",
    "\n",
    "        n_samples = 0.\n",
    "        PDUI = 0.\n",
    "\n",
    "        for gtex_sample in gtex_samples :\n",
    "            if not np.isnan(row[gtex_sample]) :\n",
    "\n",
    "                PDUI += row[gtex_sample]\n",
    "                n_samples += 1.\n",
    "\n",
    "        return PDUI / n_samples\n",
    "\n",
    "    pdui_df['mean_PDUI'] = pdui_df.apply(calc_mean_PDUI, axis=1)\n",
    "\n",
    "    snp_df = snp_df.join(pdui_df[['event_id', 'mean_PDUI']].set_index('event_id'), on='transcript_id', how='inner').copy().reset_index(drop=True)\n",
    "\n",
    "    snp_df['transcript_id'] = snp_df['transcript_id'].apply(lambda x: x.split(\"|\")[0])\n",
    "    snp_df['snp_transcript_id'] = snp_df['snp_id'] + \"_\" + snp_df['transcript_id']\n",
    "    \n",
    "    transcript_end_dict = { row['transcript_id'] : -1 for _, row in snp_df.iterrows() }\n",
    "\n",
    "    with open(\"/home/jlinder2/Downloads/GRCh37_latest_genomic.gtf\", 'rt') as f :\n",
    "        for line_raw in f.readlines() :\n",
    "\n",
    "            a = line_raw.strip()\n",
    "\n",
    "            if len(a.split(\"\\t\")) > 8 :\n",
    "                start = a.strip().split(\"\\t\")[3]\n",
    "                end = a.strip().split(\"\\t\")[4]\n",
    "\n",
    "                strand = a.strip().split(\"\\t\")[6]\n",
    "\n",
    "                t_id = \"\"\n",
    "                if strand in ['+', '-'] :\n",
    "                    t_id = a.strip().split(\"\\t\")[8].split(\";\")[1].split(\"\\\"\")[1].split(\".\")[0]\n",
    "\n",
    "                if t_id in transcript_end_dict :\n",
    "\n",
    "                    if transcript_end_dict[t_id] == -1 :\n",
    "                        if strand == '+' :\n",
    "                            transcript_end_dict[t_id] = int(end)\n",
    "                        else :\n",
    "                            transcript_end_dict[t_id] = int(start)\n",
    "                    else :\n",
    "                        if strand == '+' :\n",
    "                            transcript_end_dict[t_id] = max(transcript_end_dict[t_id], int(end))\n",
    "                        else :\n",
    "                            transcript_end_dict[t_id] = min(transcript_end_dict[t_id], int(start))\n",
    "\n",
    "    qtl_cut_poses = []\n",
    "    for _, row in snp_df.iterrows() :\n",
    "\n",
    "        qtl_cut_poses.append(transcript_end_dict[row['transcript_id']])\n",
    "\n",
    "    snp_df['qtl_cut_pos'] = qtl_cut_poses\n",
    "    snp_df['snp_distance'] = np.abs(snp_df['qtl_cut_pos'] - snp_df['snp_pos'])\n",
    "\n",
    "    #Drop duplicates\n",
    "    snp_df['row_id'] = snp_df['gene_id'] + \"_\" + snp_df['snp_id'] + \"_\" + snp_df['transcript_id']\n",
    "    snp_df = snp_df.sort_values(by='snp_id').drop_duplicates(subset=['row_id'], keep='first')\n",
    "    \n",
    "    #Load processed lead aQTL dataframe (for sizematching)\n",
    "    lead_snp_df = pd.read_csv(\"polyadb_\" + tissue_type + \"_utr3_only_iso_lead_SNPs.csv\", sep='\\t')\n",
    "    \n",
    "    #Sort dataframe in ascending order of absolute effect size\n",
    "    snp_df['abs_effect_size'] = np.abs(snp_df['effect_size'])\n",
    "    snp_df = snp_df.sort_values(by='abs_effect_size', ascending=True).iloc[:2*len(lead_snp_df)].copy().reset_index(drop=True)\n",
    "\n",
    "    #Save final apaQTL dataframe\n",
    "    snp_df.to_csv(\"polyadb_\" + tissue_type + version_suffix + \"_SNPs.csv\", sep='\\t')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Processing data for tissue = 'Adipose_Subcutaneous'.\n",
      "Processing data for tissue = 'Adipose_Visceral_Omentum'.\n",
      "Processing data for tissue = 'Adrenal_Gland'.\n",
      "Processing data for tissue = 'Artery_Aorta'.\n",
      "Processing data for tissue = 'Artery_Coronary'.\n",
      "Processing data for tissue = 'Artery_Tibial'.\n",
      "Processing data for tissue = 'Brain_Amygdala'.\n",
      "Processing data for tissue = 'Brain_Anterior_cingulate_cortex_BA24'.\n",
      "Processing data for tissue = 'Brain_Caudate_basal_ganglia'.\n",
      "Processing data for tissue = 'Brain_Cerebellar_Hemisphere'.\n",
      "Processing data for tissue = 'Brain_Cortex'.\n",
      "Processing data for tissue = 'Brain_Frontal_Cortex_BA9'.\n",
      "Processing data for tissue = 'Brain_Hippocampus'.\n",
      "Processing data for tissue = 'Brain_Hypothalamus'.\n",
      "Processing data for tissue = 'Brain_Nucleus_accumbens_basal_ganglia'.\n",
      "Processing data for tissue = 'Brain_Putamen_basal_ganglia'.\n",
      "Processing data for tissue = 'Brain_Spinal_cord_cervical_c-1'.\n",
      "Processing data for tissue = 'Breast_Mammary_Tissue'.\n",
      "Processing data for tissue = 'Cells_EBV-transformed_lymphocytes'.\n",
      "Processing data for tissue = 'Cells_Transformed_fibroblasts'.\n",
      "Processing data for tissue = 'Colon_Sigmoid'.\n",
      "Processing data for tissue = 'Colon_Transverse'.\n",
      "Processing data for tissue = 'Esophagus_Gastroesophageal_Junction'.\n",
      "Processing data for tissue = 'Esophagus_Mucosa'.\n",
      "Processing data for tissue = 'Esophagus_Muscularis'.\n",
      "Processing data for tissue = 'Heart_Left_Ventricle'.\n",
      "Processing data for tissue = 'Liver'.\n",
      "Processing data for tissue = 'Lung'.\n",
      "Processing data for tissue = 'Muscle_Skeletal'.\n",
      "Processing data for tissue = 'Nerve_Tibial'.\n",
      "Processing data for tissue = 'Ovary'.\n",
      "Processing data for tissue = 'Pancreas'.\n",
      "Processing data for tissue = 'Pituitary'.\n",
      "Processing data for tissue = 'Prostate'.\n",
      "Processing data for tissue = 'Skin_Not_Sun_Exposed_Suprapubic'.\n",
      "Processing data for tissue = 'Skin_Sun_Exposed_Lower_leg'.\n",
      "Processing data for tissue = 'Small_Intestine_Terminal_Ileum'.\n",
      "Processing data for tissue = 'Spleen'.\n",
      "Processing data for tissue = 'Stomach'.\n",
      "Processing data for tissue = 'Testis'.\n",
      "Processing data for tissue = 'Thyroid'.\n",
      "Processing data for tissue = 'Uterus'.\n",
      "Processing data for tissue = 'Vagina'.\n",
      "Processing data for tissue = 'Whole_Blood'.\n"
     ]
    }
   ],
   "source": [
    "#Compile apaQTL data from GTEx (UTR isoforms)\n",
    "\n",
    "p_val_thresh = 1e-10\n",
    "\n",
    "def get_snp_pos(row) :\n",
    "    if row['strand'] == '+' :\n",
    "        return row['snp_pos'] - row['start'] - 1\n",
    "    else :\n",
    "        return row['end'] - row['snp_pos']\n",
    "\n",
    "def get_snp_ref(row) :\n",
    "    if row['strand'] == '+' :\n",
    "        return row['snp_ref']\n",
    "    else :\n",
    "        if row['snp_ref'] == 'A' :\n",
    "            return 'T'\n",
    "        if row['snp_ref'] == 'C' :\n",
    "            return 'G'\n",
    "        if row['snp_ref'] == 'G' :\n",
    "            return 'C'\n",
    "        if row['snp_ref'] == 'T' :\n",
    "            return 'A'\n",
    "\n",
    "def get_snp_var(row) :\n",
    "    if row['strand'] == '+' :\n",
    "        return row['snp_var']\n",
    "    else :\n",
    "        if row['snp_var'] == 'A' :\n",
    "            return 'T'\n",
    "        if row['snp_var'] == 'C' :\n",
    "            return 'G'\n",
    "        if row['snp_var'] == 'G' :\n",
    "            return 'C'\n",
    "        if row['snp_var'] == 'T' :\n",
    "            return 'A'\n",
    "\n",
    "def get_var_seq(row) :\n",
    "    ref_seq = row['seq']\n",
    "    \n",
    "    var_seq = ref_seq[:row['rel_snp_pos']] + row['rel_snp_var'] + ref_seq[row['rel_snp_pos']+1:]\n",
    "    \n",
    "    return var_seq\n",
    "\n",
    "for tissue_type in tissue_types :\n",
    "    \n",
    "    print(\"Processing data for tissue = '\" + str(tissue_type) + \"'.\")\n",
    "    \n",
    "    snp_df = pd.read_csv(tissue_type + '.cis.3aQTL.txt', sep='\\t')\n",
    "\n",
    "    snp_df['chr'] = snp_df['SNP'].apply(lambda x: x.split(\"_\")[0])\n",
    "    snp_df['start'] = snp_df['SNP'].apply(lambda x: x.split(\"_\")[1])\n",
    "    snp_df['end'] = snp_df['SNP'].apply(lambda x: str(int(x.split(\"_\")[1]) + 1))\n",
    "    snp_df['gene'] = snp_df['transcript'].apply(lambda x: x.split(\"|\")[1])\n",
    "\n",
    "    snp_df = snp_df[['chr', 'start', 'end', 'gene', 'SNP', 'transcript', 'beta', 't.stat', 'p.value']]\n",
    "\n",
    "    snp_df.to_csv(tissue_type + version_suffix + \".cis.3aQTL.coordinates.bed\", sep='\\t', header=False, index=False)\n",
    "\n",
    "    !bedtools intersect -a isoform_utr3_only_coordinates.bed -b {tissue_type + version_suffix}.cis.3aQTL.coordinates.bed -wa -wb > isoform_{tissue_type + version_suffix}_intersect.bed\n",
    "    \n",
    "    snp_bed_hg19 = pd.read_csv(\"isoform_\" + tissue_type + version_suffix + \"_intersect.bed\", sep='\\t', error_bad_lines=False, names=['chrom', 'start', 'end', 'gene', 'gene_id', 'strand', 'chr_2', 'snp_pos', 'end_2', 'gene_2', 'snp_id', 'transcript_id', 'effect_size', 'test_statistic', 'p_val'])\n",
    "    snp_bed_hg19 = snp_bed_hg19.query(\"gene == gene_2\").copy().reset_index(drop=True)\n",
    "\n",
    "    snp_bed_hg19 = snp_bed_hg19[['chrom', 'start', 'end', 'gene', 'gene_id', 'strand', 'chr_2', 'snp_pos', 'end_2', 'gene_2', 'snp_id', 'transcript_id', 'effect_size', 'test_statistic', 'p_val']]\n",
    "\n",
    "    snp_bed_hg19['snp_ref'] = snp_bed_hg19['snp_id'].apply(lambda s: s.split(\"_\")[2])\n",
    "    snp_bed_hg19['snp_var'] = snp_bed_hg19['snp_id'].apply(lambda s: s.split(\"_\")[3])\n",
    "    \n",
    "    isoform_df_sel = isoform_df[['seq', 'gene_id']].copy().reset_index(drop=True)\n",
    "\n",
    "    snp_df = snp_bed_hg19.join(isoform_df_sel.set_index('gene_id'), on='gene_id', how='inner').copy().reset_index(drop=True)\n",
    "    \n",
    "    snp_df['rel_snp_pos'] = snp_df.apply(get_snp_pos, axis=1)\n",
    "    snp_df['rel_snp_ref'] = snp_df.apply(get_snp_ref, axis=1)\n",
    "    snp_df['rel_snp_var'] = snp_df.apply(get_snp_var, axis=1)\n",
    "\n",
    "    snp_df['seq_var'] = snp_df.apply(get_var_seq, axis=1)\n",
    "    \n",
    "    #Filter by lead SNPs (remove lead SNPs)\n",
    "    lead_df = pd.read_csv(\"lead/\" + tissue_type + \".lead.3aQTL.txt\", sep='\\t')\n",
    "    lead_df['is_lead'] = True\n",
    "    \n",
    "    snp_df = snp_df.join(lead_df[['SNP', 'is_lead']].set_index(\"SNP\"), on='snp_id', how='left')\n",
    "    snp_df = snp_df.loc[snp_df['is_lead'].isnull()].copy()\n",
    "    \n",
    "    #Filter on p-value\n",
    "    snp_df = snp_df.query(\"p_val <= \" + str(p_val_thresh)).copy()\n",
    "    \n",
    "    #Calculate mean PDUIs\n",
    "    pdui_df = pd.read_csv(tissue_type + \"_combined_All_PDUIs_clean.txt\", sep='\\t')\n",
    "    pdui_df = pdui_df.join(snp_df[['transcript_id']].set_index('transcript_id'), on='event_id', how='inner').copy().reset_index(drop=True)\n",
    "\n",
    "    gtex_samples = pdui_df.columns.values.tolist()[2:]\n",
    "\n",
    "    def calc_mean_PDUI(row) :\n",
    "\n",
    "        n_samples = 0.\n",
    "        PDUI = 0.\n",
    "\n",
    "        for gtex_sample in gtex_samples :\n",
    "            if not np.isnan(row[gtex_sample]) :\n",
    "\n",
    "                PDUI += row[gtex_sample]\n",
    "                n_samples += 1.\n",
    "\n",
    "        return PDUI / n_samples\n",
    "\n",
    "    pdui_df['mean_PDUI'] = pdui_df.apply(calc_mean_PDUI, axis=1)\n",
    "\n",
    "    snp_df = snp_df.join(pdui_df[['event_id', 'mean_PDUI']].set_index('event_id'), on='transcript_id', how='inner').copy().reset_index(drop=True)\n",
    "\n",
    "    snp_df['transcript_id'] = snp_df['transcript_id'].apply(lambda x: x.split(\"|\")[0])\n",
    "    snp_df['snp_transcript_id'] = snp_df['snp_id'] + \"_\" + snp_df['transcript_id']\n",
    "    \n",
    "    transcript_end_dict = { row['transcript_id'] : -1 for _, row in snp_df.iterrows() }\n",
    "\n",
    "    with open(\"/home/jlinder2/Downloads/GRCh37_latest_genomic.gtf\", 'rt') as f :\n",
    "        for line_raw in f.readlines() :\n",
    "\n",
    "            a = line_raw.strip()\n",
    "\n",
    "            if len(a.split(\"\\t\")) > 8 :\n",
    "                start = a.strip().split(\"\\t\")[3]\n",
    "                end = a.strip().split(\"\\t\")[4]\n",
    "\n",
    "                strand = a.strip().split(\"\\t\")[6]\n",
    "\n",
    "                t_id = \"\"\n",
    "                if strand in ['+', '-'] :\n",
    "                    t_id = a.strip().split(\"\\t\")[8].split(\";\")[1].split(\"\\\"\")[1].split(\".\")[0]\n",
    "\n",
    "                if t_id in transcript_end_dict :\n",
    "\n",
    "                    if transcript_end_dict[t_id] == -1 :\n",
    "                        if strand == '+' :\n",
    "                            transcript_end_dict[t_id] = int(end)\n",
    "                        else :\n",
    "                            transcript_end_dict[t_id] = int(start)\n",
    "                    else :\n",
    "                        if strand == '+' :\n",
    "                            transcript_end_dict[t_id] = max(transcript_end_dict[t_id], int(end))\n",
    "                        else :\n",
    "                            transcript_end_dict[t_id] = min(transcript_end_dict[t_id], int(start))\n",
    "\n",
    "    qtl_cut_poses = []\n",
    "    for _, row in snp_df.iterrows() :\n",
    "\n",
    "        qtl_cut_poses.append(transcript_end_dict[row['transcript_id']])\n",
    "\n",
    "    snp_df['qtl_cut_pos'] = qtl_cut_poses\n",
    "    snp_df['snp_distance'] = np.abs(snp_df['qtl_cut_pos'] - snp_df['snp_pos'])\n",
    "\n",
    "    #Drop duplicates\n",
    "    snp_df['row_id'] = snp_df['gene_id'] + \"_\" + snp_df['snp_id'] + \"_\" + snp_df['transcript_id']\n",
    "    snp_df = snp_df.sort_values(by='snp_id').drop_duplicates(subset=['row_id'], keep='first')\n",
    "    \n",
    "    #Load processed lead aQTL dataframe (for sizematching)\n",
    "    lead_snp_df = pd.read_csv(\"isoform_\" + tissue_type + \"_utr3_only_iso_lead_SNPs.csv\", sep='\\t')\n",
    "    \n",
    "    #Sort dataframe in ascending order of absolute effect size\n",
    "    snp_df['abs_effect_size'] = np.abs(snp_df['effect_size'])\n",
    "    snp_df = snp_df.sort_values(by='abs_effect_size', ascending=True).iloc[:2*len(lead_snp_df)].copy().reset_index(drop=True)\n",
    "\n",
    "    #Save final apaQTL dataframe\n",
    "    snp_df.to_csv(\"isoform_\" + tissue_type + version_suffix + \"_SNPs.csv\", sep='\\t')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Processing tissue = 'Adipose_Subcutaneous'\n",
      "Processing tissue = 'Adipose_Visceral_Omentum'\n",
      "Processing tissue = 'Adrenal_Gland'\n",
      "Processing tissue = 'Artery_Aorta'\n",
      "Processing tissue = 'Artery_Coronary'\n",
      "Processing tissue = 'Artery_Tibial'\n",
      "Processing tissue = 'Brain_Amygdala'\n",
      "Processing tissue = 'Brain_Anterior_cingulate_cortex_BA24'\n",
      "Processing tissue = 'Brain_Caudate_basal_ganglia'\n",
      "Processing tissue = 'Brain_Cerebellar_Hemisphere'\n",
      "Processing tissue = 'Brain_Cortex'\n",
      "Processing tissue = 'Brain_Frontal_Cortex_BA9'\n",
      "Processing tissue = 'Brain_Hippocampus'\n",
      "Processing tissue = 'Brain_Hypothalamus'\n",
      "Processing tissue = 'Brain_Nucleus_accumbens_basal_ganglia'\n",
      "Processing tissue = 'Brain_Putamen_basal_ganglia'\n",
      "Processing tissue = 'Brain_Spinal_cord_cervical_c-1'\n",
      "Processing tissue = 'Breast_Mammary_Tissue'\n",
      "Processing tissue = 'Cells_EBV-transformed_lymphocytes'\n",
      "Processing tissue = 'Cells_Transformed_fibroblasts'\n",
      "Processing tissue = 'Colon_Sigmoid'\n",
      "Processing tissue = 'Colon_Transverse'\n",
      "Processing tissue = 'Esophagus_Gastroesophageal_Junction'\n",
      "Processing tissue = 'Esophagus_Mucosa'\n",
      "Processing tissue = 'Esophagus_Muscularis'\n",
      "Processing tissue = 'Heart_Left_Ventricle'\n",
      "Processing tissue = 'Liver'\n",
      "Processing tissue = 'Lung'\n",
      "Processing tissue = 'Muscle_Skeletal'\n",
      "Processing tissue = 'Nerve_Tibial'\n",
      "Processing tissue = 'Ovary'\n",
      "Processing tissue = 'Pancreas'\n",
      "Processing tissue = 'Pituitary'\n",
      "Processing tissue = 'Prostate'\n",
      "Processing tissue = 'Skin_Not_Sun_Exposed_Suprapubic'\n",
      "Processing tissue = 'Skin_Sun_Exposed_Lower_leg'\n",
      "Processing tissue = 'Small_Intestine_Terminal_Ileum'\n",
      "Processing tissue = 'Spleen'\n",
      "Processing tissue = 'Stomach'\n",
      "Processing tissue = 'Testis'\n",
      "Processing tissue = 'Thyroid'\n",
      "Processing tissue = 'Uterus'\n",
      "Processing tissue = 'Vagina'\n",
      "Processing tissue = 'Whole_Blood'\n"
     ]
    }
   ],
   "source": [
    "#Merge tissue dataframes\n",
    "\n",
    "snp_dfs = []\n",
    "\n",
    "for tissue_type in tissue_types :\n",
    "    \n",
    "    print(\"Processing tissue = '\" + tissue_type + \"'\")\n",
    "    \n",
    "    snp_df = pd.read_csv(\"polyadb_\" + tissue_type + version_suffix + \"_SNPs.csv\", sep='\\t')\n",
    "    snp_df['tissue_type'] = tissue_type\n",
    "    \n",
    "    snp_dfs.append(snp_df)\n",
    "\n",
    "snp_df = pd.concat(snp_dfs).copy().reset_index(drop=True)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "merged_df = snp_df\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "snp_dict = {}\n",
    "\n",
    "for i, tissue_type in enumerate(tissue_types) :\n",
    "    \n",
    "    snp_df = pd.read_csv(\"polyadb_\" + tissue_type + version_suffix + \"_SNPs.csv\", sep='\\t')\n",
    "    \n",
    "    for _, row in snp_df.iterrows() :\n",
    "        \n",
    "        snp_id = row['snp_id'] + \"__\" + row['gene_id'] + \"__\" + row['transcript_id']\n",
    "        \n",
    "        effect_size = row['effect_size']\n",
    "        \n",
    "        if snp_id not in snp_dict :\n",
    "            snp_dict[snp_id] = {\n",
    "                'effect_size' : []\n",
    "            }\n",
    "        \n",
    "        snp_dict[snp_id]['effect_size'].append(effect_size)\n",
    "\n",
    "snp_ids = []\n",
    "effect_sizes = []\n",
    "n_tissues = []\n",
    "\n",
    "for snp_id in snp_dict :\n",
    "    \n",
    "    snp_ids.append(snp_id)\n",
    "    effect_sizes.append(np.median(snp_dict[snp_id]['effect_size']))\n",
    "    \n",
    "    n_tissues.append(len(snp_dict[snp_id]['effect_size']))\n",
    "\n",
    "median_df = pd.DataFrame({\n",
    "    'extended_snp_id' : snp_ids,\n",
    "    'effect_size' : effect_sizes,\n",
    "    'n_tissues' : n_tissues\n",
    "})\n",
    "median_df = median_df[['extended_snp_id', 'effect_size', 'n_tissues']]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "merged_df['extended_snp_id'] = merged_df['snp_id'] + \"__\" + merged_df['gene_id'] + \"__\" + merged_df['transcript_id']\n",
    "\n",
    "merged_df = merged_df.join(median_df.set_index('extended_snp_id'), on=\"extended_snp_id\", how='inner', rsuffix='_median').copy().reset_index(drop=True)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "merged_df.to_csv(\"polyadb_merged\" + version_suffix + \"_SNPs.csv\", sep='\\t')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Processing tissue = 'Adipose_Subcutaneous'\n",
      "Processing tissue = 'Adipose_Visceral_Omentum'\n",
      "Processing tissue = 'Adrenal_Gland'\n",
      "Processing tissue = 'Artery_Aorta'\n",
      "Processing tissue = 'Artery_Coronary'\n",
      "Processing tissue = 'Artery_Tibial'\n",
      "Processing tissue = 'Brain_Amygdala'\n",
      "Processing tissue = 'Brain_Anterior_cingulate_cortex_BA24'\n",
      "Processing tissue = 'Brain_Caudate_basal_ganglia'\n",
      "Processing tissue = 'Brain_Cerebellar_Hemisphere'\n",
      "Processing tissue = 'Brain_Cortex'\n",
      "Processing tissue = 'Brain_Frontal_Cortex_BA9'\n",
      "Processing tissue = 'Brain_Hippocampus'\n",
      "Processing tissue = 'Brain_Hypothalamus'\n",
      "Processing tissue = 'Brain_Nucleus_accumbens_basal_ganglia'\n",
      "Processing tissue = 'Brain_Putamen_basal_ganglia'\n",
      "Processing tissue = 'Brain_Spinal_cord_cervical_c-1'\n",
      "Processing tissue = 'Breast_Mammary_Tissue'\n",
      "Processing tissue = 'Cells_EBV-transformed_lymphocytes'\n",
      "Processing tissue = 'Cells_Transformed_fibroblasts'\n",
      "Processing tissue = 'Colon_Sigmoid'\n",
      "Processing tissue = 'Colon_Transverse'\n",
      "Processing tissue = 'Esophagus_Gastroesophageal_Junction'\n",
      "Processing tissue = 'Esophagus_Mucosa'\n",
      "Processing tissue = 'Esophagus_Muscularis'\n",
      "Processing tissue = 'Heart_Left_Ventricle'\n",
      "Processing tissue = 'Liver'\n",
      "Processing tissue = 'Lung'\n",
      "Processing tissue = 'Muscle_Skeletal'\n",
      "Processing tissue = 'Nerve_Tibial'\n",
      "Processing tissue = 'Ovary'\n",
      "Processing tissue = 'Pancreas'\n",
      "Processing tissue = 'Pituitary'\n",
      "Processing tissue = 'Prostate'\n",
      "Processing tissue = 'Skin_Not_Sun_Exposed_Suprapubic'\n",
      "Processing tissue = 'Skin_Sun_Exposed_Lower_leg'\n",
      "Processing tissue = 'Small_Intestine_Terminal_Ileum'\n",
      "Processing tissue = 'Spleen'\n",
      "Processing tissue = 'Stomach'\n",
      "Processing tissue = 'Testis'\n",
      "Processing tissue = 'Thyroid'\n",
      "Processing tissue = 'Uterus'\n",
      "Processing tissue = 'Vagina'\n",
      "Processing tissue = 'Whole_Blood'\n"
     ]
    }
   ],
   "source": [
    "#Merge tissue dataframes (UTR isoforms)\n",
    "\n",
    "snp_dfs = []\n",
    "\n",
    "for tissue_type in tissue_types :\n",
    "    \n",
    "    print(\"Processing tissue = '\" + tissue_type + \"'\")\n",
    "    \n",
    "    snp_df = pd.read_csv(\"isoform_\" + tissue_type + version_suffix + \"_SNPs.csv\", sep='\\t')\n",
    "    snp_df['tissue_type'] = tissue_type\n",
    "    \n",
    "    snp_dfs.append(snp_df)\n",
    "\n",
    "snp_df = pd.concat(snp_dfs).copy().reset_index(drop=True)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "merged_df = snp_df\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "snp_dict = {}\n",
    "\n",
    "for i, tissue_type in enumerate(tissue_types) :\n",
    "    \n",
    "    snp_df = pd.read_csv(\"isoform_\" + tissue_type + version_suffix + \"_SNPs.csv\", sep='\\t')\n",
    "    \n",
    "    for _, row in snp_df.iterrows() :\n",
    "        \n",
    "        snp_id = row['snp_id'] + \"__\" + row['gene_id'] + \"__\" + row['transcript_id']\n",
    "        \n",
    "        effect_size = row['effect_size']\n",
    "        \n",
    "        if snp_id not in snp_dict :\n",
    "            snp_dict[snp_id] = {\n",
    "                'effect_size' : []\n",
    "            }\n",
    "        \n",
    "        snp_dict[snp_id]['effect_size'].append(effect_size)\n",
    "\n",
    "snp_ids = []\n",
    "effect_sizes = []\n",
    "n_tissues = []\n",
    "\n",
    "for snp_id in snp_dict :\n",
    "    \n",
    "    snp_ids.append(snp_id)\n",
    "    effect_sizes.append(np.median(snp_dict[snp_id]['effect_size']))\n",
    "    \n",
    "    n_tissues.append(len(snp_dict[snp_id]['effect_size']))\n",
    "\n",
    "median_df = pd.DataFrame({\n",
    "    'extended_snp_id' : snp_ids,\n",
    "    'effect_size' : effect_sizes,\n",
    "    'n_tissues' : n_tissues\n",
    "})\n",
    "median_df = median_df[['extended_snp_id', 'effect_size', 'n_tissues']]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "merged_df['extended_snp_id'] = merged_df['snp_id'] + \"__\" + merged_df['gene_id'] + \"__\" + merged_df['transcript_id']\n",
    "\n",
    "merged_df = merged_df.join(median_df.set_index('extended_snp_id'), on=\"extended_snp_id\", how='inner', rsuffix='_median').copy().reset_index(drop=True)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "merged_df.to_csv(\"isoform_merged\" + version_suffix + \"_SNPs.csv\", sep='\\t')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python [conda env:tensorflow]",
   "language": "python",
   "name": "conda-env-tensorflow-py"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
