{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy\n",
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "pd.options.mode.chained_assignment = None\n",
    "def bf_apadb2(df, val=[   2,   50,  200,  200,  0.3,  50,  50],\n",
    "                  ids=['utr',  'i', 'j', 'rds','rel','up','dn']):\n",
    "    \n",
    "    print('3\\'UTRs         =    ' + str(val[0]))\n",
    "    print('interval (bp) >=   ' + str(val[1]) + ' to ' + str(val[2]) + ' bp')\n",
    "    print('locus          =  -' + str(val[5]) + ' to +' + str(val[6]) + ' bp')\n",
    "    print('read depth    >=  ' + str(val[3]))\n",
    "    print('weakest site  >=    ' + str(val[4]))\n",
    "    print('-----------------------------------')\n",
    "\n",
    "    \n",
    "    print('debug length:   ' + \"{:>12}\".format(\"{:,}\".format(len(df))) + ' genes ')\n",
    "    # filtering/reconstructing\n",
    "    df['genefam']        = df['gene'].str.split('\\\\.').apply(lambda x: x[0])\n",
    "    lost_puppy        = len(df.groupby('genefam')) \n",
    "    df['apa']         = df.groupby('genefam')['genefam'].transform(len)\n",
    "    df['start']       = df['start']-400#200\n",
    "    df['end']         = df['end']+400#200\n",
    "    df                = df.drop('apa', 1)\n",
    "    df['total_reads'] = df.groupby('genefam')['reads'].transform(lambda x: sum(x))\n",
    "    df['rel_use']     = (df['reads'] / df['total_reads']).round(3)\n",
    "    df['interval']    = df.groupby('genefam')['mode'].transform(lambda x: x.max() - x.min())\n",
    "    df                = df.drop('genefam', 1)\n",
    "\n",
    "    print('input :   ' + \"{:>12}\".format(\"{:,}\".format(lost_puppy)) + ' genes ')\n",
    "    print('output:   ' + \"{:>12}\".format(\"{:,}\".format(len(df))) + ' genes ')\n",
    "\n",
    "    return df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3'UTRs         =    20\n",
      "interval (bp) >=   0 to 40000 bp\n",
      "locus          =  -0 to +0 bp\n",
      "read depth    >=  1\n",
      "weakest site  >=    0.0\n",
      "-----------------------------------\n",
      "debug length:         71,829 genes \n",
      "input :         16,786 genes \n",
      "output:         71,829 genes \n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>gene</th>\n",
       "      <th>reads</th>\n",
       "      <th>strand</th>\n",
       "      <th>feature</th>\n",
       "      <th>mode</th>\n",
       "      <th>total_reads</th>\n",
       "      <th>rel_use</th>\n",
       "      <th>interval</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr17</td>\n",
       "      <td>61860</td>\n",
       "      <td>62712</td>\n",
       "      <td>RPH3AL.1</td>\n",
       "      <td>122</td>\n",
       "      <td>-</td>\n",
       "      <td>UTR3</td>\n",
       "      <td>62297</td>\n",
       "      <td>130</td>\n",
       "      <td>0.938</td>\n",
       "      <td>136061</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr17</td>\n",
       "      <td>197956</td>\n",
       "      <td>198758</td>\n",
       "      <td>RPH3AL.2</td>\n",
       "      <td>8</td>\n",
       "      <td>-</td>\n",
       "      <td>Intron</td>\n",
       "      <td>198358</td>\n",
       "      <td>130</td>\n",
       "      <td>0.062</td>\n",
       "      <td>136061</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr17</td>\n",
       "      <td>289358</td>\n",
       "      <td>290185</td>\n",
       "      <td>FAM101B.1</td>\n",
       "      <td>24</td>\n",
       "      <td>-</td>\n",
       "      <td>UTR3</td>\n",
       "      <td>289766</td>\n",
       "      <td>650</td>\n",
       "      <td>0.037</td>\n",
       "      <td>3013</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr17</td>\n",
       "      <td>289535</td>\n",
       "      <td>290347</td>\n",
       "      <td>FAM101B.2</td>\n",
       "      <td>170</td>\n",
       "      <td>-</td>\n",
       "      <td>UTR3</td>\n",
       "      <td>289946</td>\n",
       "      <td>650</td>\n",
       "      <td>0.262</td>\n",
       "      <td>3013</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr17</td>\n",
       "      <td>289616</td>\n",
       "      <td>290421</td>\n",
       "      <td>FAM101B.3</td>\n",
       "      <td>12</td>\n",
       "      <td>-</td>\n",
       "      <td>UTR3</td>\n",
       "      <td>290017</td>\n",
       "      <td>650</td>\n",
       "      <td>0.018</td>\n",
       "      <td>3013</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     chr   start     end       gene  reads strand feature    mode  \\\n",
       "0  chr17   61860   62712   RPH3AL.1    122      -    UTR3   62297   \n",
       "1  chr17  197956  198758   RPH3AL.2      8      -  Intron  198358   \n",
       "2  chr17  289358  290185  FAM101B.1     24      -    UTR3  289766   \n",
       "3  chr17  289535  290347  FAM101B.2    170      -    UTR3  289946   \n",
       "4  chr17  289616  290421  FAM101B.3     12      -    UTR3  290017   \n",
       "\n",
       "   total_reads  rel_use  interval  \n",
       "0          130    0.938    136061  \n",
       "1          130    0.062    136061  \n",
       "2          650    0.037      3013  \n",
       "3          650    0.262      3013  \n",
       "4          650    0.018      3013  "
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "apadb_bed    = 'unprocessed_data/apadb/hg19.apadb_v2_final.bed'#'../../data/genome_hg19/features/hg19.apadb_v2_final.bed'\n",
    "\n",
    "hg19_fai     = 'unprocessed_data/hg19/hg19.fa.fai'\n",
    "hg19_fa      = 'unprocessed_data/hg19/hg19.fa'\n",
    "\n",
    "# read apadb .bed into df\n",
    "df_columns = ['chr', 'start', 'end', 'gene', 'reads', 'strand', 'feature', 'mode']\n",
    "df_bed = pd.read_csv(apadb_bed, sep='\\t', header=None, names=df_columns, usecols=[0,1,2,3,4,5,6,8])\n",
    "\n",
    "filtered = bf_apadb2(df_bed,[20, 0, 40000, 1, 0.00, 0, 0])\n",
    "filtered.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>gene</th>\n",
       "      <th>reads</th>\n",
       "      <th>strand</th>\n",
       "      <th>feature</th>\n",
       "      <th>mode</th>\n",
       "      <th>total_reads</th>\n",
       "      <th>rel_use</th>\n",
       "      <th>interval</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>71824</th>\n",
       "      <td>chr18</td>\n",
       "      <td>77897015</td>\n",
       "      <td>77897830</td>\n",
       "      <td>ADNP2.3</td>\n",
       "      <td>29</td>\n",
       "      <td>+</td>\n",
       "      <td>UTR3</td>\n",
       "      <td>77897423</td>\n",
       "      <td>748</td>\n",
       "      <td>0.039</td>\n",
       "      <td>1604</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>71825</th>\n",
       "      <td>chr18</td>\n",
       "      <td>77897126</td>\n",
       "      <td>77897938</td>\n",
       "      <td>ADNP2.2</td>\n",
       "      <td>5</td>\n",
       "      <td>+</td>\n",
       "      <td>UTR3</td>\n",
       "      <td>77897528</td>\n",
       "      <td>748</td>\n",
       "      <td>0.007</td>\n",
       "      <td>1604</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>71826</th>\n",
       "      <td>chr18</td>\n",
       "      <td>77897805</td>\n",
       "      <td>77898649</td>\n",
       "      <td>ADNP2.1</td>\n",
       "      <td>702</td>\n",
       "      <td>+</td>\n",
       "      <td>UTR3</td>\n",
       "      <td>77898227</td>\n",
       "      <td>748</td>\n",
       "      <td>0.939</td>\n",
       "      <td>1604</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>71827</th>\n",
       "      <td>chr18</td>\n",
       "      <td>77906737</td>\n",
       "      <td>77907543</td>\n",
       "      <td>PARD6G-AS1.2</td>\n",
       "      <td>48</td>\n",
       "      <td>+</td>\n",
       "      <td>Intron</td>\n",
       "      <td>77907140</td>\n",
       "      <td>87</td>\n",
       "      <td>0.552</td>\n",
       "      <td>13154</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>71828</th>\n",
       "      <td>chr18</td>\n",
       "      <td>77919885</td>\n",
       "      <td>77920713</td>\n",
       "      <td>PARD6G-AS1.1</td>\n",
       "      <td>39</td>\n",
       "      <td>+</td>\n",
       "      <td>Intron</td>\n",
       "      <td>77920294</td>\n",
       "      <td>87</td>\n",
       "      <td>0.448</td>\n",
       "      <td>13154</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         chr     start       end          gene  reads strand feature  \\\n",
       "71824  chr18  77897015  77897830       ADNP2.3     29      +    UTR3   \n",
       "71825  chr18  77897126  77897938       ADNP2.2      5      +    UTR3   \n",
       "71826  chr18  77897805  77898649       ADNP2.1    702      +    UTR3   \n",
       "71827  chr18  77906737  77907543  PARD6G-AS1.2     48      +  Intron   \n",
       "71828  chr18  77919885  77920713  PARD6G-AS1.1     39      +  Intron   \n",
       "\n",
       "           mode  total_reads  rel_use  interval  \n",
       "71824  77897423          748    0.039      1604  \n",
       "71825  77897528          748    0.007      1604  \n",
       "71826  77898227          748    0.939      1604  \n",
       "71827  77907140           87    0.552     13154  \n",
       "71828  77920294           87    0.448     13154  "
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "filtered['interval']    = filtered['interval'].astype(int)\n",
    "filtered['mode']        = filtered['mode'].astype(int)\n",
    "filtered['start']       = filtered['start'].astype(int)\n",
    "filtered['end']         = filtered['end'].astype(int)\n",
    "filtered['reads']       = filtered['reads'].astype(int)\n",
    "filtered['total_reads'] = filtered['total_reads'].astype(int)\n",
    "filtered.tail()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chr17  61860   62712   RPH3AL.1   UTR3    -  130  0.938  136061\n",
      "chr17  197956  198758  RPH3AL.2   Intron  -  130  0.062  136061\n",
      "chr17  289358  290185  FAM101B.1  UTR3    -  650  0.037  3013\n",
      "chr17  289535  290347  FAM101B.2  UTR3    -  650  0.262  3013\n",
      "chr17  289616  290421  FAM101B.3  UTR3    -  650  0.018  3013\n",
      "\n",
      ">RPH3AL.1\n",
      "CTGTCTTCAGCTTGAGGAGCTGGGAAGCTCTGGTGGATGCTATGAACTCACTTGCTGAAGAGCAGCGTTCAGGTGCATCCCCAGCCAGGGCACGTGGCTCCCTCAGCCATGAATTCACTTCTCTTCAGGAGGTTTGGCTTGGCATGAAAATACTTCATTCAGAGTATGGGCAAATGCTTCTGGAAAACCCTTCCCTGAAGAGAGAGAACGTGTGTGTGTGTGTCGGTGATCACACCCTCCCATCCTTCCTGCCTCCTGCCCCAAACCCCGGGTTCCTGGGTCTGGAAGGGCCTTCTCTCCAAGCTGGGAGCTCCTGGGCCCCCACCATTCACTTTTTGTCCTTGCTGCTGGCAAACAGTAAAGAAACTCACTTTCCCTGTGGCACGTTATGCTTCAGAATTAAAACAATGAAGATTAAAATTTGCACCGAGCCAGTGTGTTGATCGAAGACCACGATTGCCTGTGTTTCTGAGATGCGTCCATGGAAAAATGGAAAAAACTGTGGTGCGTTGACTTGCTGGAACCCTTCCTAAGCCGCAGTGAAAGGAGGGGCTAGATCTGTGTGTGTTCATGCAGCTCACACACCTGATGCTGAACGGATAAGCAAGGCCATCTACATACAAAAGTACACACAAATAATCAGAATGAGTTCTGCAAACACATACGTATGTATGCAAAAATGTCTTTTAAATGCACAAGgacttctggttatatatggtgcactgaatatattcatattttcctctgtccctcaaaaccccactagaatgacgtaaaggaatgaaaatggcataaagcacaaagacaaaaagaacaggagagaagatggcagtgaatgacaaatatctgaag\n",
      ">RPH3AL.2\n",
      "TATGCAGTGGCTCTTCAGAGCCAGCACAATGGAGGTGACCCTTCTCCCCGTGCACCTTCCAGAGATTAACACGTAAACAAAGAAATAAACACAGATTCCAACTGTGATGCATGTGGTGAAGAAAGAACTGGGGCCCAGTATTTGTTGACCTTTCAGTGTTCTGCACTCCCAAGGGAGGGCCACGTGGTGGGGATGGAACACAGGCATGGACAGTACGTACCTTCCTCCTTTAGCAAACAGGGATGATGGGCGCCTGTGATTAGTCCACACAAAAGCATTCGGGAGGGTATAACACTCAAATGTAATATTAGTGGCGTGGCAAGATTCTTCCGCATTGATGACCATTAGTGTGCTATTTTTGTTTTCTCTATGAAGAAGAAAATTAAAATTTTTTTCATTGCATTTGGTAGCACTTTACTTTTTTttttttttcttgaacagtgttaggtttacagaaaaaattgaacaaaaagtacagagttcccgtctaacctcttacctctttcaatttcccatgattaacatcttgctttggtgaggtacatttgttacaataaatgtggtaatactgatacattattattaactgaaacgtgtagtgtacacttagggttggatcttggaggtgtaggtagtgtggacttcaaccaatgtgtaaggacatgtggggagaataatggctttgctgccttcaaaatcccctgcaatccatgtcctcatccttcctgttccacgcccaaatccctggtctttttactgtctccatagtttggccttttccagaatgtcatagatgtagttg\n",
      ">FAM101B.1\n",
      "AACAGAACCCTTCAGGGTGGGTCAGAGGACGCCATCCACAGTGGATTCGTGTTCGTTTGCAGGTGGAAGCAGTGATTTTTAGGACCCACTGATTAAAAACAAACATTCCCAAGTGTCTCTGAGAGATGCTGTTTATTTGTTAATTAAAAAGCTTTTTTCTCTGTCTTTTAAATTATGGCTTTCATGTAATAAGGATATTTTTAGTGAAAAATTGTTTTCCTTTCAAATTACAGACCTTTTAAAAAAACTTAATTTGAGCGAGTACCTTTTCATTTGACACTTTTCCTGTTTCTAACCTTAGGAAACCAGAATAGCGTTTGGCAGACACGACGTTTTCAGTTTACCTTTGACACCTGCCCCACTCCATTTTGCTTTGTGATGTCTTCATTTAACAATAAATTATCTGAAAAAACAAAACTTAGAGAGATGCTTCTGTTTTTAAAGTAGAATTATGTTTGTTTACGCAAAATGAGAAAAACAGCTCCTCATCTTGAGAAATTTTAAGACGTGATTATATTTAACAGTATTAATCTACAAGTACAAGATTTTCCGAGTGTggctgggcatggtggctcacacctataatcccagcgcttcacaaggccaaggctggaggatcacttgaggccaggagttcgagatcagccttggcaatgcagtgtgaccccatctctacaaaaattaaaaattaAAAAAGAAAGGggccaggcgccgtagctccctcctgtaatcccagcactttgggaggctgaggcgggcggatcacctgaggtcaggagtccgagaccaacctgaccaacctggtgaaaccccgtctctactaaaaa\n",
      ">FAM101B.2\n",
      "GAAGTAAGTAGCAGTGAGCGATTGTGAATGTGTAATGTAAATGGAAAACCGGGTTTTACCGTGTTAAGTTATTCACTAGGGAGCCAGTCGTAGTTCTTTGTAATCCTCTTTCTTCCAAACCTGCTTTGCTGAAAGTTGCAGAAAAGGAAGTGTGTGGAGAGAAACAGAACCCTTCAGGGTGGGTCAGAGGACGCCATCCACAGTGGATTCGTGTTCGTTTGCAGGTGGAAGCAGTGATTTTTAGGACCCACTGATTAAAAACAAACATTCCCAAGTGTCTCTGAGAGATGCTGTTTATTTGTTAATTAAAAAGCTTTTTTCTCTGTCTTTTAAATTATGGCTTTCATGTAATAAGGATATTTTTAGTGAAAAATTGTTTTCCTTTCAAATTACAGACCTTTTAAAAAAACTTAATTTGAGCGAGTACCTTTTCATTTGACACTTTTCCTGTTTCTAACCTTAGGAAACCAGAATAGCGTTTGGCAGACACGACGTTTTCAGTTTACCTTTGACACCTGCCCCACTCCATTTTGCTTTGTGATGTCTTCATTTAACAATAAATTATCTGAAAAAACAAAACTTAGAGAGATGCTTCTGTTTTTAAAGTAGAATTATGTTTGTTTACGCAAAATGAGAAAAACAGCTCCTCATCTTGAGAAATTTTAAGACGTGATTATATTTAACAGTATTAATCTACAAGTACAAGATTTTCCGAGTGTggctgggcatggtggctcacacctataatcccagcgcttcacaaggccaaggctggaggatcacttgaggccaggagttcgagatcagccttg\n",
      ">FAM101B.3\n",
      "TTTTTGTCTTAGGGGGACTGTGTTAAGTTCAGACAAATCATGCTGGGTGTGTAGAGAGTGTGAAATACGTCAGTGAAGTAAGTAGCAGTGAGCGATTGTGAATGTGTAATGTAAATGGAAAACCGGGTTTTACCGTGTTAAGTTATTCACTAGGGAGCCAGTCGTAGTTCTTTGTAATCCTCTTTCTTCCAAACCTGCTTTGCTGAAAGTTGCAGAAAAGGAAGTGTGTGGAGAGAAACAGAACCCTTCAGGGTGGGTCAGAGGACGCCATCCACAGTGGATTCGTGTTCGTTTGCAGGTGGAAGCAGTGATTTTTAGGACCCACTGATTAAAAACAAACATTCCCAAGTGTCTCTGAGAGATGCTGTTTATTTGTTAATTAAAAAGCTTTTTTCTCTGTCTTTTAAATTATGGCTTTCATGTAATAAGGATATTTTTAGTGAAAAATTGTTTTCCTTTCAAATTACAGACCTTTTAAAAAAACTTAATTTGAGCGAGTACCTTTTCATTTGACACTTTTCCTGTTTCTAACCTTAGGAAACCAGAATAGCGTTTGGCAGACACGACGTTTTCAGTTTACCTTTGACACCTGCCCCACTCCATTTTGCTTTGTGATGTCTTCATTTAACAATAAATTATCTGAAAAAACAAAACTTAGAGAGATGCTTCTGTTTTTAAAGTAGAATTATGTTTGTTTACGCAAAATGAGAAAAACAGCTCCTCATCTTGAGAAATTTTAAGACGTGATTATATTTAACAGTATTAATCTACAAGTACAAGATTTTCCGAGTGTggctgggcatgg\n",
      "\n"
     ]
    }
   ],
   "source": [
    "output_id = 'apadb_v2_seqs'\n",
    "\n",
    "# bed\n",
    "output_bed = output_id + '.bed'\n",
    "bed_columns = ['chr', 'start', 'end', 'gene', 'feature', 'strand', 'total_reads', 'rel_use', 'interval']\n",
    "filtered.to_csv(output_bed, sep='\\t', header=False, columns=bed_columns, index=False)\n",
    "\n",
    "# fasta\n",
    "output_fa = output_id + '.fa'\n",
    "!bedtools getfasta -name -s -fi \"$hg19_fa\" -bed \"$output_bed\" -fo \"$output_fa\"\n",
    "\n",
    "# file tops\n",
    "!head -5 \"$output_bed\" | column -t ; echo\n",
    "!head -10 \"$output_fa\" ; echo\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "71689\n",
      "GAAGTAAGTAGCAGTGAGCGATTGTGAATGTGTAATGTAAATGGAAAACCGGGTTTTACCGTGTTAAGTTATTCACTAGGGAGCCAGTCGTAGTTCTTTGTAATCCTCTTTCTTCCAAACCTGCTTTGCTGAAAGTTGCAGAAAAGGAAGTGTGTGGAGAGAAACAGAACCCTTCAGGGTGGGTCAGAGGACGCCATCCACAGTGGATTCGTGTTCGTTTGCAGGTGGAAGCAGTGATTTTTAGGACCCACTGATTAAAAACAAACATTCCCAAGTGTCTCTGAGAGATGCTGTTTATTTGTTAATTAAAAAGCTTTTTTCTCTGTCTTTTAAATTATGGCTTTCATGTAATAAGGATATTTTTAGTGAAAAATTGTTTTCCTTTCAAATTACAGACCTTTTAAAAAAACTTAATTTGAGCGAGTACCTTTTCATTTGACACTTTTCCTGTTTCTAACCTTAGGAAACCAGAATAGCGTTTGGCAGACACGACGTTTTCAGTTTACCTTTGACACCTGCCCCACTCCATTTTGCTTTGTGATGTCTTCATTTAACAATAAATTATCTGAAAAAACAAAACTTAGAGAGATGCTTCTGTTTTTAAAGTAGAATTATGTTTGTTTACGCAAAATGAGAAAAACAGCTCCTCATCTTGAGAAATTTTAAGACGTGATTATATTTAACAGTATTAATCTACAAGTACAAGATTTTCCGAGTGTGGCTGGGCATGGTGGCTCACACCTATAATCCCAGCGCTTCACAAGGCCAAGGCTGGAGGATCACTTGAGGCCAGGAGTTCGAGATCAGCCTTG\n",
      "71689\n",
      "{'chr': 'chr17', 'cut_start': 289935, 'cut_end': 289947, 'count': 170, 'strand': '-', 'type': 'UTR3', 'mirna': 'None', 'seq_start': 289535, 'seq_end': 290347, 'total_count': 650, 'cut_mode': 289946, 'total_ratio': 0.262, 'sort_order': 3}\n",
      "133\n"
     ]
    }
   ],
   "source": [
    "\n",
    "seq_dict = {}\n",
    "with open(output_fa, 'r') as f :\n",
    "    i = 0\n",
    "    seq_id = ''\n",
    "    for line in f :\n",
    "        if i % 2 == 0 :\n",
    "            seq_id = line[1:].rstrip()\n",
    "        else :\n",
    "            seq_dict[seq_id] = line.rstrip().upper()\n",
    "        \n",
    "        i += 1\n",
    "\n",
    "df_dict = {}\n",
    "for index, row in filtered.iterrows() :\n",
    "    df_dict[row['gene']] = {}\n",
    "    \n",
    "    df_dict[row['gene']]['seq_start'] = row['start']\n",
    "    df_dict[row['gene']]['seq_end'] = row['end']\n",
    "    df_dict[row['gene']]['count'] = row['reads']\n",
    "    df_dict[row['gene']]['total_count'] = row['total_reads']\n",
    "    df_dict[row['gene']]['cut_mode'] = row['mode']\n",
    "    df_dict[row['gene']]['total_ratio'] = row['rel_use']\n",
    "    df_dict[row['gene']]['sort_order'] = index\n",
    "    \n",
    "\n",
    "\n",
    "print(len(seq_dict))\n",
    "print(seq_dict['FAM101B.2'])\n",
    "\n",
    "gene_dict = {}\n",
    "\n",
    "site_dict = {}\n",
    "n_mismatch = 0\n",
    "\n",
    "with open('unprocessed_data/apadb/hg19.apadb_v2_final.bed', 'r') as f :\n",
    "    for line in f :\n",
    "        lineparts = line.rstrip().split('\\t')\n",
    "        site_id = lineparts[3]\n",
    "        \n",
    "        site_dict[site_id] = {}\n",
    "        \n",
    "        site_dict[site_id]['chr'] = lineparts[0]\n",
    "        site_dict[site_id]['cut_start'] = int(lineparts[1])\n",
    "        site_dict[site_id]['cut_end'] = int(lineparts[2])\n",
    "        site_dict[site_id]['count'] = int(lineparts[4])\n",
    "        site_dict[site_id]['strand'] = lineparts[5]\n",
    "        site_dict[site_id]['type'] = lineparts[6]\n",
    "        site_dict[site_id]['mirna'] = lineparts[9]\n",
    "        \n",
    "        if df_dict[site_id]['count'] != site_dict[site_id]['count'] :\n",
    "            n_mismatch += 1\n",
    "        \n",
    "        site_dict[site_id]['seq_start'] = df_dict[site_id]['seq_start']\n",
    "        site_dict[site_id]['seq_end'] = df_dict[site_id]['seq_end']\n",
    "        site_dict[site_id]['total_count'] = df_dict[site_id]['total_count']\n",
    "        site_dict[site_id]['cut_mode'] = df_dict[site_id]['cut_mode']\n",
    "        site_dict[site_id]['total_ratio'] = df_dict[site_id]['total_ratio']\n",
    "        site_dict[site_id]['sort_order'] = df_dict[site_id]['sort_order']\n",
    "        \n",
    "        gene_id = site_id.split('.')[0]\n",
    "        site_num = int(site_id.split('.')[1])\n",
    "        \n",
    "        if gene_id not in gene_dict :\n",
    "            gene_dict[gene_id] = 0\n",
    "        gene_dict[gene_id] += 1\n",
    "\n",
    "print(len(site_dict))\n",
    "print(site_dict['FAM101B.2'])\n",
    "\n",
    "print(n_mismatch)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "#Make Valid PAS lookup hierarchy\n",
    "\n",
    "cano_pas1 = 'AATAAA'\n",
    "cano_pas2 = 'ATTAAA'\n",
    "\n",
    "valid_pas = []\n",
    "\n",
    "valid_pas.append({})\n",
    "valid_pas[0]['AATAAA'] = True\n",
    "\n",
    "valid_pas.append({})\n",
    "valid_pas[1]['ATTAAA'] = True\n",
    "\n",
    "valid_pas.append({})\n",
    "valid_pas[2]['AGTAAA'] = True\n",
    "valid_pas[2]['TATAAA'] = True\n",
    "valid_pas[2]['CATAAA'] = True\n",
    "valid_pas[2]['GATAAA'] = True\n",
    "\n",
    "valid_pas.append({})\n",
    "for pos in range(0, 6) :\n",
    "    for base in ['A', 'C', 'G', 'T'] :\n",
    "        valid_pas[3][cano_pas1[:pos] + base + cano_pas1[pos+1:]] = True\n",
    "\n",
    "valid_pas.append({})\n",
    "for pos1 in range(0, 6) :\n",
    "    for pos2 in range(pos1 + 1, 6) :\n",
    "        for base1 in ['A', 'C', 'G', 'T'] :\n",
    "            for base2 in ['A', 'C', 'G', 'T'] :\n",
    "                valid_pas[4][cano_pas1[:pos1] + base1 + cano_pas1[pos1+1:pos2] + base2 + cano_pas1[pos2+1:]] = True\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "def align_seq(seq, cut_start, cut_end, before_cut = 30, after_cut = 5) :\n",
    "    \n",
    "    align_j = cut_start - 25\n",
    "    aligned = -1\n",
    "    \n",
    "    for i in range(0, len(valid_pas)) :\n",
    "        for j in range(cut_start - before_cut, cut_start + after_cut) :\n",
    "\n",
    "            candidate_pas = seq[j:j+6]\n",
    "\n",
    "            if candidate_pas in valid_pas[i] :\n",
    "                align_j = j\n",
    "                aligned = i\n",
    "                \n",
    "                if j >= cut_start :\n",
    "                    break\n",
    "        if aligned != -1 :\n",
    "            break\n",
    "    \n",
    "    aligned_seq = (seq[align_j-50:])[:186]\n",
    "    aligned_seq_long = (seq[align_j-100:])[:186 + 50 + 50]\n",
    "    \n",
    "    if len(aligned_seq_long) != 186 + 50 + 50 :\n",
    "        print('WARNING! Aligned seq not long enough. Length = ' + str(len(aligned_seq_long)))\n",
    "    \n",
    "    \n",
    "    aligned_wide_seq = (seq[align_j-125:])[:256]\n",
    "    aligned_wide_seq_long = (seq[align_j-175:])[:256 + 50 + 50]\n",
    "    \n",
    "    if len(aligned_wide_seq_long) != 256 + 50 + 50 :\n",
    "        print('WARNING! Aligned seq not long enough. Length = ' + str(len(aligned_wide_seq_long)))\n",
    "    \n",
    "    return aligned_seq, aligned_seq_long, aligned, (align_j - cut_start), aligned_wide_seq, aligned_wide_seq_long\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "         gene   gene_id  sitenum  num_sites  pas  \\\n",
      "53932  ABCB10  ABCB10.5        1          5    0   \n",
      "53931  ABCB10  ABCB10.4        2          5    0   \n",
      "53930  ABCB10  ABCB10.3        3          5    2   \n",
      "53929  ABCB10  ABCB10.2        4          5    1   \n",
      "53928  ABCB10  ABCB10.1        5          5    1   \n",
      "\n",
      "                                                     seq  \\\n",
      "53932  GTTAAAGATTGAAGCTATTGTCAAATGACAACTTTAAAAAGGCAAT...   \n",
      "53931  CTATTTCATGAAAAGCATGGAATATTATATTTTATTGTTCATAATT...   \n",
      "53930  CATAATTAATGAATAAAATTGATATGAATGAATATAGTGTTCTTTG...   \n",
      "53929  TCTGATACATGATGTTCAATTTTATCTTTAGGTAATATTTTATATC...   \n",
      "53928  TCAGGAATAAAGAAAAGACTAACATTACACATATCCAAAAACATGT...   \n",
      "\n",
      "                                                 seq_ext  \\\n",
      "53932  TGGAATATTTTAATTAATATAGCATGGCACCTCATTTTCTTTTGCC...   \n",
      "53931  CATGAACTAAGCATTTATTAGTTCCCTGATTAGACTGGAAGAAGAA...   \n",
      "53930  AGAAGAAACCACTATTTCATGAAAAGCATGGAATATTATATTTTAT...   \n",
      "53929  TGTGCCATAGAAGTATTTACGAAATTGCATTTCATTGTTATGTTTT...   \n",
      "53928  TATAAAACTTCTCACTACATTGTTTCTTAGTAGAATTTGGCTGTGG...   \n",
      "\n",
      "                                                wide_seq  \\\n",
      "53932  TCAGGTTTTGTATTTTCTTTTCTTGTGGAATATTTTAATTAATATA...   \n",
      "53931  AATGTAAATCAAATGGAAGTTTTCCCATGAACTAAGCATTTATTAG...   \n",
      "53930  TTATTAGTTCCCTGATTAGACTGGAAGAAGAAACCACTATTTCATG...   \n",
      "53929  GGTTGAATCTGAGGAAAATAATCCTTGTGCCATAGAAGTATTTACG...   \n",
      "53928  ACCAAAGATGCAGTCTGTCATTTCTTATAAAACTTCTCACTACATT...   \n",
      "\n",
      "                                            wide_seq_ext  count  total_count  \\\n",
      "53932  GAGTTTTAATAATTGTAACTTTTTAAATGTCTATAGCACTGAAGTT...     10          564   \n",
      "53931  AAGTGCTTTTTCTCCATGGATGAGGCTAGACCCTAAGAAGTAATTA...     73          564   \n",
      "53930  GTAATTAAGTCAATGTAAATCAAATGGAAGTTTTCCCATGAACTAA...      6          564   \n",
      "53929  TGTGTTCTTTATAAAGTGTGATTTTCAGAAAGCAAACAACACAATT...    467          564   \n",
      "53928  TAGTGAACTTTATCTGTGTCTGTCACTTTTTTTTTTTTTATGACCC...      8          564   \n",
      "\n",
      "       ratio    pas_pos  cut_start    cut_end   cut_mode chrom strand  \\\n",
      "53932  0.018  229653571  229653537  229653551  229653538  chr1      -   \n",
      "53931  0.129  229652609  229652589  229652595  229652591  chr1      -   \n",
      "53930  0.011  229652570  229652539  229652568  229652540  chr1      -   \n",
      "53929  0.828  229652351  229652327  229652337  229652330  chr1      -   \n",
      "53928  0.014  229652165  229652139  229652140  229652140  chr1      -   \n",
      "\n",
      "                                                   mirna  site_type  \n",
      "53932  miR-590-3p,miR-183,miR-219-5p/508/508-3p/4782-...       UTR3  \n",
      "53931   miR-590-3p,miR-183,miR-219-5p/508/508-3p/4782-3p       UTR3  \n",
      "53930                                 miR-590-3p,miR-183       UTR3  \n",
      "53929                                               None       UTR3  \n",
      "53928                                               None  Extension  \n",
      "      gene gene_id  sitenum  num_sites  pas  \\\n",
      "16166  UTY   UTY.3        1          3    0   \n",
      "16165  UTY   UTY.2        2          3    1   \n",
      "16164  UTY   UTY.1        3          3    1   \n",
      "71009  ZFY   ZFY.2        1          2    0   \n",
      "71010  ZFY   ZFY.1        2          2    0   \n",
      "\n",
      "                                                     seq  \\\n",
      "16166  AAAAATGTAAGCAAAATTACAGTATGTAAAACACACATTGCTAATG...   \n",
      "16165  TAAGATGTAAATAATCTTATATATTCATAAGTGTTATAAAATCTCA...   \n",
      "16164  GGAGGAAAGTATTAAAAAATCTATATCGCTAGTAAATTGTAATAAG...   \n",
      "71009  GTAAATGTGTATTTGAGAATATGCAAAAATAAAAATAATATTTTAC...   \n",
      "71010  ATACCAATACTAAGTTAACTCTTTCTTTAAAACACGCAGTTTTAAG...   \n",
      "\n",
      "                                                 seq_ext  \\\n",
      "16166  CCCAAAGTGCTGGGATTACAGGTGTGAGCCACCATGCCCGGTAAAC...   \n",
      "16165  CTGTTGACATTAAATGATATCAGCAGTCCAGAAGTCTTGGCAAAAT...   \n",
      "16164  CTTTGAAATGCTATACTGTCCTGCTTGTACAACATGGTTTGGGGTG...   \n",
      "71009  AAATACTTAAAATAACTATGAAGAATTCATATAGGAAGTGTGATAC...   \n",
      "71010  GCAAAAATAAAAATAATATTTTACTGTTAATAAATGTTTACTTGTA...   \n",
      "\n",
      "                                                wide_seq  \\\n",
      "16166  TCAGGTGGTCTGCTTGCCTCAGCATCCCAAAGTGCTGGGATTACAG...   \n",
      "16165  ATTCAGTATTTTTTTCAAAAGAATTCTGTTGACATTAAATGATATC...   \n",
      "16164  GACTTATTTTTTTAAAGCTAGATGACTTTGAAATGCTATACTGTCC...   \n",
      "71009  TGTCCCTTCCTCTGTACAAATTGTAAAATACTTAAAATAACTATGA...   \n",
      "71010  ATTGTAAATGTGTATTTGAGAATATGCAAAAATAAAAATAATATTT...   \n",
      "\n",
      "                                            wide_seq_ext  count  total_count  \\\n",
      "16166  TAGAGACAGGGTTTCACCATGTTGGCCAGGCTTGTCTTGAACTCCT...     16           44   \n",
      "16165  TAAGTGCCTTCTGAAACTGCTGCAGTTTCTCTTTGGGGGTATTGGT...      7           44   \n",
      "16164  TTATGGAAAAGAGATTCAGAATGATAACCTCTTTTAGAGAACAAAT...     21           44   \n",
      "71009  CAAGGACATAAATTATTATTTGGTTGAAAAATAGCCTATTAGATAT...     48           54   \n",
      "71010  GTAAAATACTTAAAATAACTATGAAGAATTCATATAGGAAGTGTGA...      6           54   \n",
      "\n",
      "       ratio   pas_pos  cut_start   cut_end  cut_mode chrom strand  \\\n",
      "16166  0.364  15434937   15434909  15434918  15434915  chrY      -   \n",
      "16165  0.159  15409410   15409388  15409392  15409390  chrY      -   \n",
      "16164  0.477  15360340   15360318  15360338  15360319  chrY      -   \n",
      "71009  0.889   2850524    2850522   2850547   2850525  chrY      +   \n",
      "71010  0.111   2850596    2850610   2850623   2850618  chrY      +   \n",
      "\n",
      "                                                   mirna  site_type  \n",
      "16166  miR-290-5p/292-5p/371-5p/293,miR-590-3p,miR-14...       UTR3  \n",
      "16165  miR-290-5p/292-5p/371-5p/293,miR-590-3p,miR-14...       UTR3  \n",
      "16164                                               None       UTR3  \n",
      "71009                                               None       UTR3  \n",
      "71010                                               None  Extension  \n",
      "Total number of members: 71689\n"
     ]
    }
   ],
   "source": [
    "#Global APADB dataframe generation\n",
    "\n",
    "gene = []\n",
    "\n",
    "prox_id = []\n",
    "prox_seq = []\n",
    "prox_seq_ext = []\n",
    "prox_wide_seq = []\n",
    "prox_wide_seq_ext = []\n",
    "prox_count = []\n",
    "total_count = []\n",
    "prox_ratio = []\n",
    "\n",
    "prox_mirna = []\n",
    "\n",
    "cut_start = []\n",
    "cut_end = []\n",
    "cut_mode = []\n",
    "\n",
    "prox_type = []\n",
    "prox_pas = []\n",
    "prox_sitenum = []\n",
    "num_sites = []\n",
    "\n",
    "prox_chrom = []\n",
    "prox_strand = []\n",
    "\n",
    "pas_pos = []\n",
    "\n",
    "for site_id in site_dict :\n",
    "    \n",
    "    gene_id = site_id.split('.')[0]\n",
    "    site_num = int(site_id.split('.')[1])\n",
    "    \n",
    "    gene.append(gene_id)\n",
    "    \n",
    "    p_id = site_id\n",
    "    \n",
    "    p_seq = seq_dict[p_id]\n",
    "    p_count = site_dict[p_id]['count']\n",
    "    p_total_count = site_dict[p_id]['total_count']\n",
    "    p_usage = site_dict[p_id]['total_ratio']\n",
    "    p_mirna = site_dict[p_id]['mirna']\n",
    "    \n",
    "    p_start = site_dict[p_id]['cut_start']\n",
    "    p_end = site_dict[p_id]['cut_end']\n",
    "    p_mode = site_dict[p_id]['cut_mode']\n",
    "    \n",
    "    p_type = site_dict[p_id]['type']\n",
    "    \n",
    "    p_seq, p_seq_long, p_aligned, p_shift, p_wide_seq, p_wide_seq_long = align_seq(p_seq, 400, len(p_seq) - 800, before_cut = 35, after_cut = 5) #200, ..., 400\n",
    "    \n",
    "    if site_dict[p_id]['strand'] == '+' :\n",
    "        pas_pos.append(p_shift + p_start)\n",
    "    else :\n",
    "        pas_pos.append(-p_shift + p_end)\n",
    "    \n",
    "    prox_id.append(p_id)\n",
    "    prox_seq.append(p_seq)\n",
    "    prox_seq_ext.append(p_seq_long)\n",
    "    \n",
    "    prox_wide_seq.append(p_wide_seq)\n",
    "    prox_wide_seq_ext.append(p_wide_seq_long)\n",
    "    \n",
    "    prox_count.append(p_count)\n",
    "    total_count.append(p_total_count)\n",
    "    prox_ratio.append(p_usage)\n",
    "    \n",
    "    prox_mirna.append(p_mirna)\n",
    "    \n",
    "    cut_start.append(p_start)\n",
    "    cut_end.append(p_end)\n",
    "    cut_mode.append(p_mode)\n",
    "    \n",
    "    prox_type.append(p_type)\n",
    "    \n",
    "    prox_pas.append(p_aligned)\n",
    "    \n",
    "    prox_sitenum.append(gene_dict[gene_id] - site_num + 1)\n",
    "    \n",
    "    num_sites.append(gene_dict[gene_id])\n",
    "    \n",
    "    prox_chrom.append(site_dict[p_id]['chr'])\n",
    "    prox_strand.append(site_dict[p_id]['strand'])\n",
    "\n",
    "\n",
    "df = pd.DataFrame({\n",
    "        'gene' : gene,\n",
    "        'gene_id'  : prox_id,\n",
    "        'sitenum'  : prox_sitenum,\n",
    "        'num_sites' : num_sites,\n",
    "        'pas'  : prox_pas,\n",
    "        'seq'  : prox_seq,\n",
    "        'seq_ext'  : prox_seq_ext,\n",
    "        'wide_seq'  : prox_wide_seq,\n",
    "        'wide_seq_ext'  : prox_wide_seq_ext,\n",
    "        'count'  : prox_count,\n",
    "        'total_count'  : total_count,\n",
    "        'ratio' : prox_ratio,\n",
    "        'cut_start'  : cut_start,\n",
    "        'cut_end'  : cut_end,\n",
    "        'cut_mode'  : cut_mode,\n",
    "        'chrom' : prox_chrom,\n",
    "        'strand' : prox_strand,\n",
    "        'mirna'  : prox_mirna,\n",
    "        'site_type'  : prox_type,\n",
    "        'pas_pos' : pas_pos\n",
    "    })\n",
    "\n",
    "df = df[['gene',\n",
    "        'gene_id',\n",
    "        'sitenum',\n",
    "        'num_sites',\n",
    "        'pas',\n",
    "        'seq',\n",
    "        'seq_ext',\n",
    "        'wide_seq',\n",
    "        'wide_seq_ext',\n",
    "        'count',\n",
    "        'total_count',\n",
    "        'ratio',\n",
    "        'pas_pos',\n",
    "        'cut_start',\n",
    "        'cut_end',\n",
    "        'cut_mode',\n",
    "        'chrom',\n",
    "        'strand',\n",
    "        'mirna',\n",
    "        'site_type']]\n",
    "\n",
    "df = df.sort_values(by=['chrom', 'gene', 'sitenum'])\n",
    "\n",
    "print(df.head())\n",
    "print(df.tail())\n",
    "\n",
    "print('Total number of members: ' + str(len(df)))\n",
    "\n",
    "\n",
    "df.to_csv('apadb_processed_v2.csv', header=True, index=False, sep=',')\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "         gene chrom strand proximal_id distal_id  proximal_sitenum  \\\n",
      "41557  ABCB10  chr1      -    ABCB10.5  ABCB10.4                 1   \n",
      "41556  ABCB10  chr1      -    ABCB10.4  ABCB10.3                 2   \n",
      "41555  ABCB10  chr1      -    ABCB10.3  ABCB10.2                 3   \n",
      "41554  ABCB10  chr1      -    ABCB10.2  ABCB10.1                 4   \n",
      "21825   ABCD3  chr1      +     ABCD3.5   ABCD3.4                 1   \n",
      "\n",
      "       distal_sitenum  proximal_pas  distal_pas  \\\n",
      "41557               2             0           0   \n",
      "41556               3             0           2   \n",
      "41555               4             2           1   \n",
      "41554               5             1           1   \n",
      "21825               2             0           0   \n",
      "\n",
      "                                            proximal_seq    ...      \\\n",
      "41557  GTTAAAGATTGAAGCTATTGTCAAATGACAACTTTAAAAAGGCAAT...    ...       \n",
      "41556  CTATTTCATGAAAAGCATGGAATATTATATTTTATTGTTCATAATT...    ...       \n",
      "41555  CATAATTAATGAATAAAATTGATATGAATGAATATAGTGTTCTTTG...    ...       \n",
      "41554  TCTGATACATGATGTTCAATTTTATCTTTAGGTAATATTTTATATC...    ...       \n",
      "21825  TTCGAGACAAGCCTGGACAAAAAGCGAGACCCGCTTCTTTAAAAAA...    ...       \n",
      "\n",
      "      distal_cutsite_length  proximal_cut_start  proximal_cut_mode  \\\n",
      "41557                     6           229653537          229653538   \n",
      "41556                    29           229652589          229652591   \n",
      "41555                    10           229652539          229652540   \n",
      "41554                     1           229652327          229652330   \n",
      "21825                    27            94944268           94944278   \n",
      "\n",
      "       proximal_cut_end  distal_cut_start distal_cut_mode distal_cut_end  \\\n",
      "41557         229653551         229652589       229652591      229652595   \n",
      "41556         229652595         229652539       229652540      229652568   \n",
      "41555         229652568         229652327       229652330      229652337   \n",
      "41554         229652337         229652139       229652140      229652140   \n",
      "21825          94944278          94944642        94944644       94944669   \n",
      "\n",
      "       proximal_type  distal_type  num_sites  \n",
      "41557           UTR3         UTR3          5  \n",
      "41556           UTR3         UTR3          5  \n",
      "41555           UTR3         UTR3          5  \n",
      "41554           UTR3    Extension          5  \n",
      "21825         Intron       Intron          5  \n",
      "\n",
      "[5 rows x 28 columns]\n",
      "          gene chrom strand proximal_id  distal_id  proximal_sitenum  \\\n",
      "54369  TXLNG2P  chrY      +   TXLNG2P.3  TXLNG2P.2                 6   \n",
      "54370  TXLNG2P  chrY      +   TXLNG2P.2  TXLNG2P.1                 7   \n",
      "12106      UTY  chrY      -       UTY.3      UTY.2                 1   \n",
      "12105      UTY  chrY      -       UTY.2      UTY.1                 2   \n",
      "54355      ZFY  chrY      +       ZFY.2      ZFY.1                 1   \n",
      "\n",
      "       distal_sitenum  proximal_pas  distal_pas  \\\n",
      "54369               7             3           1   \n",
      "54370               8             1           3   \n",
      "12106               2             0           1   \n",
      "12105               3             1           1   \n",
      "54355               2             0           0   \n",
      "\n",
      "                                            proximal_seq    ...      \\\n",
      "54369  CTTGTTCTGTTGTTTTTGGCATACTGTTTATGTTAGATACACTGTG...    ...       \n",
      "54370  TGTTTCTGGACACTGTAGCTGACCAAGAAAATGTTTAAATGTTGCG...    ...       \n",
      "12106  AAAAATGTAAGCAAAATTACAGTATGTAAAACACACATTGCTAATG...    ...       \n",
      "12105  TAAGATGTAAATAATCTTATATATTCATAAGTGTTATAAAATCTCA...    ...       \n",
      "54355  GTAAATGTGTATTTGAGAATATGCAAAAATAAAAATAATATTTTAC...    ...       \n",
      "\n",
      "      distal_cutsite_length  proximal_cut_start  proximal_cut_mode  \\\n",
      "54369                     9            21767383           21767386   \n",
      "54370                     1            21767695           21767701   \n",
      "12106                     4            15434909           15434915   \n",
      "12105                    20            15409388           15409390   \n",
      "54355                    13             2850522            2850525   \n",
      "\n",
      "       proximal_cut_end  distal_cut_start distal_cut_mode distal_cut_end  \\\n",
      "54369          21767398          21767695        21767701       21767704   \n",
      "54370          21767704          21768157        21768159       21768158   \n",
      "12106          15434918          15409388        15409390       15409392   \n",
      "12105          15409392          15360318        15360319       15360338   \n",
      "54355           2850547           2850610         2850618        2850623   \n",
      "\n",
      "       proximal_type  distal_type  num_sites  \n",
      "54369           UTR5         UTR5          8  \n",
      "54370           UTR5         UTR5          8  \n",
      "12106           UTR3         UTR3          3  \n",
      "12105           UTR3         UTR3          3  \n",
      "54355           UTR3    Extension          2  \n",
      "\n",
      "[5 rows x 28 columns]\n",
      "Total number of members: 54903\n"
     ]
    }
   ],
   "source": [
    "chrom = []\n",
    "strand = []\n",
    "gene = []\n",
    "\n",
    "prox_id = []\n",
    "dist_id = []\n",
    "\n",
    "prox_seq = []\n",
    "dist_seq = []\n",
    "\n",
    "prox_count = []\n",
    "dist_count = []\n",
    "total_count = []\n",
    "\n",
    "site_distance = []\n",
    "\n",
    "prox_mirna = []\n",
    "dist_mirna = []\n",
    "\n",
    "prox_cutsite_length = []\n",
    "dist_cutsite_length = []\n",
    "\n",
    "prox_cut_start = []\n",
    "prox_cut_mode = []\n",
    "prox_cut_end = []\n",
    "dist_cut_start = []\n",
    "dist_cut_mode = []\n",
    "dist_cut_end = []\n",
    "\n",
    "prox_type = []\n",
    "dist_type = []\n",
    "\n",
    "prox_pas = []\n",
    "dist_pas = []\n",
    "\n",
    "prox_sitenum = []\n",
    "dist_sitenum = []\n",
    "\n",
    "num_sites = []\n",
    "\n",
    "for site_id in site_dict :\n",
    "    \n",
    "    gene_id = site_id.split('.')[0]\n",
    "    site_num = int(site_id.split('.')[1])\n",
    "    \n",
    "    if gene_id + '.' + str(site_num-1) not in site_dict :\n",
    "        continue\n",
    "    \n",
    "    p_id = site_id\n",
    "    d_id = gene_id + '.' + str(site_num-1)\n",
    "    \n",
    "    gene.append(gene_id)\n",
    "    \n",
    "    p_seq = seq_dict[p_id]\n",
    "    p_count = site_dict[p_id]['count']\n",
    "    p_mirna = site_dict[p_id]['mirna']\n",
    "    p_cutsite_length = numpy.abs(site_dict[p_id]['cut_end'] - site_dict[p_id]['cut_start'])\n",
    "    p_cut_start = site_dict[p_id]['cut_start']\n",
    "    p_cut_mode = site_dict[p_id]['cut_mode']\n",
    "    p_cut_end = site_dict[p_id]['cut_end']\n",
    "    p_type = site_dict[p_id]['type']\n",
    "    \n",
    "    d_seq = seq_dict[d_id]\n",
    "    d_count = site_dict[d_id]['count']\n",
    "    d_mirna = site_dict[d_id]['mirna']\n",
    "    d_cutsite_length = numpy.abs(site_dict[d_id]['cut_end'] - site_dict[d_id]['cut_start'])\n",
    "    d_cut_start = site_dict[d_id]['cut_start']\n",
    "    d_cut_mode = site_dict[d_id]['cut_mode']\n",
    "    d_cut_end = site_dict[d_id]['cut_end']\n",
    "    d_type = site_dict[d_id]['type']\n",
    "    \n",
    "    site_dist = numpy.abs(site_dict[p_id]['cut_end'] - site_dict[d_id]['cut_end'])\n",
    "    \n",
    "    \n",
    "    p_seq, _, p_aligned, _, _, _ = align_seq(p_seq, 400, len(p_seq) - 800, before_cut = 30, after_cut = 5) #200, ..., 400\n",
    "    d_seq, _, d_aligned, _, _, _ = align_seq(d_seq, 400, len(d_seq) - 800, before_cut = 30, after_cut = 5)\n",
    "    \n",
    "    chrom.append(site_dict[p_id]['chr'])\n",
    "    strand.append(site_dict[p_id]['strand'])\n",
    "    \n",
    "    prox_id.append(p_id)\n",
    "    dist_id.append(d_id)\n",
    "    \n",
    "    prox_seq.append(p_seq)\n",
    "    dist_seq.append(d_seq)\n",
    "    \n",
    "    prox_count.append(p_count)\n",
    "    dist_count.append(d_count)\n",
    "    total_count.append(p_count + d_count)\n",
    "    \n",
    "    site_distance.append(site_dist)\n",
    "    \n",
    "    prox_mirna.append(p_mirna)\n",
    "    dist_mirna.append(d_mirna)\n",
    "    \n",
    "    prox_cutsite_length.append(p_cutsite_length)\n",
    "    dist_cutsite_length.append(d_cutsite_length)\n",
    "    \n",
    "    prox_cut_start.append(p_cut_start)\n",
    "    prox_cut_mode.append(p_cut_mode)\n",
    "    prox_cut_end.append(p_cut_end)\n",
    "    dist_cut_start.append(d_cut_start)\n",
    "    dist_cut_mode.append(d_cut_mode)\n",
    "    dist_cut_end.append(d_cut_end)\n",
    "    \n",
    "    prox_type.append(p_type)\n",
    "    dist_type.append(d_type)\n",
    "    \n",
    "    prox_pas.append(p_aligned)\n",
    "    dist_pas.append(d_aligned)\n",
    "    \n",
    "    prox_sitenum.append(gene_dict[gene_id] - site_num + 1)\n",
    "    dist_sitenum.append(gene_dict[gene_id] - site_num + 2)\n",
    "    \n",
    "    num_sites.append(gene_dict[gene_id])\n",
    "\n",
    "\n",
    "df = pd.DataFrame({\n",
    "        'gene' : gene,\n",
    "        'chrom' : chrom,\n",
    "        'strand' : strand,\n",
    "        'proximal_id'  : prox_id,\n",
    "        'distal_id'  : dist_id,\n",
    "        'proximal_sitenum'  : prox_sitenum,\n",
    "        'distal_sitenum'  : dist_sitenum,\n",
    "        'proximal_pas'  : prox_pas,\n",
    "        'distal_pas'  : dist_pas,\n",
    "        'proximal_seq'  : prox_seq,\n",
    "        'distal_seq'  : dist_seq,\n",
    "        'proximal_count'  : prox_count,\n",
    "        'distal_count'  : dist_count,\n",
    "        'total_count'  : total_count,\n",
    "        'site_distance'  : site_distance,\n",
    "        'proximal_mirna'  : prox_mirna,\n",
    "        'distal_mirna'  : dist_mirna,\n",
    "        'proximal_cutsite_length'  : prox_cutsite_length,\n",
    "        'distal_cutsite_length'  : dist_cutsite_length,\n",
    "        'proximal_cut_start'  : prox_cut_start,\n",
    "        'proximal_cut_mode'  : prox_cut_mode,\n",
    "        'proximal_cut_end'  : prox_cut_end,\n",
    "        'distal_cut_start'  : dist_cut_start,\n",
    "        'distal_cut_mode'  : dist_cut_mode,\n",
    "        'distal_cut_end'  : dist_cut_end,\n",
    "        'proximal_type'  : prox_type,\n",
    "        'distal_type'  : dist_type,\n",
    "        'num_sites' : num_sites\n",
    "    })\n",
    "\n",
    "df = df[['gene',\n",
    "        'chrom',\n",
    "        'strand',\n",
    "        'proximal_id',\n",
    "        'distal_id',\n",
    "        'proximal_sitenum',\n",
    "        'distal_sitenum',\n",
    "        'proximal_pas',\n",
    "        'distal_pas',\n",
    "        'proximal_seq',\n",
    "        'distal_seq',\n",
    "        'proximal_count',\n",
    "        'distal_count',\n",
    "        'total_count',\n",
    "        'site_distance',\n",
    "        'proximal_mirna',\n",
    "        'distal_mirna',\n",
    "        'proximal_cutsite_length',\n",
    "        'distal_cutsite_length',\n",
    "        'proximal_cut_start',\n",
    "        'proximal_cut_mode',\n",
    "        'proximal_cut_end',\n",
    "        'distal_cut_start',\n",
    "        'distal_cut_mode',\n",
    "        'distal_cut_end',\n",
    "        'proximal_type',\n",
    "        'distal_type',\n",
    "        'num_sites']]\n",
    "\n",
    "df = df.sort_values(by=['chrom', 'gene', 'proximal_sitenum'])\n",
    "\n",
    "\n",
    "print(df.head())\n",
    "print(df.tail())\n",
    "\n",
    "print('Total number of members: ' + str(len(df)))\n",
    "\n",
    "\n",
    "df.to_csv('apadb_processed_v2_pairs.csv', header=True, index=False, sep=',')\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "anaconda-cloud": {},
  "kernelspec": {
   "display_name": "Python [conda env:tensorflow]",
   "language": "python",
   "name": "conda-env-tensorflow-py"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 1
}
